// floating-ui/dist/floating-ui.mjs
/**
* @vue/shared v3.4.19
* (c) 2018-present Yuxi (Evan) You and Vue contributors
* @license MIT
**/
function Ye(e, t) {
  const n = new Set(e.split(","));
  return t ? (o) => n.has(o.toLowerCase()) : (o) => n.has(o);
}
const X = {}.NODE_ENV !== "production" ? Object.freeze({}) : {}, Ft = {}.NODE_ENV !== "production" ? Object.freeze([]) : [], se = () => {
}, As = () => !1, mn = (e) => e.charCodeAt(0) === 111 && e.charCodeAt(1) === 110 && // uppercase letter
(e.charCodeAt(2) > 122 || e.charCodeAt(2) < 97), jn = (e) => e.startsWith("onUpdate:"), ne = Object.assign, oi = (e, t) => {
  const n = e.indexOf(t);
  n > -1 && e.splice(n, 1);
}, sl = Object.prototype.hasOwnProperty, z = (e, t) => sl.call(e, t), R = Array.isArray, yt = (e) => eo(e) === "[object Map]", Rs = (e) => eo(e) === "[object Set]", M = (e) => typeof e == "function", oe = (e) => typeof e == "string", Wt = (e) => typeof e == "symbol", q = (e) => e !== null && typeof e == "object", ii = (e) => (q(e) || M(e)) && M(e.then) && M(e.catch), Ms = Object.prototype.toString, eo = (e) => Ms.call(e), si = (e) => eo(e).slice(8, -1), Is = (e) => eo(e) === "[object Object]", ri = (e) => oe(e) && e !== "NaN" && e[0] !== "-" && "" + parseInt(e, 10) === e, Zt = /* @__PURE__ */ Ye(
  // the leading comma is intentional so empty string "" is also included
  ",key,ref,ref_for,ref_key,onVnodeBeforeMount,onVnodeMounted,onVnodeBeforeUpdate,onVnodeUpdated,onVnodeBeforeUnmount,onVnodeUnmounted"
), rl = /* @__PURE__ */ Ye(
  "bind,cloak,else-if,else,for,html,if,model,on,once,pre,show,slot,text,memo"
), to = (e) => {
  const t = /* @__PURE__ */ Object.create(null);
  return (n) => t[n] || (t[n] = e(n));
}, ll = /-(\w)/g, Le = to((e) => e.replace(ll, (t, n) => n ? n.toUpperCase() : "")), cl = /\B([A-Z])/g, Ge = to(
  (e) => e.replace(cl, "-$1").toLowerCase()
), Dt = to((e) => e.charAt(0).toUpperCase() + e.slice(1)), _t = to((e) => e ? `on${Dt(e)}` : ""), rt = (e, t) => !Object.is(e, t), Gt = (e, t) => {
  for (let n = 0; n < e.length; n++)
    e[n](t);
}, kn = (e, t, n) => {
  Object.defineProperty(e, t, {
    configurable: !0,
    enumerable: !1,
    value: n
  });
}, ul = (e) => {
  const t = parseFloat(e);
  return isNaN(t) ? e : t;
};
let Mi;
const li = () => Mi || (Mi = typeof globalThis < "u" ? globalThis : typeof self < "u" ? self : typeof window < "u" ? window : typeof global < "u" ? global : {});
function Ht(e) {
  if (R(e)) {
    const t = {};
    for (let n = 0; n < e.length; n++) {
      const o = e[n], i = oe(o) ? pl(o) : Ht(o);
      if (i)
        for (const s in i)
          t[s] = i[s];
    }
    return t;
  } else if (oe(e) || q(e))
    return e;
}
const al = /;(?![^(]*\))/g, fl = /:([^]+)/, dl = /\/\*[^]*?\*\//g;
function pl(e) {
  const t = {};
  return e.replace(dl, "").split(al).forEach((n) => {
    if (n) {
      const o = n.split(fl);
      o.length > 1 && (t[o[0].trim()] = o[1].trim());
    }
  }), t;
}
function gn(e) {
  let t = "";
  if (oe(e))
    t = e;
  else if (R(e))
    for (let n = 0; n < e.length; n++) {
      const o = gn(e[n]);
      o && (t += o + " ");
    }
  else if (q(e))
    for (const n in e)
      e[n] && (t += n + " ");
  return t.trim();
}
const hl = "html,body,base,head,link,meta,style,title,address,article,aside,footer,header,hgroup,h1,h2,h3,h4,h5,h6,nav,section,div,dd,dl,dt,figcaption,figure,picture,hr,img,li,main,ol,p,pre,ul,a,b,abbr,bdi,bdo,br,cite,code,data,dfn,em,i,kbd,mark,q,rp,rt,ruby,s,samp,small,span,strong,sub,sup,time,u,var,wbr,area,audio,map,track,video,embed,object,param,source,canvas,script,noscript,del,ins,caption,col,colgroup,table,thead,tbody,td,th,tr,button,datalist,fieldset,form,input,label,legend,meter,optgroup,option,output,progress,select,textarea,details,dialog,menu,summary,template,blockquote,iframe,tfoot", ml = "svg,animate,animateMotion,animateTransform,circle,clipPath,color-profile,defs,desc,discard,ellipse,feBlend,feColorMatrix,feComponentTransfer,feComposite,feConvolveMatrix,feDiffuseLighting,feDisplacementMap,feDistantLight,feDropShadow,feFlood,feFuncA,feFuncB,feFuncG,feFuncR,feGaussianBlur,feImage,feMerge,feMergeNode,feMorphology,feOffset,fePointLight,feSpecularLighting,feSpotLight,feTile,feTurbulence,filter,foreignObject,g,hatch,hatchpath,image,line,linearGradient,marker,mask,mesh,meshgradient,meshpatch,meshrow,metadata,mpath,path,pattern,polygon,polyline,radialGradient,rect,set,solidcolor,stop,switch,symbol,text,textPath,title,tspan,unknown,use,view", gl = "annotation,annotation-xml,maction,maligngroup,malignmark,math,menclose,merror,mfenced,mfrac,mfraction,mglyph,mi,mlabeledtr,mlongdiv,mmultiscripts,mn,mo,mover,mpadded,mphantom,mprescripts,mroot,mrow,ms,mscarries,mscarry,msgroup,msline,mspace,msqrt,msrow,mstack,mstyle,msub,msubsup,msup,mtable,mtd,mtext,mtr,munder,munderover,none,semantics", _l = /* @__PURE__ */ Ye(hl), yl = /* @__PURE__ */ Ye(ml), wl = /* @__PURE__ */ Ye(gl), El = "itemscope,allowfullscreen,formnovalidate,ismap,nomodule,novalidate,readonly", bl = /* @__PURE__ */ Ye(El);
function Ls(e) {
  return !!e || e === "";
}
const vl = (e) => oe(e) ? e : e == null ? "" : R(e) || q(e) && (e.toString === Ms || !M(e.toString)) ? JSON.stringify(e, Fs, 2) : String(e), Fs = (e, t) => t && t.__v_isRef ? Fs(e, t.value) : yt(t) ? {
  [`Map(${t.size})`]: [...t.entries()].reduce(
    (n, [o, i], s) => (n[vo(o, s) + " =>"] = i, n),
    {}
  )
} : Rs(t) ? {
  [`Set(${t.size})`]: [...t.values()].map((n) => vo(n))
} : Wt(t) ? vo(t) : q(t) && !R(t) && !Is(t) ? String(t) : t, vo = (e, t = "") => {
  var n;
  return Wt(e) ? `Symbol(${(n = e.description) != null ? n : t})` : e;
};
/**
* @vue/reactivity v3.4.19
* (c) 2018-present Yuxi (Evan) You and Vue contributors
* @license MIT
**/
function rn(e, ...t) {
  console.warn(`[Vue warn] ${e}`, ...t);
}
let $e;
class Nl {
  constructor(t = !1) {
    this.detached = t, this._active = !0, this.effects = [], this.cleanups = [], this.parent = $e, !t && $e && (this.index = ($e.scopes || ($e.scopes = [])).push(
      this
    ) - 1);
  }
  get active() {
    return this._active;
  }
  run(t) {
    if (this._active) {
      const n = $e;
      try {
        return $e = this, t();
      } finally {
        $e = n;
      }
    } else
      ({}).NODE_ENV !== "production" && rn("cannot run an inactive effect scope.");
  }
  /**
   * This should only be called on non-detached scopes
   * @internal
   */
  on() {
    $e = this;
  }
  /**
   * This should only be called on non-detached scopes
   * @internal
   */
  off() {
    $e = this.parent;
  }
  stop(t) {
    if (this._active) {
      let n, o;
      for (n = 0, o = this.effects.length; n < o; n++)
        this.effects[n].stop();
      for (n = 0, o = this.cleanups.length; n < o; n++)
        this.cleanups[n]();
      if (this.scopes)
        for (n = 0, o = this.scopes.length; n < o; n++)
          this.scopes[n].stop(!0);
      if (!this.detached && this.parent && !t) {
        const i = this.parent.scopes.pop();
        i && i !== this && (this.parent.scopes[this.index] = i, i.index = this.index);
      }
      this.parent = void 0, this._active = !1;
    }
  }
}
function Ol(e, t = $e) {
  t && t.active && t.effects.push(e);
}
function xl() {
  return $e;
}
let wt;
class ci {
  constructor(t, n, o, i) {
    this.fn = t, this.trigger = n, this.scheduler = o, this.active = !0, this.deps = [], this._dirtyLevel = 4, this._trackId = 0, this._runnings = 0, this._shouldSchedule = !1, this._depsLength = 0, Ol(this, i);
  }
  get dirty() {
    if (this._dirtyLevel === 2 || this._dirtyLevel === 3) {
      this._dirtyLevel = 1, ft();
      for (let t = 0; t < this._depsLength; t++) {
        const n = this.deps[t];
        if (n.computed && ($l(n.computed), this._dirtyLevel >= 4))
          break;
      }
      this._dirtyLevel === 1 && (this._dirtyLevel = 0), dt();
    }
    return this._dirtyLevel >= 4;
  }
  set dirty(t) {
    this._dirtyLevel = t ? 4 : 0;
  }
  run() {
    if (this._dirtyLevel = 0, !this.active)
      return this.fn();
    let t = st, n = wt;
    try {
      return st = !0, wt = this, this._runnings++, Ii(this), this.fn();
    } finally {
      Li(this), this._runnings--, wt = n, st = t;
    }
  }
  stop() {
    var t;
    this.active && (Ii(this), Li(this), (t = this.onStop) == null || t.call(this), this.active = !1);
  }
}
function $l(e) {
  return e.value;
}
function Ii(e) {
  e._trackId++, e._depsLength = 0;
}
function Li(e) {
  if (e.deps.length > e._depsLength) {
    for (let t = e._depsLength; t < e.deps.length; t++)
      Hs(e.deps[t], e);
    e.deps.length = e._depsLength;
  }
}
function Hs(e, t) {
  const n = e.get(t);
  n !== void 0 && t._trackId !== n && (e.delete(t), e.size === 0 && e.cleanup());
}
let st = !0, Ao = 0;
const js = [];
function ft() {
  js.push(st), st = !1;
}
function dt() {
  const e = js.pop();
  st = e === void 0 ? !0 : e;
}
function ui() {
  Ao++;
}
function ai() {
  for (Ao--; !Ao && Ro.length; )
    Ro.shift()();
}
function ks(e, t, n) {
  var o;
  if (t.get(e) !== e._trackId) {
    t.set(e, e._trackId);
    const i = e.deps[e._depsLength];
    i !== t ? (i && Hs(i, e), e.deps[e._depsLength++] = t) : e._depsLength++, {}.NODE_ENV !== "production" && ((o = e.onTrack) == null || o.call(e, ne({ effect: e }, n)));
  }
}
const Ro = [];
function Bs(e, t, n) {
  var o;
  ui();
  for (const i of e.keys()) {
    let s;
    i._dirtyLevel < t && (s ?? (s = e.get(i) === i._trackId)) && (i._shouldSchedule || (i._shouldSchedule = i._dirtyLevel === 0), i._dirtyLevel = t), i._shouldSchedule && (s ?? (s = e.get(i) === i._trackId)) && ({}.NODE_ENV !== "production" && ((o = i.onTrigger) == null || o.call(i, ne({ effect: i }, n))), i.trigger(), (!i._runnings || i.allowRecurse) && i._dirtyLevel !== 2 && (i._shouldSchedule = !1, i.scheduler && Ro.push(i.scheduler)));
  }
  ai();
}
const zs = (e, t) => {
  const n = /* @__PURE__ */ new Map();
  return n.cleanup = e, n.computed = t, n;
}, Mo = /* @__PURE__ */ new WeakMap(), Et = Symbol({}.NODE_ENV !== "production" ? "iterate" : ""), Io = Symbol({}.NODE_ENV !== "production" ? "Map key iterate" : "");
function ae(e, t, n) {
  if (st && wt) {
    let o = Mo.get(e);
    o || Mo.set(e, o = /* @__PURE__ */ new Map());
    let i = o.get(n);
    i || o.set(n, i = zs(() => o.delete(n))), ks(
      wt,
      i,
      {}.NODE_ENV !== "production" ? {
        target: e,
        type: t,
        key: n
      } : void 0
    );
  }
}
function Ie(e, t, n, o, i, s) {
  const r = Mo.get(e);
  if (!r)
    return;
  let l = [];
  if (t === "clear")
    l = [...r.values()];
  else if (n === "length" && R(e)) {
    const u = Number(o);
    r.forEach((a, p) => {
      (p === "length" || !Wt(p) && p >= u) && l.push(a);
    });
  } else
    switch (n !== void 0 && l.push(r.get(n)), t) {
      case "add":
        R(e) ? ri(n) && l.push(r.get("length")) : (l.push(r.get(Et)), yt(e) && l.push(r.get(Io)));
        break;
      case "delete":
        R(e) || (l.push(r.get(Et)), yt(e) && l.push(r.get(Io)));
        break;
      case "set":
        yt(e) && l.push(r.get(Et));
        break;
    }
  ui();
  for (const u of l)
    u && Bs(
      u,
      4,
      {}.NODE_ENV !== "production" ? {
        target: e,
        type: t,
        key: n,
        newValue: o,
        oldValue: i,
        oldTarget: s
      } : void 0
    );
  ai();
}
const Cl = /* @__PURE__ */ Ye("__proto__,__v_isRef,__isVue"), Ws = new Set(
  /* @__PURE__ */ Object.getOwnPropertyNames(Symbol).filter((e) => e !== "arguments" && e !== "caller").map((e) => Symbol[e]).filter(Wt)
), Fi = /* @__PURE__ */ Dl();
function Dl() {
  const e = {};
  return ["includes", "indexOf", "lastIndexOf"].forEach((t) => {
    e[t] = function(...n) {
      const o = k(this);
      for (let s = 0, r = this.length; s < r; s++)
        ae(o, "get", s + "");
      const i = o[t](...n);
      return i === -1 || i === !1 ? o[t](...n.map(k)) : i;
    };
  }), ["push", "pop", "shift", "unshift", "splice"].forEach((t) => {
    e[t] = function(...n) {
      ft(), ui();
      const o = k(this)[t].apply(this, n);
      return ai(), dt(), o;
    };
  }), e;
}
function Tl(e) {
  const t = k(this);
  return ae(t, "has", e), t.hasOwnProperty(e);
}
class Us {
  constructor(t = !1, n = !1) {
    this._isReadonly = t, this._shallow = n;
  }
  get(t, n, o) {
    const i = this._isReadonly, s = this._shallow;
    if (n === "__v_isReactive")
      return !i;
    if (n === "__v_isReadonly")
      return i;
    if (n === "__v_isShallow")
      return s;
    if (n === "__v_raw")
      return o === (i ? s ? Zs : Xs : s ? Js : Ys).get(t) || // receiver is not the reactive proxy, but has the same prototype
      // this means the reciever is a user proxy of the reactive proxy
      Object.getPrototypeOf(t) === Object.getPrototypeOf(o) ? t : void 0;
    const r = R(t);
    if (!i) {
      if (r && z(Fi, n))
        return Reflect.get(Fi, n, o);
      if (n === "hasOwnProperty")
        return Tl;
    }
    const l = Reflect.get(t, n, o);
    return (Wt(n) ? Ws.has(n) : Cl(n)) || (i || ae(t, "get", n), s) ? l : fe(l) ? r && ri(n) ? l : l.value : q(l) ? i ? Qs(l) : di(l) : l;
  }
}
class Ks extends Us {
  constructor(t = !1) {
    super(!1, t);
  }
  set(t, n, o, i) {
    let s = t[n];
    if (!this._shallow) {
      const u = lt(s);
      if (!vt(o) && !lt(o) && (s = k(s), o = k(o)), !R(t) && fe(s) && !fe(o))
        return u ? !1 : (s.value = o, !0);
    }
    const r = R(t) && ri(n) ? Number(n) < t.length : z(t, n), l = Reflect.set(t, n, o, i);
    return t === k(i) && (r ? rt(o, s) && Ie(t, "set", n, o, s) : Ie(t, "add", n, o)), l;
  }
  deleteProperty(t, n) {
    const o = z(t, n), i = t[n], s = Reflect.deleteProperty(t, n);
    return s && o && Ie(t, "delete", n, void 0, i), s;
  }
  has(t, n) {
    const o = Reflect.has(t, n);
    return (!Wt(n) || !Ws.has(n)) && ae(t, "has", n), o;
  }
  ownKeys(t) {
    return ae(
      t,
      "iterate",
      R(t) ? "length" : Et
    ), Reflect.ownKeys(t);
  }
}
class Gs extends Us {
  constructor(t = !1) {
    super(!0, t);
  }
  set(t, n) {
    return {}.NODE_ENV !== "production" && rn(
      `Set operation on key "${String(n)}" failed: target is readonly.`,
      t
    ), !0;
  }
  deleteProperty(t, n) {
    return {}.NODE_ENV !== "production" && rn(
      `Delete operation on key "${String(n)}" failed: target is readonly.`,
      t
    ), !0;
  }
}
const Sl = /* @__PURE__ */ new Ks(), Vl = /* @__PURE__ */ new Gs(), Pl = /* @__PURE__ */ new Ks(
  !0
), Al = /* @__PURE__ */ new Gs(!0), fi = (e) => e, no = (e) => Reflect.getPrototypeOf(e);
function xn(e, t, n = !1, o = !1) {
  e = e.__v_raw;
  const i = k(e), s = k(t);
  n || (rt(t, s) && ae(i, "get", t), ae(i, "get", s));
  const { has: r } = no(i), l = o ? fi : n ? pi : ln;
  if (r.call(i, t))
    return l(e.get(t));
  if (r.call(i, s))
    return l(e.get(s));
  e !== i && e.get(t);
}
function $n(e, t = !1) {
  const n = this.__v_raw, o = k(n), i = k(e);
  return t || (rt(e, i) && ae(o, "has", e), ae(o, "has", i)), e === i ? n.has(e) : n.has(e) || n.has(i);
}
function Cn(e, t = !1) {
  return e = e.__v_raw, !t && ae(k(e), "iterate", Et), Reflect.get(e, "size", e);
}
function Hi(e) {
  e = k(e);
  const t = k(this);
  return no(t).has.call(t, e) || (t.add(e), Ie(t, "add", e, e)), this;
}
function ji(e, t) {
  t = k(t);
  const n = k(this), { has: o, get: i } = no(n);
  let s = o.call(n, e);
  s ? {}.NODE_ENV !== "production" && qs(n, o, e) : (e = k(e), s = o.call(n, e));
  const r = i.call(n, e);
  return n.set(e, t), s ? rt(t, r) && Ie(n, "set", e, t, r) : Ie(n, "add", e, t), this;
}
function ki(e) {
  const t = k(this), { has: n, get: o } = no(t);
  let i = n.call(t, e);
  i ? {}.NODE_ENV !== "production" && qs(t, n, e) : (e = k(e), i = n.call(t, e));
  const s = o ? o.call(t, e) : void 0, r = t.delete(e);
  return i && Ie(t, "delete", e, void 0, s), r;
}
function Bi() {
  const e = k(this), t = e.size !== 0, n = {}.NODE_ENV !== "production" ? yt(e) ? new Map(e) : new Set(e) : void 0, o = e.clear();
  return t && Ie(e, "clear", void 0, void 0, n), o;
}
function Dn(e, t) {
  return function(o, i) {
    const s = this, r = s.__v_raw, l = k(r), u = t ? fi : e ? pi : ln;
    return !e && ae(l, "iterate", Et), r.forEach((a, p) => o.call(i, u(a), u(p), s));
  };
}
function Tn(e, t, n) {
  return function(...o) {
    const i = this.__v_raw, s = k(i), r = yt(s), l = e === "entries" || e === Symbol.iterator && r, u = e === "keys" && r, a = i[e](...o), p = n ? fi : t ? pi : ln;
    return !t && ae(
      s,
      "iterate",
      u ? Io : Et
    ), {
      // iterator protocol
      next() {
        const { value: f, done: h } = a.next();
        return h ? { value: f, done: h } : {
          value: l ? [p(f[0]), p(f[1])] : p(f),
          done: h
        };
      },
      // iterable protocol
      [Symbol.iterator]() {
        return this;
      }
    };
  };
}
function Ze(e) {
  return function(...t) {
    if ({}.NODE_ENV !== "production") {
      const n = t[0] ? `on key "${t[0]}" ` : "";
      console.warn(
        `${Dt(e)} operation ${n}failed: target is readonly.`,
        k(this)
      );
    }
    return e === "delete" ? !1 : e === "clear" ? void 0 : this;
  };
}
function Rl() {
  const e = {
    get(s) {
      return xn(this, s);
    },
    get size() {
      return Cn(this);
    },
    has: $n,
    add: Hi,
    set: ji,
    delete: ki,
    clear: Bi,
    forEach: Dn(!1, !1)
  }, t = {
    get(s) {
      return xn(this, s, !1, !0);
    },
    get size() {
      return Cn(this);
    },
    has: $n,
    add: Hi,
    set: ji,
    delete: ki,
    clear: Bi,
    forEach: Dn(!1, !0)
  }, n = {
    get(s) {
      return xn(this, s, !0);
    },
    get size() {
      return Cn(this, !0);
    },
    has(s) {
      return $n.call(this, s, !0);
    },
    add: Ze("add"),
    set: Ze("set"),
    delete: Ze("delete"),
    clear: Ze("clear"),
    forEach: Dn(!0, !1)
  }, o = {
    get(s) {
      return xn(this, s, !0, !0);
    },
    get size() {
      return Cn(this, !0);
    },
    has(s) {
      return $n.call(this, s, !0);
    },
    add: Ze("add"),
    set: Ze("set"),
    delete: Ze("delete"),
    clear: Ze("clear"),
    forEach: Dn(!0, !0)
  };
  return ["keys", "values", "entries", Symbol.iterator].forEach((s) => {
    e[s] = Tn(
      s,
      !1,
      !1
    ), n[s] = Tn(
      s,
      !0,
      !1
    ), t[s] = Tn(
      s,
      !1,
      !0
    ), o[s] = Tn(
      s,
      !0,
      !0
    );
  }), [
    e,
    n,
    t,
    o
  ];
}
const [
  Ml,
  Il,
  Ll,
  Fl
] = /* @__PURE__ */ Rl();
function oo(e, t) {
  const n = t ? e ? Fl : Ll : e ? Il : Ml;
  return (o, i, s) => i === "__v_isReactive" ? !e : i === "__v_isReadonly" ? e : i === "__v_raw" ? o : Reflect.get(
    z(n, i) && i in o ? n : o,
    i,
    s
  );
}
const Hl = {
  get: /* @__PURE__ */ oo(!1, !1)
}, jl = {
  get: /* @__PURE__ */ oo(!1, !0)
}, kl = {
  get: /* @__PURE__ */ oo(!0, !1)
}, Bl = {
  get: /* @__PURE__ */ oo(!0, !0)
};
function qs(e, t, n) {
  const o = k(n);
  if (o !== n && t.call(e, o)) {
    const i = si(e);
    console.warn(
      `Reactive ${i} contains both the raw and reactive versions of the same object${i === "Map" ? " as keys" : ""}, which can lead to inconsistencies. Avoid differentiating between the raw and reactive versions of an object and only use the reactive version if possible.`
    );
  }
}
const Ys = /* @__PURE__ */ new WeakMap(), Js = /* @__PURE__ */ new WeakMap(), Xs = /* @__PURE__ */ new WeakMap(), Zs = /* @__PURE__ */ new WeakMap();
function zl(e) {
  switch (e) {
    case "Object":
    case "Array":
      return 1;
    case "Map":
    case "Set":
    case "WeakMap":
    case "WeakSet":
      return 2;
    default:
      return 0;
  }
}
function Wl(e) {
  return e.__v_skip || !Object.isExtensible(e) ? 0 : zl(si(e));
}
function di(e) {
  return lt(e) ? e : io(
    e,
    !1,
    Sl,
    Hl,
    Ys
  );
}
function Ul(e) {
  return io(
    e,
    !1,
    Pl,
    jl,
    Js
  );
}
function Qs(e) {
  return io(
    e,
    !0,
    Vl,
    kl,
    Xs
  );
}
function It(e) {
  return io(
    e,
    !0,
    Al,
    Bl,
    Zs
  );
}
function io(e, t, n, o, i) {
  if (!q(e))
    return {}.NODE_ENV !== "production" && console.warn(`value cannot be made reactive: ${String(e)}`), e;
  if (e.__v_raw && !(t && e.__v_isReactive))
    return e;
  const s = i.get(e);
  if (s)
    return s;
  const r = Wl(e);
  if (r === 0)
    return e;
  const l = new Proxy(
    e,
    r === 2 ? o : n
  );
  return i.set(e, l), l;
}
function bt(e) {
  return lt(e) ? bt(e.__v_raw) : !!(e && e.__v_isReactive);
}
function lt(e) {
  return !!(e && e.__v_isReadonly);
}
function vt(e) {
  return !!(e && e.__v_isShallow);
}
function Lo(e) {
  return bt(e) || lt(e);
}
function k(e) {
  const t = e && e.__v_raw;
  return t ? k(t) : e;
}
function er(e) {
  return Object.isExtensible(e) && kn(e, "__v_skip", !0), e;
}
const ln = (e) => q(e) ? di(e) : e, pi = (e) => q(e) ? Qs(e) : e, Kl = "Computed is still dirty after getter evaluation, likely because a computed is mutating its own dependency in its getter. State mutations in computed getters should be avoided.  Check the docs for more details: https://vuejs.org/guide/essentials/computed.html#getters-should-be-side-effect-free";
class tr {
  constructor(t, n, o, i) {
    this._setter = n, this.dep = void 0, this.__v_isRef = !0, this.__v_isReadonly = !1, this.effect = new ci(
      () => t(this._value),
      () => Pn(
        this,
        this.effect._dirtyLevel === 2 ? 2 : 3
      )
    ), this.effect.computed = this, this.effect.active = this._cacheable = !i, this.__v_isReadonly = o;
  }
  get value() {
    const t = k(this);
    return (!t._cacheable || t.effect.dirty) && rt(t._value, t._value = t.effect.run()) && Pn(t, 4), nr(t), t.effect._dirtyLevel >= 2 && ({}.NODE_ENV !== "production" && rn(Kl), Pn(t, 2)), t._value;
  }
  set value(t) {
    this._setter(t);
  }
  // #region polyfill _dirty for backward compatibility third party code for Vue <= 3.3.x
  get _dirty() {
    return this.effect.dirty;
  }
  set _dirty(t) {
    this.effect.dirty = t;
  }
  // #endregion
}
function Gl(e, t, n = !1) {
  let o, i;
  const s = M(e);
  s ? (o = e, i = {}.NODE_ENV !== "production" ? () => {
    rn("Write operation failed: computed value is readonly");
  } : se) : (o = e.get, i = e.set);
  const r = new tr(o, i, s || !i, n);
  return {}.NODE_ENV !== "production" && t && !n && (r.effect.onTrack = t.onTrack, r.effect.onTrigger = t.onTrigger), r;
}
function nr(e) {
  var t;
  st && wt && (e = k(e), ks(
    wt,
    (t = e.dep) != null ? t : e.dep = zs(
      () => e.dep = void 0,
      e instanceof tr ? e : void 0
    ),
    {}.NODE_ENV !== "production" ? {
      target: e,
      type: "get",
      key: "value"
    } : void 0
  ));
}
function Pn(e, t = 4, n) {
  e = k(e);
  const o = e.dep;
  o && Bs(
    o,
    t,
    {}.NODE_ENV !== "production" ? {
      target: e,
      type: "set",
      key: "value",
      newValue: n
    } : void 0
  );
}
function fe(e) {
  return !!(e && e.__v_isRef === !0);
}
function Fo(e) {
  return ql(e, !1);
}
function ql(e, t) {
  return fe(e) ? e : new Yl(e, t);
}
class Yl {
  constructor(t, n) {
    this.__v_isShallow = n, this.dep = void 0, this.__v_isRef = !0, this._rawValue = n ? t : k(t), this._value = n ? t : ln(t);
  }
  get value() {
    return nr(this), this._value;
  }
  set value(t) {
    const n = this.__v_isShallow || vt(t) || lt(t);
    t = n ? t : k(t), rt(t, this._rawValue) && (this._rawValue = t, this._value = n ? t : ln(t), Pn(this, 4, t));
  }
}
function Jl(e) {
  return fe(e) ? e.value : e;
}
const Xl = {
  get: (e, t, n) => Jl(Reflect.get(e, t, n)),
  set: (e, t, n, o) => {
    const i = e[t];
    return fe(i) && !fe(n) ? (i.value = n, !0) : Reflect.set(e, t, n, o);
  }
};
function or(e) {
  return bt(e) ? e : new Proxy(e, Xl);
}
/**
* @vue/runtime-core v3.4.19
* (c) 2018-present Yuxi (Evan) You and Vue contributors
* @license MIT
**/
const Nt = [];
function An(e) {
  Nt.push(e);
}
function Rn() {
  Nt.pop();
}
function N(e, ...t) {
  ft();
  const n = Nt.length ? Nt[Nt.length - 1].component : null, o = n && n.appContext.config.warnHandler, i = Zl();
  if (o)
    Ue(
      o,
      n,
      11,
      [
        e + t.join(""),
        n && n.proxy,
        i.map(
          ({ vnode: s }) => `at <${fo(n, s.type)}>`
        ).join(`
`),
        i
      ]
    );
  else {
    const s = [`[Vue warn]: ${e}`, ...t];
    i.length && s.push(`
`, ...Ql(i)), console.warn(...s);
  }
  dt();
}
function Zl() {
  let e = Nt[Nt.length - 1];
  if (!e)
    return [];
  const t = [];
  for (; e; ) {
    const n = t[0];
    n && n.vnode === e ? n.recurseCount++ : t.push({
      vnode: e,
      recurseCount: 0
    });
    const o = e.component && e.component.parent;
    e = o && o.vnode;
  }
  return t;
}
function Ql(e) {
  const t = [];
  return e.forEach((n, o) => {
    t.push(...o === 0 ? [] : [`
`], ...ec(n));
  }), t;
}
function ec({ vnode: e, recurseCount: t }) {
  const n = t > 0 ? `... (${t} recursive calls)` : "", o = e.component ? e.component.parent == null : !1, i = ` at <${fo(
    e.component,
    e.type,
    o
  )}`, s = ">" + n;
  return e.props ? [i, ...tc(e.props), s] : [i + s];
}
function tc(e) {
  const t = [], n = Object.keys(e);
  return n.slice(0, 3).forEach((o) => {
    t.push(...ir(o, e[o]));
  }), n.length > 3 && t.push(" ..."), t;
}
function ir(e, t, n) {
  return oe(t) ? (t = JSON.stringify(t), n ? t : [`${e}=${t}`]) : typeof t == "number" || typeof t == "boolean" || t == null ? n ? t : [`${e}=${t}`] : fe(t) ? (t = ir(e, k(t.value), !0), n ? t : [`${e}=Ref<`, t, ">"]) : M(t) ? [`${e}=fn${t.name ? `<${t.name}>` : ""}`] : (t = k(t), n ? t : [`${e}=`, t]);
}
const hi = {
  sp: "serverPrefetch hook",
  bc: "beforeCreate hook",
  c: "created hook",
  bm: "beforeMount hook",
  m: "mounted hook",
  bu: "beforeUpdate hook",
  u: "updated",
  bum: "beforeUnmount hook",
  um: "unmounted hook",
  a: "activated hook",
  da: "deactivated hook",
  ec: "errorCaptured hook",
  rtc: "renderTracked hook",
  rtg: "renderTriggered hook",
  0: "setup function",
  1: "render function",
  2: "watcher getter",
  3: "watcher callback",
  4: "watcher cleanup function",
  5: "native event handler",
  6: "component event handler",
  7: "vnode hook",
  8: "directive hook",
  9: "transition hook",
  10: "app errorHandler",
  11: "app warnHandler",
  12: "ref function",
  13: "async component loader",
  14: "scheduler flush. This is likely a Vue internals bug. Please open an issue at https://github.com/vuejs/core ."
};
function Ue(e, t, n, o) {
  try {
    return o ? e(...o) : e();
  } catch (i) {
    _n(i, t, n);
  }
}
function Te(e, t, n, o) {
  if (M(e)) {
    const s = Ue(e, t, n, o);
    return s && ii(s) && s.catch((r) => {
      _n(r, t, n);
    }), s;
  }
  const i = [];
  for (let s = 0; s < e.length; s++)
    i.push(Te(e[s], t, n, o));
  return i;
}
function _n(e, t, n, o = !0) {
  const i = t ? t.vnode : null;
  if (t) {
    let s = t.parent;
    const r = t.proxy, l = {}.NODE_ENV !== "production" ? hi[n] : `https://vuejs.org/error-reference/#runtime-${n}`;
    for (; s; ) {
      const a = s.ec;
      if (a) {
        for (let p = 0; p < a.length; p++)
          if (a[p](e, r, l) === !1)
            return;
      }
      s = s.parent;
    }
    const u = t.appContext.config.errorHandler;
    if (u) {
      Ue(
        u,
        null,
        10,
        [e, r, l]
      );
      return;
    }
  }
  nc(e, n, i, o);
}
function nc(e, t, n, o = !0) {
  if ({}.NODE_ENV !== "production") {
    const i = hi[t];
    if (n && An(n), N(`Unhandled error${i ? ` during execution of ${i}` : ""}`), n && Rn(), o)
      throw e;
    console.error(e);
  } else
    console.error(e);
}
let cn = !1, Ho = !1;
const pe = [];
let Re = 0;
const jt = [];
let ze = null, tt = 0;
const sr = /* @__PURE__ */ Promise.resolve();
let mi = null;
const oc = 100;
function rr(e) {
  const t = mi || sr;
  return e ? t.then(this ? e.bind(this) : e) : t;
}
function ic(e) {
  let t = Re + 1, n = pe.length;
  for (; t < n; ) {
    const o = t + n >>> 1, i = pe[o], s = un(i);
    s < e || s === e && i.pre ? t = o + 1 : n = o;
  }
  return t;
}
function so(e) {
  (!pe.length || !pe.includes(
    e,
    cn && e.allowRecurse ? Re + 1 : Re
  )) && (e.id == null ? pe.push(e) : pe.splice(ic(e.id), 0, e), lr());
}
function lr() {
  !cn && !Ho && (Ho = !0, mi = sr.then(ar));
}
function sc(e) {
  const t = pe.indexOf(e);
  t > Re && pe.splice(t, 1);
}
function cr(e) {
  R(e) ? jt.push(...e) : (!ze || !ze.includes(
    e,
    e.allowRecurse ? tt + 1 : tt
  )) && jt.push(e), lr();
}
function zi(e, t, n = cn ? Re + 1 : 0) {
  for ({}.NODE_ENV !== "production" && (t = t || /* @__PURE__ */ new Map()); n < pe.length; n++) {
    const o = pe[n];
    if (o && o.pre) {
      if (e && o.id !== e.uid || {}.NODE_ENV !== "production" && gi(t, o))
        continue;
      pe.splice(n, 1), n--, o();
    }
  }
}
function ur(e) {
  if (jt.length) {
    const t = [...new Set(jt)].sort(
      (n, o) => un(n) - un(o)
    );
    if (jt.length = 0, ze) {
      ze.push(...t);
      return;
    }
    for (ze = t, {}.NODE_ENV !== "production" && (e = e || /* @__PURE__ */ new Map()), tt = 0; tt < ze.length; tt++)
      ({}).NODE_ENV !== "production" && gi(e, ze[tt]) || ze[tt]();
    ze = null, tt = 0;
  }
}
const un = (e) => e.id == null ? 1 / 0 : e.id, rc = (e, t) => {
  const n = un(e) - un(t);
  if (n === 0) {
    if (e.pre && !t.pre)
      return -1;
    if (t.pre && !e.pre)
      return 1;
  }
  return n;
};
function ar(e) {
  Ho = !1, cn = !0, {}.NODE_ENV !== "production" && (e = e || /* @__PURE__ */ new Map()), pe.sort(rc);
  const t = {}.NODE_ENV !== "production" ? (n) => gi(e, n) : se;
  try {
    for (Re = 0; Re < pe.length; Re++) {
      const n = pe[Re];
      if (n && n.active !== !1) {
        if ({}.NODE_ENV !== "production" && t(n))
          continue;
        Ue(n, null, 14);
      }
    }
  } finally {
    Re = 0, pe.length = 0, ur(e), cn = !1, mi = null, (pe.length || jt.length) && ar(e);
  }
}
function gi(e, t) {
  if (!e.has(t))
    e.set(t, 1);
  else {
    const n = e.get(t);
    if (n > oc) {
      const o = t.ownerInstance, i = o && $i(o.type);
      return _n(
        `Maximum recursive updates exceeded${i ? ` in component <${i}>` : ""}. This means you have a reactive effect that is mutating its own dependencies and thus recursively triggering itself. Possible sources include component template, render function, updated hook or watcher source function.`,
        null,
        10
      ), !0;
    } else
      e.set(t, n + 1);
  }
}
let Ot = !1;
const Mt = /* @__PURE__ */ new Set();
({}).NODE_ENV !== "production" && (li().__VUE_HMR_RUNTIME__ = {
  createRecord: No(fr),
  rerender: No(uc),
  reload: No(ac)
});
const Tt = /* @__PURE__ */ new Map();
function lc(e) {
  const t = e.type.__hmrId;
  let n = Tt.get(t);
  n || (fr(t, e.type), n = Tt.get(t)), n.instances.add(e);
}
function cc(e) {
  Tt.get(e.type.__hmrId).instances.delete(e);
}
function fr(e, t) {
  return Tt.has(e) ? !1 : (Tt.set(e, {
    initialDef: Qt(t),
    instances: /* @__PURE__ */ new Set()
  }), !0);
}
function Qt(e) {
  return jr(e) ? e.__vccOpts : e;
}
function uc(e, t) {
  const n = Tt.get(e);
  n && (n.initialDef.render = t, [...n.instances].forEach((o) => {
    t && (o.render = t, Qt(o.type).render = t), o.renderCache = [], Ot = !0, o.effect.dirty = !0, o.update(), Ot = !1;
  }));
}
function ac(e, t) {
  const n = Tt.get(e);
  if (!n)
    return;
  t = Qt(t), Wi(n.initialDef, t);
  const o = [...n.instances];
  for (const i of o) {
    const s = Qt(i.type);
    Mt.has(s) || (s !== n.initialDef && Wi(s, t), Mt.add(s)), i.appContext.propsCache.delete(i.type), i.appContext.emitsCache.delete(i.type), i.appContext.optionsCache.delete(i.type), i.ceReload ? (Mt.add(s), i.ceReload(t.styles), Mt.delete(s)) : i.parent ? (i.parent.effect.dirty = !0, so(i.parent.update)) : i.appContext.reload ? i.appContext.reload() : typeof window < "u" ? window.location.reload() : console.warn(
      "[HMR] Root or manually mounted instance modified. Full reload required."
    );
  }
  cr(() => {
    for (const i of o)
      Mt.delete(
        Qt(i.type)
      );
  });
}
function Wi(e, t) {
  ne(e, t);
  for (const n in e)
    n !== "__file" && !(n in t) && delete e[n];
}
function No(e) {
  return (t, n) => {
    try {
      return e(t, n);
    } catch (o) {
      console.error(o), console.warn(
        "[HMR] Something went wrong during Vue component hot-reload. Full reload required."
      );
    }
  };
}
let Me, Jt = [], jo = !1;
function yn(e, ...t) {
  Me ? Me.emit(e, ...t) : jo || Jt.push({ event: e, args: t });
}
function dr(e, t) {
  var n, o;
  Me = e, Me ? (Me.enabled = !0, Jt.forEach(({ event: i, args: s }) => Me.emit(i, ...s)), Jt = []) : /* handle late devtools injection - only do this if we are in an actual */ /* browser environment to avoid the timer handle stalling test runner exit */ /* (#4815) */ typeof window < "u" && // some envs mock window but not fully
  window.HTMLElement && // also exclude jsdom
  !((o = (n = window.navigator) == null ? void 0 : n.userAgent) != null && o.includes("jsdom")) ? ((t.__VUE_DEVTOOLS_HOOK_REPLAY__ = t.__VUE_DEVTOOLS_HOOK_REPLAY__ || []).push((s) => {
    dr(s, t);
  }), setTimeout(() => {
    Me || (t.__VUE_DEVTOOLS_HOOK_REPLAY__ = null, jo = !0, Jt = []);
  }, 3e3)) : (jo = !0, Jt = []);
}
function fc(e, t) {
  yn("app:init", e, t, {
    Fragment: we,
    Text: wn,
    Comment: ye,
    Static: Ln
  });
}
function dc(e) {
  yn("app:unmount", e);
}
const pc = /* @__PURE__ */ _i(
  "component:added"
  /* COMPONENT_ADDED */
), pr = /* @__PURE__ */ _i(
  "component:updated"
  /* COMPONENT_UPDATED */
), hc = /* @__PURE__ */ _i(
  "component:removed"
  /* COMPONENT_REMOVED */
), mc = (e) => {
  Me && typeof Me.cleanupBuffer == "function" && // remove the component if it wasn't buffered
  !Me.cleanupBuffer(e) && hc(e);
};
function _i(e) {
  return (t) => {
    yn(
      e,
      t.appContext.app,
      t.uid,
      t.parent ? t.parent.uid : void 0,
      t
    );
  };
}
const gc = /* @__PURE__ */ hr(
  "perf:start"
  /* PERFORMANCE_START */
), _c = /* @__PURE__ */ hr(
  "perf:end"
  /* PERFORMANCE_END */
);
function hr(e) {
  return (t, n, o) => {
    yn(e, t.appContext.app, t.uid, t, n, o);
  };
}
function yc(e, t, n) {
  yn(
    "component:emit",
    e.appContext.app,
    e,
    t,
    n
  );
}
function wc(e, t, ...n) {
  if (e.isUnmounted)
    return;
  const o = e.vnode.props || X;
  if ({}.NODE_ENV !== "production") {
    const {
      emitsOptions: p,
      propsOptions: [f]
    } = e;
    if (p)
      if (!(t in p))
        (!f || !(_t(t) in f)) && N(
          `Component emitted event "${t}" but it is neither declared in the emits option nor as an "${_t(t)}" prop.`
        );
      else {
        const h = p[t];
        M(h) && (h(...n) || N(
          `Invalid event arguments: event validation failed for event "${t}".`
        ));
      }
  }
  let i = n;
  const s = t.startsWith("update:"), r = s && t.slice(7);
  if (r && r in o) {
    const p = `${r === "modelValue" ? "model" : r}Modifiers`, { number: f, trim: h } = o[p] || X;
    h && (i = n.map((g) => oe(g) ? g.trim() : g)), f && (i = n.map(ul));
  }
  if ({}.NODE_ENV !== "production" && yc(e, t, i), {}.NODE_ENV !== "production") {
    const p = t.toLowerCase();
    p !== t && o[_t(p)] && N(
      `Event "${p}" is emitted in component ${fo(
        e,
        e.type
      )} but the handler is registered for "${t}". Note that HTML attributes are case-insensitive and you cannot use v-on to listen to camelCase events when using in-DOM templates. You should probably use "${Ge(
        t
      )}" instead of "${t}".`
    );
  }
  let l, u = o[l = _t(t)] || // also try camelCase event handler (#2249)
  o[l = _t(Le(t))];
  !u && s && (u = o[l = _t(Ge(t))]), u && Te(
    u,
    e,
    6,
    i
  );
  const a = o[l + "Once"];
  if (a) {
    if (!e.emitted)
      e.emitted = {};
    else if (e.emitted[l])
      return;
    e.emitted[l] = !0, Te(
      a,
      e,
      6,
      i
    );
  }
}
function mr(e, t, n = !1) {
  const o = t.emitsCache, i = o.get(e);
  if (i !== void 0)
    return i;
  const s = e.emits;
  let r = {}, l = !1;
  if (!M(e)) {
    const u = (a) => {
      const p = mr(a, t, !0);
      p && (l = !0, ne(r, p));
    };
    !n && t.mixins.length && t.mixins.forEach(u), e.extends && u(e.extends), e.mixins && e.mixins.forEach(u);
  }
  return !s && !l ? (q(e) && o.set(e, null), null) : (R(s) ? s.forEach((u) => r[u] = null) : ne(r, s), q(e) && o.set(e, r), r);
}
function ro(e, t) {
  return !e || !mn(t) ? !1 : (t = t.slice(2).replace(/Once$/, ""), z(e, t[0].toLowerCase() + t.slice(1)) || z(e, Ge(t)) || z(e, t));
}
let ue = null, lo = null;
function Bn(e) {
  const t = ue;
  return ue = e, lo = e && e.type.__scopeId || null, t;
}
function Ec(e) {
  lo = e;
}
function bc() {
  lo = null;
}
const vc = (e) => zn;
function zn(e, t = ue, n) {
  if (!t || e._n)
    return e;
  const o = (...i) => {
    o._d && os(-1);
    const s = Bn(t);
    let r;
    try {
      r = e(...i);
    } finally {
      Bn(s), o._d && os(1);
    }
    return {}.NODE_ENV !== "production" && pr(t), r;
  };
  return o._n = !0, o._c = !0, o._d = !0, o;
}
let ko = !1;
function Wn() {
  ko = !0;
}
function Oo(e) {
  const {
    type: t,
    vnode: n,
    proxy: o,
    withProxy: i,
    props: s,
    propsOptions: [r],
    slots: l,
    attrs: u,
    emit: a,
    render: p,
    renderCache: f,
    data: h,
    setupState: g,
    ctx: O,
    inheritAttrs: $
  } = e;
  let V, P;
  const H = Bn(e);
  ({}).NODE_ENV !== "production" && (ko = !1);
  try {
    if (n.shapeFlag & 4) {
      const I = i || o, K = {}.NODE_ENV !== "production" && g.__isScriptSetup ? new Proxy(I, {
        get(D, Y, J) {
          return N(
            `Property '${String(
              Y
            )}' was accessed via 'this'. Avoid using 'this' in templates.`
          ), Reflect.get(D, Y, J);
        }
      }) : I;
      V = Ce(
        p.call(
          K,
          I,
          f,
          s,
          g,
          h,
          O
        )
      ), P = u;
    } else {
      const I = t;
      ({}).NODE_ENV !== "production" && u === s && Wn(), V = Ce(
        I.length > 1 ? I(
          s,
          {}.NODE_ENV !== "production" ? {
            get attrs() {
              return Wn(), u;
            },
            slots: l,
            emit: a
          } : { attrs: u, slots: l, emit: a }
        ) : I(
          s,
          null
          /* we know it doesn't need it */
        )
      ), P = t.props ? u : Nc(u);
    }
  } catch (I) {
    nn.length = 0, _n(I, e, 1), V = ge(ye);
  }
  let A = V, j;
  if ({}.NODE_ENV !== "production" && V.patchFlag > 0 && V.patchFlag & 2048 && ([A, j] = gr(V)), P && $ !== !1) {
    const I = Object.keys(P), { shapeFlag: K } = A;
    if (I.length) {
      if (K & 7)
        r && I.some(jn) && (P = Oc(
          P,
          r
        )), A = ct(A, P);
      else if ({}.NODE_ENV !== "production" && !ko && A.type !== ye) {
        const D = Object.keys(u), Y = [], J = [];
        for (let Z = 0, ie = D.length; Z < ie; Z++) {
          const Q = D[Z];
          mn(Q) ? jn(Q) || Y.push(Q[2].toLowerCase() + Q.slice(3)) : J.push(Q);
        }
        J.length && N(
          `Extraneous non-props attributes (${J.join(", ")}) were passed to component but could not be automatically inherited because component renders fragment or text root nodes.`
        ), Y.length && N(
          `Extraneous non-emits event listeners (${Y.join(", ")}) were passed to component but could not be automatically inherited because component renders fragment or text root nodes. If the listener is intended to be a component custom event listener only, declare it using the "emits" option.`
        );
      }
    }
  }
  return n.dirs && ({}.NODE_ENV !== "production" && !Ui(A) && N(
    "Runtime directive used on component with non-element root node. The directives will not function as intended."
  ), A = ct(A), A.dirs = A.dirs ? A.dirs.concat(n.dirs) : n.dirs), n.transition && ({}.NODE_ENV !== "production" && !Ui(A) && N(
    "Component inside <Transition> renders non-element root node that cannot be animated."
  ), A.transition = n.transition), {}.NODE_ENV !== "production" && j ? j(A) : V = A, Bn(H), V;
}
const gr = (e) => {
  const t = e.children, n = e.dynamicChildren, o = yi(t, !1);
  if (o) {
    if ({}.NODE_ENV !== "production" && o.patchFlag > 0 && o.patchFlag & 2048)
      return gr(o);
  } else
    return [e, void 0];
  const i = t.indexOf(o), s = n ? n.indexOf(o) : -1, r = (l) => {
    t[i] = l, n && (s > -1 ? n[s] = l : l.patchFlag > 0 && (e.dynamicChildren = [...n, l]));
  };
  return [Ce(o), r];
};
function yi(e, t = !0) {
  let n;
  for (let o = 0; o < e.length; o++) {
    const i = e[o];
    if (Bt(i)) {
      if (i.type !== ye || i.children === "v-if") {
        if (n)
          return;
        if (n = i, {}.NODE_ENV !== "production" && t && n.patchFlag > 0 && n.patchFlag & 2048)
          return yi(n.children);
      }
    } else
      return;
  }
  return n;
}
const Nc = (e) => {
  let t;
  for (const n in e)
    (n === "class" || n === "style" || mn(n)) && ((t || (t = {}))[n] = e[n]);
  return t;
}, Oc = (e, t) => {
  const n = {};
  for (const o in e)
    (!jn(o) || !(o.slice(9) in t)) && (n[o] = e[o]);
  return n;
}, Ui = (e) => e.shapeFlag & 7 || e.type === ye;
function xc(e, t, n) {
  const { props: o, children: i, component: s } = e, { props: r, children: l, patchFlag: u } = t, a = s.emitsOptions;
  if ({}.NODE_ENV !== "production" && (i || l) && Ot || t.dirs || t.transition)
    return !0;
  if (n && u >= 0) {
    if (u & 1024)
      return !0;
    if (u & 16)
      return o ? Ki(o, r, a) : !!r;
    if (u & 8) {
      const p = t.dynamicProps;
      for (let f = 0; f < p.length; f++) {
        const h = p[f];
        if (r[h] !== o[h] && !ro(a, h))
          return !0;
      }
    }
  } else
    return (i || l) && (!l || !l.$stable) ? !0 : o === r ? !1 : o ? r ? Ki(o, r, a) : !0 : !!r;
  return !1;
}
function Ki(e, t, n) {
  const o = Object.keys(t);
  if (o.length !== Object.keys(e).length)
    return !0;
  for (let i = 0; i < o.length; i++) {
    const s = o[i];
    if (t[s] !== e[s] && !ro(n, s))
      return !0;
  }
  return !1;
}
function $c({ vnode: e, parent: t }, n) {
  for (; t; ) {
    const o = t.subTree;
    if (o.suspense && o.suspense.activeBranch === e && (o.el = e.el), o === e)
      (e = t.vnode).el = n, t = t.parent;
    else
      break;
  }
}
const Bo = "components";
function zo(e, t) {
  return Dc(Bo, e, !0, t) || e;
}
const Cc = Symbol.for("v-ndc");
function Dc(e, t, n = !0, o = !1) {
  const i = ue || ce;
  if (i) {
    const s = i.type;
    if (e === Bo) {
      const l = $i(
        s,
        !1
      );
      if (l && (l === t || l === Le(t) || l === Dt(Le(t))))
        return s;
    }
    const r = (
      // local registration
      // check instance[type] first which is resolved for options API
      Gi(i[e] || s[e], t) || // global registration
      Gi(i.appContext[e], t)
    );
    if (!r && o)
      return s;
    if ({}.NODE_ENV !== "production" && n && !r) {
      const l = e === Bo ? `
If this is a native custom element, make sure to exclude it from component resolution via compilerOptions.isCustomElement.` : "";
      N(`Failed to resolve ${e.slice(0, -1)}: ${t}${l}`);
    }
    return r;
  } else
    ({}).NODE_ENV !== "production" && N(
      `resolve${Dt(e.slice(0, -1))} can only be used in render() or setup().`
    );
}
function Gi(e, t) {
  return e && (e[t] || e[Le(t)] || e[Dt(Le(t))]);
}
const Tc = (e) => e.__isSuspense;
function Sc(e, t) {
  t && t.pendingBranch ? R(e) ? t.effects.push(...e) : t.effects.push(e) : cr(e);
}
const Vc = Symbol.for("v-scx"), Pc = () => {
  {
    const e = Mn(Vc);
    return e || {}.NODE_ENV !== "production" && N(
      "Server rendering context not provided. Make sure to only call useSSRContext() conditionally in the server build."
    ), e;
  }
}, Sn = {};
function xo(e, t, n) {
  return {}.NODE_ENV !== "production" && !M(t) && N(
    "`watch(fn, options?)` signature has been moved to a separate API. Use `watchEffect(fn, options?)` instead. `watch` now only supports `watch(source, cb, options?) signature."
  ), _r(e, t, n);
}
function _r(e, t, {
  immediate: n,
  deep: o,
  flush: i,
  once: s,
  onTrack: r,
  onTrigger: l
} = X) {
  if (t && s) {
    const D = t;
    t = (...Y) => {
      D(...Y), K();
    };
  }
  ({}).NODE_ENV !== "production" && o !== void 0 && typeof o == "number" && N(
    'watch() "deep" option with number value will be used as watch depth in future versions. Please use a boolean instead to avoid potential breakage.'
  ), {}.NODE_ENV !== "production" && !t && (n !== void 0 && N(
    'watch() "immediate" option is only respected when using the watch(source, callback, options?) signature.'
  ), o !== void 0 && N(
    'watch() "deep" option is only respected when using the watch(source, callback, options?) signature.'
  ), s !== void 0 && N(
    'watch() "once" option is only respected when using the watch(source, callback, options?) signature.'
  ));
  const u = (D) => {
    N(
      "Invalid watch source: ",
      D,
      "A watch source can only be a getter/effect function, a ref, a reactive object, or an array of these types."
    );
  }, a = ce, p = (D) => o === !0 ? D : (
    // for deep: false, only traverse root-level properties
    Lt(D, o === !1 ? 1 : void 0)
  );
  let f, h = !1, g = !1;
  if (fe(e) ? (f = () => e.value, h = vt(e)) : bt(e) ? (f = () => p(e), h = !0) : R(e) ? (g = !0, h = e.some((D) => bt(D) || vt(D)), f = () => e.map((D) => {
    if (fe(D))
      return D.value;
    if (bt(D))
      return p(D);
    if (M(D))
      return Ue(D, a, 2);
    ({}).NODE_ENV !== "production" && u(D);
  })) : M(e) ? t ? f = () => Ue(e, a, 2) : f = () => (O && O(), Te(
    e,
    a,
    3,
    [$]
  )) : (f = se, {}.NODE_ENV !== "production" && u(e)), t && o) {
    const D = f;
    f = () => Lt(D());
  }
  let O, $ = (D) => {
    O = j.onStop = () => {
      Ue(D, a, 4), O = j.onStop = void 0;
    };
  }, V;
  if (ao)
    if ($ = se, t ? n && Te(t, a, 3, [
      f(),
      g ? [] : void 0,
      $
    ]) : f(), i === "sync") {
      const D = Pc();
      V = D.__watcherHandles || (D.__watcherHandles = []);
    } else
      return se;
  let P = g ? new Array(e.length).fill(Sn) : Sn;
  const H = () => {
    if (!(!j.active || !j.dirty))
      if (t) {
        const D = j.run();
        (o || h || (g ? D.some((Y, J) => rt(Y, P[J])) : rt(D, P))) && (O && O(), Te(t, a, 3, [
          D,
          // pass undefined as the old value when it's changed for the first time
          P === Sn ? void 0 : g && P[0] === Sn ? [] : P,
          $
        ]), P = D);
      } else
        j.run();
  };
  H.allowRecurse = !!t;
  let A;
  i === "sync" ? A = H : i === "post" ? A = () => _e(H, a && a.suspense) : (H.pre = !0, a && (H.id = a.uid), A = () => so(H));
  const j = new ci(f, se, A), I = xl(), K = () => {
    j.stop(), I && oi(I.effects, j);
  };
  return {}.NODE_ENV !== "production" && (j.onTrack = r, j.onTrigger = l), t ? n ? H() : P = j.run() : i === "post" ? _e(
    j.run.bind(j),
    a && a.suspense
  ) : j.run(), V && V.push(K), K;
}
function Ac(e, t, n) {
  const o = this.proxy, i = oe(e) ? e.includes(".") ? yr(o, e) : () => o[e] : e.bind(o, o);
  let s;
  M(t) ? s = t : (s = t.handler, n = t);
  const r = bn(this), l = _r(i, s.bind(o), n);
  return r(), l;
}
function yr(e, t) {
  const n = t.split(".");
  return () => {
    let o = e;
    for (let i = 0; i < n.length && o; i++)
      o = o[n[i]];
    return o;
  };
}
function Lt(e, t, n = 0, o) {
  if (!q(e) || e.__v_skip)
    return e;
  if (t && t > 0) {
    if (n >= t)
      return e;
    n++;
  }
  if (o = o || /* @__PURE__ */ new Set(), o.has(e))
    return e;
  if (o.add(e), fe(e))
    Lt(e.value, t, n, o);
  else if (R(e))
    for (let i = 0; i < e.length; i++)
      Lt(e[i], t, n, o);
  else if (Rs(e) || yt(e))
    e.forEach((i) => {
      Lt(i, t, n, o);
    });
  else if (Is(e))
    for (const i in e)
      Lt(e[i], t, n, o);
  return e;
}
function wr(e) {
  rl(e) && N("Do not use built-in directive ids as custom directive id: " + e);
}
function ht(e, t, n, o) {
  const i = e.dirs, s = t && t.dirs;
  for (let r = 0; r < i.length; r++) {
    const l = i[r];
    s && (l.oldValue = s[r].value);
    let u = l.dir[o];
    u && (ft(), Te(u, n, 8, [
      e.el,
      l,
      e,
      t
    ]), dt());
  }
}
/*! #__NO_SIDE_EFFECTS__ */
// @__NO_SIDE_EFFECTS__
function wi(e, t) {
  return M(e) ? (
    // #8326: extend call and options.name access are considered side-effects
    // by Rollup, so we have to wrap it in a pure-annotated IIFE.
    /* @__PURE__ */ (() => ne({ name: e.name }, t, { setup: e }))()
  ) : e;
}
const en = (e) => !!e.type.__asyncLoader, Ei = (e) => e.type.__isKeepAlive;
function Rc(e, t) {
  Er(e, "a", t);
}
function Mc(e, t) {
  Er(e, "da", t);
}
function Er(e, t, n = ce) {
  const o = e.__wdc || (e.__wdc = () => {
    let i = n;
    for (; i; ) {
      if (i.isDeactivated)
        return;
      i = i.parent;
    }
    return e();
  });
  if (co(t, o, n), n) {
    let i = n.parent;
    for (; i && i.parent; )
      Ei(i.parent.vnode) && Ic(o, t, n, i), i = i.parent;
  }
}
function Ic(e, t, n, o) {
  const i = co(
    t,
    e,
    o,
    !0
    /* prepend */
  );
  br(() => {
    oi(o[t], i);
  }, n);
}
function co(e, t, n = ce, o = !1) {
  if (n) {
    const i = n[e] || (n[e] = []), s = t.__weh || (t.__weh = (...r) => {
      if (n.isUnmounted)
        return;
      ft();
      const l = bn(n), u = Te(t, n, e, r);
      return l(), dt(), u;
    });
    return o ? i.unshift(s) : i.push(s), s;
  } else if ({}.NODE_ENV !== "production") {
    const i = _t(hi[e].replace(/ hook$/, ""));
    N(
      `${i} is called when there is no active component instance to be associated with. Lifecycle injection APIs can only be used during execution of setup(). If you are using async setup(), make sure to register lifecycle hooks before the first await statement.`
    );
  }
}
const Je = (e) => (t, n = ce) => (
  // post-create lifecycle registrations are noops during SSR (except for serverPrefetch)
  (!ao || e === "sp") && co(e, (...o) => t(...o), n)
), Lc = Je("bm"), Fc = Je("m"), Hc = Je("bu"), jc = Je("u"), kc = Je("bum"), br = Je("um"), Bc = Je("sp"), zc = Je(
  "rtg"
), Wc = Je(
  "rtc"
);
function Uc(e, t = ce) {
  co("ec", e, t);
}
function Kc(e, t, n = {}, o, i) {
  if (ue.isCE || ue.parent && en(ue.parent) && ue.parent.isCE)
    return t !== "default" && (n.name = t), ge("slot", n, o && o());
  let s = e[t];
  ({}).NODE_ENV !== "production" && s && s.length > 1 && (N(
    "SSR-optimized slot function detected in a non-SSR-optimized render function. You need to mark this component with $dynamic-slots in the parent template."
  ), s = () => []), s && s._c && (s._d = !1), Ke();
  const r = s && vr(s(n)), l = En(
    we,
    {
      key: n.key || // slot content array of a dynamic conditional slot may have a branch
      // key attached in the `createSlots` helper, respect that
      r && r.key || `_${t}`
    },
    r || (o ? o() : []),
    r && e._ === 1 ? 64 : -2
  );
  return !i && l.scopeId && (l.slotScopeIds = [l.scopeId + "-s"]), s && s._c && (s._d = !0), l;
}
function vr(e) {
  return e.some((t) => Bt(t) ? !(t.type === ye || t.type === we && !vr(t.children)) : !0) ? e : null;
}
const Wo = (e) => e ? Fr(e) ? xi(e) || e.proxy : Wo(e.parent) : null, xt = (
  // Move PURE marker to new line to workaround compiler discarding it
  // due to type annotation
  /* @__PURE__ */ ne(/* @__PURE__ */ Object.create(null), {
    $: (e) => e,
    $el: (e) => e.vnode.el,
    $data: (e) => e.data,
    $props: (e) => ({}).NODE_ENV !== "production" ? It(e.props) : e.props,
    $attrs: (e) => ({}).NODE_ENV !== "production" ? It(e.attrs) : e.attrs,
    $slots: (e) => ({}).NODE_ENV !== "production" ? It(e.slots) : e.slots,
    $refs: (e) => ({}).NODE_ENV !== "production" ? It(e.refs) : e.refs,
    $parent: (e) => Wo(e.parent),
    $root: (e) => Wo(e.root),
    $emit: (e) => e.emit,
    $options: (e) => vi(e),
    $forceUpdate: (e) => e.f || (e.f = () => {
      e.effect.dirty = !0, so(e.update);
    }),
    $nextTick: (e) => e.n || (e.n = rr.bind(e.proxy)),
    $watch: (e) => Ac.bind(e)
  })
), bi = (e) => e === "_" || e === "$", $o = (e, t) => e !== X && !e.__isScriptSetup && z(e, t), Nr = {
  get({ _: e }, t) {
    const { ctx: n, setupState: o, data: i, props: s, accessCache: r, type: l, appContext: u } = e;
    if ({}.NODE_ENV !== "production" && t === "__isVue")
      return !0;
    let a;
    if (t[0] !== "$") {
      const g = r[t];
      if (g !== void 0)
        switch (g) {
          case 1:
            return o[t];
          case 2:
            return i[t];
          case 4:
            return n[t];
          case 3:
            return s[t];
        }
      else {
        if ($o(o, t))
          return r[t] = 1, o[t];
        if (i !== X && z(i, t))
          return r[t] = 2, i[t];
        if (
          // only cache other properties when instance has declared (thus stable)
          // props
          (a = e.propsOptions[0]) && z(a, t)
        )
          return r[t] = 3, s[t];
        if (n !== X && z(n, t))
          return r[t] = 4, n[t];
        Uo && (r[t] = 0);
      }
    }
    const p = xt[t];
    let f, h;
    if (p)
      return t === "$attrs" ? (ae(e, "get", t), {}.NODE_ENV !== "production" && Wn()) : {}.NODE_ENV !== "production" && t === "$slots" && ae(e, "get", t), p(e);
    if (
      // css module (injected by vue-loader)
      (f = l.__cssModules) && (f = f[t])
    )
      return f;
    if (n !== X && z(n, t))
      return r[t] = 4, n[t];
    if (
      // global properties
      h = u.config.globalProperties, z(h, t)
    )
      return h[t];
    ({}).NODE_ENV !== "production" && ue && (!oe(t) || // #1091 avoid internal isRef/isVNode checks on component instance leading
    // to infinite warning loop
    t.indexOf("__v") !== 0) && (i !== X && bi(t[0]) && z(i, t) ? N(
      `Property ${JSON.stringify(
        t
      )} must be accessed via $data because it starts with a reserved character ("$" or "_") and is not proxied on the render context.`
    ) : e === ue && N(
      `Property ${JSON.stringify(t)} was accessed during render but is not defined on instance.`
    ));
  },
  set({ _: e }, t, n) {
    const { data: o, setupState: i, ctx: s } = e;
    return $o(i, t) ? (i[t] = n, !0) : {}.NODE_ENV !== "production" && i.__isScriptSetup && z(i, t) ? (N(`Cannot mutate <script setup> binding "${t}" from Options API.`), !1) : o !== X && z(o, t) ? (o[t] = n, !0) : z(e.props, t) ? ({}.NODE_ENV !== "production" && N(`Attempting to mutate prop "${t}". Props are readonly.`), !1) : t[0] === "$" && t.slice(1) in e ? ({}.NODE_ENV !== "production" && N(
      `Attempting to mutate public property "${t}". Properties starting with $ are reserved and readonly.`
    ), !1) : ({}.NODE_ENV !== "production" && t in e.appContext.config.globalProperties ? Object.defineProperty(s, t, {
      enumerable: !0,
      configurable: !0,
      value: n
    }) : s[t] = n, !0);
  },
  has({
    _: { data: e, setupState: t, accessCache: n, ctx: o, appContext: i, propsOptions: s }
  }, r) {
    let l;
    return !!n[r] || e !== X && z(e, r) || $o(t, r) || (l = s[0]) && z(l, r) || z(o, r) || z(xt, r) || z(i.config.globalProperties, r);
  },
  defineProperty(e, t, n) {
    return n.get != null ? e._.accessCache[t] = 0 : z(n, "value") && this.set(e, t, n.value, null), Reflect.defineProperty(e, t, n);
  }
};
({}).NODE_ENV !== "production" && (Nr.ownKeys = (e) => (N(
  "Avoid app logic that relies on enumerating keys on a component instance. The keys will be empty in production mode to avoid performance overhead."
), Reflect.ownKeys(e)));
function Gc(e) {
  const t = {};
  return Object.defineProperty(t, "_", {
    configurable: !0,
    enumerable: !1,
    get: () => e
  }), Object.keys(xt).forEach((n) => {
    Object.defineProperty(t, n, {
      configurable: !0,
      enumerable: !1,
      get: () => xt[n](e),
      // intercepted by the proxy so no need for implementation,
      // but needed to prevent set errors
      set: se
    });
  }), t;
}
function qc(e) {
  const {
    ctx: t,
    propsOptions: [n]
  } = e;
  n && Object.keys(n).forEach((o) => {
    Object.defineProperty(t, o, {
      enumerable: !0,
      configurable: !0,
      get: () => e.props[o],
      set: se
    });
  });
}
function Yc(e) {
  const { ctx: t, setupState: n } = e;
  Object.keys(k(n)).forEach((o) => {
    if (!n.__isScriptSetup) {
      if (bi(o[0])) {
        N(
          `setup() return property ${JSON.stringify(
            o
          )} should not start with "$" or "_" which are reserved prefixes for Vue internals.`
        );
        return;
      }
      Object.defineProperty(t, o, {
        enumerable: !0,
        configurable: !0,
        get: () => n[o],
        set: se
      });
    }
  });
}
function qi(e) {
  return R(e) ? e.reduce(
    (t, n) => (t[n] = null, t),
    {}
  ) : e;
}
function Jc() {
  const e = /* @__PURE__ */ Object.create(null);
  return (t, n) => {
    e[n] ? N(`${t} property "${n}" is already defined in ${e[n]}.`) : e[n] = t;
  };
}
let Uo = !0;
function Xc(e) {
  const t = vi(e), n = e.proxy, o = e.ctx;
  Uo = !1, t.beforeCreate && Yi(t.beforeCreate, e, "bc");
  const {
    // state
    data: i,
    computed: s,
    methods: r,
    watch: l,
    provide: u,
    inject: a,
    // lifecycle
    created: p,
    beforeMount: f,
    mounted: h,
    beforeUpdate: g,
    updated: O,
    activated: $,
    deactivated: V,
    beforeDestroy: P,
    beforeUnmount: H,
    destroyed: A,
    unmounted: j,
    render: I,
    renderTracked: K,
    renderTriggered: D,
    errorCaptured: Y,
    serverPrefetch: J,
    // public API
    expose: Z,
    inheritAttrs: ie,
    // assets
    components: Q,
    directives: ve,
    filters: de
  } = t, re = {}.NODE_ENV !== "production" ? Jc() : null;
  if ({}.NODE_ENV !== "production") {
    const [L] = e.propsOptions;
    if (L)
      for (const B in L)
        re("Props", B);
  }
  if (a && Zc(a, o, re), r)
    for (const L in r) {
      const B = r[L];
      M(B) ? ({}.NODE_ENV !== "production" ? Object.defineProperty(o, L, {
        value: B.bind(n),
        configurable: !0,
        enumerable: !0,
        writable: !0
      }) : o[L] = B.bind(n), {}.NODE_ENV !== "production" && re("Methods", L)) : {}.NODE_ENV !== "production" && N(
        `Method "${L}" has type "${typeof B}" in the component definition. Did you reference the function correctly?`
      );
    }
  if (i) {
    ({}).NODE_ENV !== "production" && !M(i) && N(
      "The data option must be a function. Plain object usage is no longer supported."
    );
    const L = i.call(n, n);
    if ({}.NODE_ENV !== "production" && ii(L) && N(
      "data() returned a Promise - note data() cannot be async; If you intend to perform data fetching before component renders, use async setup() + <Suspense>."
    ), !q(L))
      ({}).NODE_ENV !== "production" && N("data() should return an object.");
    else if (e.data = di(L), {}.NODE_ENV !== "production")
      for (const B in L)
        re("Data", B), bi(B[0]) || Object.defineProperty(o, B, {
          configurable: !0,
          enumerable: !0,
          get: () => L[B],
          set: se
        });
  }
  if (Uo = !0, s)
    for (const L in s) {
      const B = s[L], Ve = M(B) ? B.bind(n, n) : M(B.get) ? B.get.bind(n, n) : se;
      ({}).NODE_ENV !== "production" && Ve === se && N(`Computed property "${L}" has no getter.`);
      const _o = !M(B) && M(B.set) ? B.set.bind(n) : {}.NODE_ENV !== "production" ? () => {
        N(
          `Write operation failed: computed property "${L}" is readonly.`
        );
      } : se, Ut = Lu({
        get: Ve,
        set: _o
      });
      Object.defineProperty(o, L, {
        enumerable: !0,
        configurable: !0,
        get: () => Ut.value,
        set: (Pt) => Ut.value = Pt
      }), {}.NODE_ENV !== "production" && re("Computed", L);
    }
  if (l)
    for (const L in l)
      Or(l[L], o, n, L);
  if (u) {
    const L = M(u) ? u.call(n) : u;
    Reflect.ownKeys(L).forEach((B) => {
      iu(B, L[B]);
    });
  }
  p && Yi(p, e, "c");
  function ee(L, B) {
    R(B) ? B.forEach((Ve) => L(Ve.bind(n))) : B && L(B.bind(n));
  }
  if (ee(Lc, f), ee(Fc, h), ee(Hc, g), ee(jc, O), ee(Rc, $), ee(Mc, V), ee(Uc, Y), ee(Wc, K), ee(zc, D), ee(kc, H), ee(br, j), ee(Bc, J), R(Z))
    if (Z.length) {
      const L = e.exposed || (e.exposed = {});
      Z.forEach((B) => {
        Object.defineProperty(L, B, {
          get: () => n[B],
          set: (Ve) => n[B] = Ve
        });
      });
    } else
      e.exposed || (e.exposed = {});
  I && e.render === se && (e.render = I), ie != null && (e.inheritAttrs = ie), Q && (e.components = Q), ve && (e.directives = ve);
}
function Zc(e, t, n = se) {
  R(e) && (e = Ko(e));
  for (const o in e) {
    const i = e[o];
    let s;
    q(i) ? "default" in i ? s = Mn(
      i.from || o,
      i.default,
      !0
    ) : s = Mn(i.from || o) : s = Mn(i), fe(s) ? Object.defineProperty(t, o, {
      enumerable: !0,
      configurable: !0,
      get: () => s.value,
      set: (r) => s.value = r
    }) : t[o] = s, {}.NODE_ENV !== "production" && n("Inject", o);
  }
}
function Yi(e, t, n) {
  Te(
    R(e) ? e.map((o) => o.bind(t.proxy)) : e.bind(t.proxy),
    t,
    n
  );
}
function Or(e, t, n, o) {
  const i = o.includes(".") ? yr(n, o) : () => n[o];
  if (oe(e)) {
    const s = t[e];
    M(s) ? xo(i, s) : {}.NODE_ENV !== "production" && N(`Invalid watch handler specified by key "${e}"`, s);
  } else if (M(e))
    xo(i, e.bind(n));
  else if (q(e))
    if (R(e))
      e.forEach((s) => Or(s, t, n, o));
    else {
      const s = M(e.handler) ? e.handler.bind(n) : t[e.handler];
      M(s) ? xo(i, s, e) : {}.NODE_ENV !== "production" && N(`Invalid watch handler specified by key "${e.handler}"`, s);
    }
  else
    ({}).NODE_ENV !== "production" && N(`Invalid watch option: "${o}"`, e);
}
function vi(e) {
  const t = e.type, { mixins: n, extends: o } = t, {
    mixins: i,
    optionsCache: s,
    config: { optionMergeStrategies: r }
  } = e.appContext, l = s.get(t);
  let u;
  return l ? u = l : !i.length && !n && !o ? u = t : (u = {}, i.length && i.forEach(
    (a) => Un(u, a, r, !0)
  ), Un(u, t, r)), q(t) && s.set(t, u), u;
}
function Un(e, t, n, o = !1) {
  const { mixins: i, extends: s } = t;
  s && Un(e, s, n, !0), i && i.forEach(
    (r) => Un(e, r, n, !0)
  );
  for (const r in t)
    if (o && r === "expose")
      ({}).NODE_ENV !== "production" && N(
        '"expose" option is ignored when declared in mixins or extends. It should only be declared in the base component itself.'
      );
    else {
      const l = Qc[r] || n && n[r];
      e[r] = l ? l(e[r], t[r]) : t[r];
    }
  return e;
}
const Qc = {
  data: Ji,
  props: Xi,
  emits: Xi,
  // objects
  methods: Xt,
  computed: Xt,
  // lifecycle
  beforeCreate: me,
  created: me,
  beforeMount: me,
  mounted: me,
  beforeUpdate: me,
  updated: me,
  beforeDestroy: me,
  beforeUnmount: me,
  destroyed: me,
  unmounted: me,
  activated: me,
  deactivated: me,
  errorCaptured: me,
  serverPrefetch: me,
  // assets
  components: Xt,
  directives: Xt,
  // watch
  watch: tu,
  // provide / inject
  provide: Ji,
  inject: eu
};
function Ji(e, t) {
  return t ? e ? function() {
    return ne(
      M(e) ? e.call(this, this) : e,
      M(t) ? t.call(this, this) : t
    );
  } : t : e;
}
function eu(e, t) {
  return Xt(Ko(e), Ko(t));
}
function Ko(e) {
  if (R(e)) {
    const t = {};
    for (let n = 0; n < e.length; n++)
      t[e[n]] = e[n];
    return t;
  }
  return e;
}
function me(e, t) {
  return e ? [...new Set([].concat(e, t))] : t;
}
function Xt(e, t) {
  return e ? ne(/* @__PURE__ */ Object.create(null), e, t) : t;
}
function Xi(e, t) {
  return e ? R(e) && R(t) ? [.../* @__PURE__ */ new Set([...e, ...t])] : ne(
    /* @__PURE__ */ Object.create(null),
    qi(e),
    qi(t ?? {})
  ) : t;
}
function tu(e, t) {
  if (!e)
    return t;
  if (!t)
    return e;
  const n = ne(/* @__PURE__ */ Object.create(null), e);
  for (const o in t)
    n[o] = me(e[o], t[o]);
  return n;
}
function xr() {
  return {
    app: null,
    config: {
      isNativeTag: As,
      performance: !1,
      globalProperties: {},
      optionMergeStrategies: {},
      errorHandler: void 0,
      warnHandler: void 0,
      compilerOptions: {}
    },
    mixins: [],
    components: {},
    directives: {},
    provides: /* @__PURE__ */ Object.create(null),
    optionsCache: /* @__PURE__ */ new WeakMap(),
    propsCache: /* @__PURE__ */ new WeakMap(),
    emitsCache: /* @__PURE__ */ new WeakMap()
  };
}
let nu = 0;
function ou(e, t) {
  return function(o, i = null) {
    M(o) || (o = ne({}, o)), i != null && !q(i) && ({}.NODE_ENV !== "production" && N("root props passed to app.mount() must be an object."), i = null);
    const s = xr(), r = /* @__PURE__ */ new WeakSet();
    let l = !1;
    const u = s.app = {
      _uid: nu++,
      _component: o,
      _props: i,
      _container: null,
      _context: s,
      _instance: null,
      version: cs,
      get config() {
        return s.config;
      },
      set config(a) {
        ({}).NODE_ENV !== "production" && N(
          "app.config cannot be replaced. Modify individual options instead."
        );
      },
      use(a, ...p) {
        return r.has(a) ? {}.NODE_ENV !== "production" && N("Plugin has already been applied to target app.") : a && M(a.install) ? (r.add(a), a.install(u, ...p)) : M(a) ? (r.add(a), a(u, ...p)) : {}.NODE_ENV !== "production" && N(
          'A plugin must either be a function or an object with an "install" function.'
        ), u;
      },
      mixin(a) {
        return s.mixins.includes(a) ? {}.NODE_ENV !== "production" && N(
          "Mixin has already been applied to target app" + (a.name ? `: ${a.name}` : "")
        ) : s.mixins.push(a), u;
      },
      component(a, p) {
        return {}.NODE_ENV !== "production" && Xo(a, s.config), p ? ({}.NODE_ENV !== "production" && s.components[a] && N(`Component "${a}" has already been registered in target app.`), s.components[a] = p, u) : s.components[a];
      },
      directive(a, p) {
        return {}.NODE_ENV !== "production" && wr(a), p ? ({}.NODE_ENV !== "production" && s.directives[a] && N(`Directive "${a}" has already been registered in target app.`), s.directives[a] = p, u) : s.directives[a];
      },
      mount(a, p, f) {
        if (l)
          ({}).NODE_ENV !== "production" && N(
            "App has already been mounted.\nIf you want to remount the same app, move your app creation logic into a factory function and create fresh app instances for each mount - e.g. `const createMyApp = () => createApp(App)`"
          );
        else {
          ({}).NODE_ENV !== "production" && a.__vue_app__ && N(
            "There is already an app instance mounted on the host container.\n If you want to mount another app on the same host container, you need to unmount the previous app by calling `app.unmount()` first."
          );
          const h = ge(o, i);
          return h.appContext = s, f === !0 ? f = "svg" : f === !1 && (f = void 0), {}.NODE_ENV !== "production" && (s.reload = () => {
            e(
              ct(h),
              a,
              f
            );
          }), p && t ? t(h, a) : e(h, a, f), l = !0, u._container = a, a.__vue_app__ = u, {}.NODE_ENV !== "production" && (u._instance = h.component, fc(u, cs)), xi(h.component) || h.component.proxy;
        }
      },
      unmount() {
        l ? (e(null, u._container), {}.NODE_ENV !== "production" && (u._instance = null, dc(u)), delete u._container.__vue_app__) : {}.NODE_ENV !== "production" && N("Cannot unmount an app that is not mounted.");
      },
      provide(a, p) {
        return {}.NODE_ENV !== "production" && a in s.provides && N(
          `App already provides property with key "${String(a)}". It will be overwritten with the new value.`
        ), s.provides[a] = p, u;
      },
      runWithContext(a) {
        const p = tn;
        tn = u;
        try {
          return a();
        } finally {
          tn = p;
        }
      }
    };
    return u;
  };
}
let tn = null;
function iu(e, t) {
  if (!ce)
    ({}).NODE_ENV !== "production" && N("provide() can only be used inside setup().");
  else {
    let n = ce.provides;
    const o = ce.parent && ce.parent.provides;
    o === n && (n = ce.provides = Object.create(o)), n[e] = t;
  }
}
function Mn(e, t, n = !1) {
  const o = ce || ue;
  if (o || tn) {
    const i = o ? o.parent == null ? o.vnode.appContext && o.vnode.appContext.provides : o.parent.provides : tn._context.provides;
    if (i && e in i)
      return i[e];
    if (arguments.length > 1)
      return n && M(t) ? t.call(o && o.proxy) : t;
    ({}).NODE_ENV !== "production" && N(`injection "${String(e)}" not found.`);
  } else
    ({}).NODE_ENV !== "production" && N("inject() can only be used inside setup() or functional components.");
}
function su(e, t, n, o = !1) {
  const i = {}, s = {};
  kn(s, uo, 1), e.propsDefaults = /* @__PURE__ */ Object.create(null), $r(e, t, i, s);
  for (const r in e.propsOptions[0])
    r in i || (i[r] = void 0);
  ({}).NODE_ENV !== "production" && Dr(t || {}, i, e), n ? e.props = o ? i : Ul(i) : e.type.props ? e.props = i : e.props = s, e.attrs = s;
}
function ru(e) {
  for (; e; ) {
    if (e.type.__hmrId)
      return !0;
    e = e.parent;
  }
}
function lu(e, t, n, o) {
  const {
    props: i,
    attrs: s,
    vnode: { patchFlag: r }
  } = e, l = k(i), [u] = e.propsOptions;
  let a = !1;
  if (
    // always force full diff in dev
    // - #1942 if hmr is enabled with sfc component
    // - vite#872 non-sfc component used by sfc component
    !({}.NODE_ENV !== "production" && ru(e)) && (o || r > 0) && !(r & 16)
  ) {
    if (r & 8) {
      const p = e.vnode.dynamicProps;
      for (let f = 0; f < p.length; f++) {
        let h = p[f];
        if (ro(e.emitsOptions, h))
          continue;
        const g = t[h];
        if (u)
          if (z(s, h))
            g !== s[h] && (s[h] = g, a = !0);
          else {
            const O = Le(h);
            i[O] = Go(
              u,
              l,
              O,
              g,
              e,
              !1
            );
          }
        else
          g !== s[h] && (s[h] = g, a = !0);
      }
    }
  } else {
    $r(e, t, i, s) && (a = !0);
    let p;
    for (const f in l)
      (!t || // for camelCase
      !z(t, f) && // it's possible the original props was passed in as kebab-case
      // and converted to camelCase (#955)
      ((p = Ge(f)) === f || !z(t, p))) && (u ? n && // for camelCase
      (n[f] !== void 0 || // for kebab-case
      n[p] !== void 0) && (i[f] = Go(
        u,
        l,
        f,
        void 0,
        e,
        !0
      )) : delete i[f]);
    if (s !== l)
      for (const f in s)
        (!t || !z(t, f)) && (delete s[f], a = !0);
  }
  a && Ie(e, "set", "$attrs"), {}.NODE_ENV !== "production" && Dr(t || {}, i, e);
}
function $r(e, t, n, o) {
  const [i, s] = e.propsOptions;
  let r = !1, l;
  if (t)
    for (let u in t) {
      if (Zt(u))
        continue;
      const a = t[u];
      let p;
      i && z(i, p = Le(u)) ? !s || !s.includes(p) ? n[p] = a : (l || (l = {}))[p] = a : ro(e.emitsOptions, u) || (!(u in o) || a !== o[u]) && (o[u] = a, r = !0);
    }
  if (s) {
    const u = k(n), a = l || X;
    for (let p = 0; p < s.length; p++) {
      const f = s[p];
      n[f] = Go(
        i,
        u,
        f,
        a[f],
        e,
        !z(a, f)
      );
    }
  }
  return r;
}
function Go(e, t, n, o, i, s) {
  const r = e[n];
  if (r != null) {
    const l = z(r, "default");
    if (l && o === void 0) {
      const u = r.default;
      if (r.type !== Function && !r.skipFactory && M(u)) {
        const { propsDefaults: a } = i;
        if (n in a)
          o = a[n];
        else {
          const p = bn(i);
          o = a[n] = u.call(
            null,
            t
          ), p();
        }
      } else
        o = u;
    }
    r[
      0
      /* shouldCast */
    ] && (s && !l ? o = !1 : r[
      1
      /* shouldCastTrue */
    ] && (o === "" || o === Ge(n)) && (o = !0));
  }
  return o;
}
function Cr(e, t, n = !1) {
  const o = t.propsCache, i = o.get(e);
  if (i)
    return i;
  const s = e.props, r = {}, l = [];
  let u = !1;
  if (!M(e)) {
    const p = (f) => {
      u = !0;
      const [h, g] = Cr(f, t, !0);
      ne(r, h), g && l.push(...g);
    };
    !n && t.mixins.length && t.mixins.forEach(p), e.extends && p(e.extends), e.mixins && e.mixins.forEach(p);
  }
  if (!s && !u)
    return q(e) && o.set(e, Ft), Ft;
  if (R(s))
    for (let p = 0; p < s.length; p++) {
      ({}).NODE_ENV !== "production" && !oe(s[p]) && N("props must be strings when using array syntax.", s[p]);
      const f = Le(s[p]);
      Zi(f) && (r[f] = X);
    }
  else if (s) {
    ({}).NODE_ENV !== "production" && !q(s) && N("invalid props options", s);
    for (const p in s) {
      const f = Le(p);
      if (Zi(f)) {
        const h = s[p], g = r[f] = R(h) || M(h) ? { type: h } : ne({}, h);
        if (g) {
          const O = es(Boolean, g.type), $ = es(String, g.type);
          g[
            0
            /* shouldCast */
          ] = O > -1, g[
            1
            /* shouldCastTrue */
          ] = $ < 0 || O < $, (O > -1 || z(g, "default")) && l.push(f);
        }
      }
    }
  }
  const a = [r, l];
  return q(e) && o.set(e, a), a;
}
function Zi(e) {
  return e[0] !== "$" && !Zt(e) ? !0 : ({}.NODE_ENV !== "production" && N(`Invalid prop name: "${e}" is a reserved property.`), !1);
}
function qo(e) {
  return e === null ? "null" : typeof e == "function" ? e.name || "" : typeof e == "object" && e.constructor && e.constructor.name || "";
}
function Qi(e, t) {
  return qo(e) === qo(t);
}
function es(e, t) {
  return R(t) ? t.findIndex((n) => Qi(n, e)) : M(t) && Qi(t, e) ? 0 : -1;
}
function Dr(e, t, n) {
  const o = k(t), i = n.propsOptions[0];
  for (const s in i) {
    let r = i[s];
    r != null && cu(
      s,
      o[s],
      r,
      {}.NODE_ENV !== "production" ? It(o) : o,
      !z(e, s) && !z(e, Ge(s))
    );
  }
}
function cu(e, t, n, o, i) {
  const { type: s, required: r, validator: l, skipCheck: u } = n;
  if (r && i) {
    N('Missing required prop: "' + e + '"');
    return;
  }
  if (!(t == null && !r)) {
    if (s != null && s !== !0 && !u) {
      let a = !1;
      const p = R(s) ? s : [s], f = [];
      for (let h = 0; h < p.length && !a; h++) {
        const { valid: g, expectedType: O } = au(t, p[h]);
        f.push(O || ""), a = g;
      }
      if (!a) {
        N(fu(e, t, f));
        return;
      }
    }
    l && !l(t, o) && N('Invalid prop: custom validator check failed for prop "' + e + '".');
  }
}
const uu = /* @__PURE__ */ Ye(
  "String,Number,Boolean,Function,Symbol,BigInt"
);
function au(e, t) {
  let n;
  const o = qo(t);
  if (uu(o)) {
    const i = typeof e;
    n = i === o.toLowerCase(), !n && i === "object" && (n = e instanceof t);
  } else
    o === "Object" ? n = q(e) : o === "Array" ? n = R(e) : o === "null" ? n = e === null : n = e instanceof t;
  return {
    valid: n,
    expectedType: o
  };
}
function fu(e, t, n) {
  if (n.length === 0)
    return `Prop type [] for prop "${e}" won't match anything. Did you mean to use type Array instead?`;
  let o = `Invalid prop: type check failed for prop "${e}". Expected ${n.map(Dt).join(" | ")}`;
  const i = n[0], s = si(t), r = ts(t, i), l = ts(t, s);
  return n.length === 1 && ns(i) && !du(i, s) && (o += ` with value ${r}`), o += `, got ${s} `, ns(s) && (o += `with value ${l}.`), o;
}
function ts(e, t) {
  return t === "String" ? `"${e}"` : t === "Number" ? `${Number(e)}` : `${e}`;
}
function ns(e) {
  return ["string", "number", "boolean"].some((n) => e.toLowerCase() === n);
}
function du(...e) {
  return e.some((t) => t.toLowerCase() === "boolean");
}
const Tr = (e) => e[0] === "_" || e === "$stable", Ni = (e) => R(e) ? e.map(Ce) : [Ce(e)], pu = (e, t, n) => {
  if (t._n)
    return t;
  const o = zn((...i) => ({}.NODE_ENV !== "production" && ce && (!n || n.root === ce.root) && N(
    `Slot "${e}" invoked outside of the render function: this will not track dependencies used in the slot. Invoke the slot function inside the render function instead.`
  ), Ni(t(...i))), n);
  return o._c = !1, o;
}, Sr = (e, t, n) => {
  const o = e._ctx;
  for (const i in e) {
    if (Tr(i))
      continue;
    const s = e[i];
    if (M(s))
      t[i] = pu(i, s, o);
    else if (s != null) {
      ({}).NODE_ENV !== "production" && N(
        `Non-function value encountered for slot "${i}". Prefer function slots for better performance.`
      );
      const r = Ni(s);
      t[i] = () => r;
    }
  }
}, Vr = (e, t) => {
  ({}).NODE_ENV !== "production" && !Ei(e.vnode) && N(
    "Non-function value encountered for default slot. Prefer function slots for better performance."
  );
  const n = Ni(t);
  e.slots.default = () => n;
}, hu = (e, t) => {
  if (e.vnode.shapeFlag & 32) {
    const n = t._;
    n ? (e.slots = k(t), kn(t, "_", n)) : Sr(
      t,
      e.slots = {}
    );
  } else
    e.slots = {}, t && Vr(e, t);
  kn(e.slots, uo, 1);
}, mu = (e, t, n) => {
  const { vnode: o, slots: i } = e;
  let s = !0, r = X;
  if (o.shapeFlag & 32) {
    const l = t._;
    l ? {}.NODE_ENV !== "production" && Ot ? (ne(i, t), Ie(e, "set", "$slots")) : n && l === 1 ? s = !1 : (ne(i, t), !n && l === 1 && delete i._) : (s = !t.$stable, Sr(t, i)), r = t;
  } else
    t && (Vr(e, t), r = { default: 1 });
  if (s)
    for (const l in i)
      !Tr(l) && r[l] == null && delete i[l];
};
function Yo(e, t, n, o, i = !1) {
  if (R(e)) {
    e.forEach(
      (h, g) => Yo(
        h,
        t && (R(t) ? t[g] : t),
        n,
        o,
        i
      )
    );
    return;
  }
  if (en(o) && !i)
    return;
  const s = o.shapeFlag & 4 ? xi(o.component) || o.component.proxy : o.el, r = i ? null : s, { i: l, r: u } = e;
  if ({}.NODE_ENV !== "production" && !l) {
    N(
      "Missing ref owner context. ref cannot be used on hoisted vnodes. A vnode with ref must be created inside the render function."
    );
    return;
  }
  const a = t && t.r, p = l.refs === X ? l.refs = {} : l.refs, f = l.setupState;
  if (a != null && a !== u && (oe(a) ? (p[a] = null, z(f, a) && (f[a] = null)) : fe(a) && (a.value = null)), M(u))
    Ue(u, l, 12, [r, p]);
  else {
    const h = oe(u), g = fe(u);
    if (h || g) {
      const O = () => {
        if (e.f) {
          const $ = h ? z(f, u) ? f[u] : p[u] : u.value;
          i ? R($) && oi($, s) : R($) ? $.includes(s) || $.push(s) : h ? (p[u] = [s], z(f, u) && (f[u] = p[u])) : (u.value = [s], e.k && (p[e.k] = u.value));
        } else
          h ? (p[u] = r, z(f, u) && (f[u] = r)) : g ? (u.value = r, e.k && (p[e.k] = r)) : {}.NODE_ENV !== "production" && N("Invalid template ref type:", u, `(${typeof u})`);
      };
      r ? (O.id = -1, _e(O, n)) : O();
    } else
      ({}).NODE_ENV !== "production" && N("Invalid template ref type:", u, `(${typeof u})`);
  }
}
let qt, it;
function ke(e, t) {
  e.appContext.config.performance && Kn() && it.mark(`vue-${t}-${e.uid}`), {}.NODE_ENV !== "production" && gc(e, t, Kn() ? it.now() : Date.now());
}
function Be(e, t) {
  if (e.appContext.config.performance && Kn()) {
    const n = `vue-${t}-${e.uid}`, o = n + ":end";
    it.mark(o), it.measure(
      `<${fo(e, e.type)}> ${t}`,
      n,
      o
    ), it.clearMarks(n), it.clearMarks(o);
  }
  ({}).NODE_ENV !== "production" && _c(e, t, Kn() ? it.now() : Date.now());
}
function Kn() {
  return qt !== void 0 || (typeof window < "u" && window.performance ? (qt = !0, it = window.performance) : qt = !1), qt;
}
function gu() {
  const e = [];
  if ({}.NODE_ENV !== "production" && e.length) {
    const t = e.length > 1;
    console.warn(
      `Feature flag${t ? "s" : ""} ${e.join(", ")} ${t ? "are" : "is"} not explicitly defined. You are running the esm-bundler build of Vue, which expects these compile-time feature flags to be globally injected via the bundler config in order to get better tree-shaking in the production bundle.

For more details, see https://link.vuejs.org/feature-flags.`
    );
  }
}
const _e = Sc;
function _u(e) {
  return yu(e);
}
function yu(e, t) {
  gu();
  const n = li();
  n.__VUE__ = !0, {}.NODE_ENV !== "production" && dr(n.__VUE_DEVTOOLS_GLOBAL_HOOK__, n);
  const {
    insert: o,
    remove: i,
    patchProp: s,
    createElement: r,
    createText: l,
    createComment: u,
    setText: a,
    setElementText: p,
    parentNode: f,
    nextSibling: h,
    setScopeId: g = se,
    insertStaticContent: O
  } = e, $ = (c, d, m, _ = null, y = null, b = null, x = void 0, E = null, v = {}.NODE_ENV !== "production" && Ot ? !1 : !!d.dynamicChildren) => {
    if (c === d)
      return;
    c && !Yt(c, d) && (_ = On(c), Xe(c, y, b, !0), c = null), d.patchFlag === -2 && (v = !1, d.dynamicChildren = null);
    const { type: w, ref: C, shapeFlag: S } = d;
    switch (w) {
      case wn:
        V(c, d, m, _);
        break;
      case ye:
        P(c, d, m, _);
        break;
      case Ln:
        c == null ? H(d, m, _, x) : {}.NODE_ENV !== "production" && A(c, d, m, x);
        break;
      case we:
        ve(
          c,
          d,
          m,
          _,
          y,
          b,
          x,
          E,
          v
        );
        break;
      default:
        S & 1 ? K(
          c,
          d,
          m,
          _,
          y,
          b,
          x,
          E,
          v
        ) : S & 6 ? de(
          c,
          d,
          m,
          _,
          y,
          b,
          x,
          E,
          v
        ) : S & 64 || S & 128 ? w.process(
          c,
          d,
          m,
          _,
          y,
          b,
          x,
          E,
          v,
          At
        ) : {}.NODE_ENV !== "production" && N("Invalid VNode type:", w, `(${typeof w})`);
    }
    C != null && y && Yo(C, c && c.ref, b, d || c, !d);
  }, V = (c, d, m, _) => {
    if (c == null)
      o(
        d.el = l(d.children),
        m,
        _
      );
    else {
      const y = d.el = c.el;
      d.children !== c.children && a(y, d.children);
    }
  }, P = (c, d, m, _) => {
    c == null ? o(
      d.el = u(d.children || ""),
      m,
      _
    ) : d.el = c.el;
  }, H = (c, d, m, _) => {
    [c.el, c.anchor] = O(
      c.children,
      d,
      m,
      _,
      c.el,
      c.anchor
    );
  }, A = (c, d, m, _) => {
    if (d.children !== c.children) {
      const y = h(c.anchor);
      I(c), [d.el, d.anchor] = O(
        d.children,
        m,
        y,
        _
      );
    } else
      d.el = c.el, d.anchor = c.anchor;
  }, j = ({ el: c, anchor: d }, m, _) => {
    let y;
    for (; c && c !== d; )
      y = h(c), o(c, m, _), c = y;
    o(d, m, _);
  }, I = ({ el: c, anchor: d }) => {
    let m;
    for (; c && c !== d; )
      m = h(c), i(c), c = m;
    i(d);
  }, K = (c, d, m, _, y, b, x, E, v) => {
    d.type === "svg" ? x = "svg" : d.type === "math" && (x = "mathml"), c == null ? D(
      d,
      m,
      _,
      y,
      b,
      x,
      E,
      v
    ) : Z(
      c,
      d,
      y,
      b,
      x,
      E,
      v
    );
  }, D = (c, d, m, _, y, b, x, E) => {
    let v, w;
    const { props: C, shapeFlag: S, transition: T, dirs: F } = c;
    if (v = c.el = r(
      c.type,
      b,
      C && C.is,
      C
    ), S & 8 ? p(v, c.children) : S & 16 && J(
      c.children,
      v,
      null,
      _,
      y,
      Co(c, b),
      x,
      E
    ), F && ht(c, null, _, "created"), Y(v, c, c.scopeId, x, _), C) {
      for (const G in C)
        G !== "value" && !Zt(G) && s(
          v,
          G,
          null,
          C[G],
          b,
          c.children,
          _,
          y,
          je
        );
      "value" in C && s(v, "value", null, C.value, b), (w = C.onVnodeBeforeMount) && Ae(w, _, c);
    }
    ({}).NODE_ENV !== "production" && (Object.defineProperty(v, "__vnode", {
      value: c,
      enumerable: !1
    }), Object.defineProperty(v, "__vueParentComponent", {
      value: _,
      enumerable: !1
    })), F && ht(c, null, _, "beforeMount");
    const W = wu(y, T);
    W && T.beforeEnter(v), o(v, d, m), ((w = C && C.onVnodeMounted) || W || F) && _e(() => {
      w && Ae(w, _, c), W && T.enter(v), F && ht(c, null, _, "mounted");
    }, y);
  }, Y = (c, d, m, _, y) => {
    if (m && g(c, m), _)
      for (let b = 0; b < _.length; b++)
        g(c, _[b]);
    if (y) {
      let b = y.subTree;
      if ({}.NODE_ENV !== "production" && b.patchFlag > 0 && b.patchFlag & 2048 && (b = yi(b.children) || b), d === b) {
        const x = y.vnode;
        Y(
          c,
          x,
          x.scopeId,
          x.slotScopeIds,
          y.parent
        );
      }
    }
  }, J = (c, d, m, _, y, b, x, E, v = 0) => {
    for (let w = v; w < c.length; w++) {
      const C = c[w] = E ? nt(c[w]) : Ce(c[w]);
      $(
        null,
        C,
        d,
        m,
        _,
        y,
        b,
        x,
        E
      );
    }
  }, Z = (c, d, m, _, y, b, x) => {
    const E = d.el = c.el;
    let { patchFlag: v, dynamicChildren: w, dirs: C } = d;
    v |= c.patchFlag & 16;
    const S = c.props || X, T = d.props || X;
    let F;
    if (m && mt(m, !1), (F = T.onVnodeBeforeUpdate) && Ae(F, m, d, c), C && ht(d, c, m, "beforeUpdate"), m && mt(m, !0), {}.NODE_ENV !== "production" && Ot && (v = 0, x = !1, w = null), w ? (ie(
      c.dynamicChildren,
      w,
      E,
      m,
      _,
      Co(d, y),
      b
    ), {}.NODE_ENV !== "production" && In(c, d)) : x || Ve(
      c,
      d,
      E,
      null,
      m,
      _,
      Co(d, y),
      b,
      !1
    ), v > 0) {
      if (v & 16)
        Q(
          E,
          d,
          S,
          T,
          m,
          _,
          y
        );
      else if (v & 2 && S.class !== T.class && s(E, "class", null, T.class, y), v & 4 && s(E, "style", S.style, T.style, y), v & 8) {
        const W = d.dynamicProps;
        for (let G = 0; G < W.length; G++) {
          const te = W[G], le = S[te], xe = T[te];
          (xe !== le || te === "value") && s(
            E,
            te,
            le,
            xe,
            y,
            c.children,
            m,
            _,
            je
          );
        }
      }
      v & 1 && c.children !== d.children && p(E, d.children);
    } else
      !x && w == null && Q(
        E,
        d,
        S,
        T,
        m,
        _,
        y
      );
    ((F = T.onVnodeUpdated) || C) && _e(() => {
      F && Ae(F, m, d, c), C && ht(d, c, m, "updated");
    }, _);
  }, ie = (c, d, m, _, y, b, x) => {
    for (let E = 0; E < d.length; E++) {
      const v = c[E], w = d[E], C = (
        // oldVNode may be an errored async setup() component inside Suspense
        // which will not have a mounted element
        v.el && // - In the case of a Fragment, we need to provide the actual parent
        // of the Fragment itself so it can move its children.
        (v.type === we || // - In the case of different nodes, there is going to be a replacement
        // which also requires the correct parent container
        !Yt(v, w) || // - In the case of a component, it could contain anything.
        v.shapeFlag & 70) ? f(v.el) : (
          // In other cases, the parent container is not actually used so we
          // just pass the block element here to avoid a DOM parentNode call.
          m
        )
      );
      $(
        v,
        w,
        C,
        null,
        _,
        y,
        b,
        x,
        !0
      );
    }
  }, Q = (c, d, m, _, y, b, x) => {
    if (m !== _) {
      if (m !== X)
        for (const E in m)
          !Zt(E) && !(E in _) && s(
            c,
            E,
            m[E],
            null,
            x,
            d.children,
            y,
            b,
            je
          );
      for (const E in _) {
        if (Zt(E))
          continue;
        const v = _[E], w = m[E];
        v !== w && E !== "value" && s(
          c,
          E,
          w,
          v,
          x,
          d.children,
          y,
          b,
          je
        );
      }
      "value" in _ && s(c, "value", m.value, _.value, x);
    }
  }, ve = (c, d, m, _, y, b, x, E, v) => {
    const w = d.el = c ? c.el : l(""), C = d.anchor = c ? c.anchor : l("");
    let { patchFlag: S, dynamicChildren: T, slotScopeIds: F } = d;
    ({}).NODE_ENV !== "production" && // #5523 dev root fragment may inherit directives
    (Ot || S & 2048) && (S = 0, v = !1, T = null), F && (E = E ? E.concat(F) : F), c == null ? (o(w, m, _), o(C, m, _), J(
      // #10007
      // such fragment like `<></>` will be compiled into
      // a fragment which doesn't have a children.
      // In this case fallback to an empty array
      d.children || [],
      m,
      C,
      y,
      b,
      x,
      E,
      v
    )) : S > 0 && S & 64 && T && // #2715 the previous fragment could've been a BAILed one as a result
    // of renderSlot() with no valid children
    c.dynamicChildren ? (ie(
      c.dynamicChildren,
      T,
      m,
      y,
      b,
      x,
      E
    ), {}.NODE_ENV !== "production" ? In(c, d) : (
      // #2080 if the stable fragment has a key, it's a <template v-for> that may
      //  get moved around. Make sure all root level vnodes inherit el.
      // #2134 or if it's a component root, it may also get moved around
      // as the component is being moved.
      (d.key != null || y && d === y.subTree) && In(
        c,
        d,
        !0
        /* shallow */
      )
    )) : Ve(
      c,
      d,
      m,
      C,
      y,
      b,
      x,
      E,
      v
    );
  }, de = (c, d, m, _, y, b, x, E, v) => {
    d.slotScopeIds = E, c == null ? d.shapeFlag & 512 ? y.ctx.activate(
      d,
      m,
      _,
      x,
      v
    ) : re(
      d,
      m,
      _,
      y,
      b,
      x,
      v
    ) : ee(c, d, v);
  }, re = (c, d, m, _, y, b, x) => {
    const E = c.component = Du(
      c,
      _,
      y
    );
    if ({}.NODE_ENV !== "production" && E.type.__hmrId && lc(E), {}.NODE_ENV !== "production" && (An(c), ke(E, "mount")), Ei(c) && (E.ctx.renderer = At), {}.NODE_ENV !== "production" && ke(E, "init"), Su(E), {}.NODE_ENV !== "production" && Be(E, "init"), E.asyncDep) {
      if (y && y.registerDep(E, L), !c.el) {
        const v = E.subTree = ge(ye);
        P(null, v, d, m);
      }
    } else
      L(
        E,
        c,
        d,
        m,
        y,
        b,
        x
      );
    ({}).NODE_ENV !== "production" && (Rn(), Be(E, "mount"));
  }, ee = (c, d, m) => {
    const _ = d.component = c.component;
    if (xc(c, d, m))
      if (_.asyncDep && !_.asyncResolved) {
        ({}).NODE_ENV !== "production" && An(d), B(_, d, m), {}.NODE_ENV !== "production" && Rn();
        return;
      } else
        _.next = d, sc(_.update), _.effect.dirty = !0, _.update();
    else
      d.el = c.el, _.vnode = d;
  }, L = (c, d, m, _, y, b, x) => {
    const E = () => {
      if (c.isMounted) {
        let { next: C, bu: S, u: T, parent: F, vnode: W } = c;
        {
          const Rt = Pr(c);
          if (Rt) {
            C && (C.el = W.el, B(c, C, x)), Rt.asyncDep.then(() => {
              c.isUnmounted || E();
            });
            return;
          }
        }
        let G = C, te;
        ({}).NODE_ENV !== "production" && An(C || c.vnode), mt(c, !1), C ? (C.el = W.el, B(c, C, x)) : C = W, S && Gt(S), (te = C.props && C.props.onVnodeBeforeUpdate) && Ae(te, F, C, W), mt(c, !0), {}.NODE_ENV !== "production" && ke(c, "render");
        const le = Oo(c);
        ({}).NODE_ENV !== "production" && Be(c, "render");
        const xe = c.subTree;
        c.subTree = le, {}.NODE_ENV !== "production" && ke(c, "patch"), $(
          xe,
          le,
          // parent may have changed if it's in a teleport
          f(xe.el),
          // anchor may have changed if it's in a fragment
          On(xe),
          c,
          y,
          b
        ), {}.NODE_ENV !== "production" && Be(c, "patch"), C.el = le.el, G === null && $c(c, le.el), T && _e(T, y), (te = C.props && C.props.onVnodeUpdated) && _e(
          () => Ae(te, F, C, W),
          y
        ), {}.NODE_ENV !== "production" && pr(c), {}.NODE_ENV !== "production" && Rn();
      } else {
        let C;
        const { el: S, props: T } = d, { bm: F, m: W, parent: G } = c, te = en(d);
        if (mt(c, !1), F && Gt(F), !te && (C = T && T.onVnodeBeforeMount) && Ae(C, G, d), mt(c, !0), S && bo) {
          const le = () => {
            ({}).NODE_ENV !== "production" && ke(c, "render"), c.subTree = Oo(c), {}.NODE_ENV !== "production" && Be(c, "render"), {}.NODE_ENV !== "production" && ke(c, "hydrate"), bo(
              S,
              c.subTree,
              c,
              y,
              null
            ), {}.NODE_ENV !== "production" && Be(c, "hydrate");
          };
          te ? d.type.__asyncLoader().then(
            // note: we are moving the render call into an async callback,
            // which means it won't track dependencies - but it's ok because
            // a server-rendered async wrapper is already in resolved state
            // and it will never need to change.
            () => !c.isUnmounted && le()
          ) : le();
        } else {
          ({}).NODE_ENV !== "production" && ke(c, "render");
          const le = c.subTree = Oo(c);
          ({}).NODE_ENV !== "production" && Be(c, "render"), {}.NODE_ENV !== "production" && ke(c, "patch"), $(
            null,
            le,
            m,
            _,
            c,
            y,
            b
          ), {}.NODE_ENV !== "production" && Be(c, "patch"), d.el = le.el;
        }
        if (W && _e(W, y), !te && (C = T && T.onVnodeMounted)) {
          const le = d;
          _e(
            () => Ae(C, G, le),
            y
          );
        }
        (d.shapeFlag & 256 || G && en(G.vnode) && G.vnode.shapeFlag & 256) && c.a && _e(c.a, y), c.isMounted = !0, {}.NODE_ENV !== "production" && pc(c), d = m = _ = null;
      }
    }, v = c.effect = new ci(
      E,
      se,
      () => so(w),
      c.scope
      // track it in component's effect scope
    ), w = c.update = () => {
      v.dirty && v.run();
    };
    w.id = c.uid, mt(c, !0), {}.NODE_ENV !== "production" && (v.onTrack = c.rtc ? (C) => Gt(c.rtc, C) : void 0, v.onTrigger = c.rtg ? (C) => Gt(c.rtg, C) : void 0, w.ownerInstance = c), w();
  }, B = (c, d, m) => {
    d.component = c;
    const _ = c.vnode.props;
    c.vnode = d, c.next = null, lu(c, d.props, _, m), mu(c, d.children, m), ft(), zi(c), dt();
  }, Ve = (c, d, m, _, y, b, x, E, v = !1) => {
    const w = c && c.children, C = c ? c.shapeFlag : 0, S = d.children, { patchFlag: T, shapeFlag: F } = d;
    if (T > 0) {
      if (T & 128) {
        Ut(
          w,
          S,
          m,
          _,
          y,
          b,
          x,
          E,
          v
        );
        return;
      } else if (T & 256) {
        _o(
          w,
          S,
          m,
          _,
          y,
          b,
          x,
          E,
          v
        );
        return;
      }
    }
    F & 8 ? (C & 16 && je(w, y, b), S !== w && p(m, S)) : C & 16 ? F & 16 ? Ut(
      w,
      S,
      m,
      _,
      y,
      b,
      x,
      E,
      v
    ) : je(w, y, b, !0) : (C & 8 && p(m, ""), F & 16 && J(
      S,
      m,
      _,
      y,
      b,
      x,
      E,
      v
    ));
  }, _o = (c, d, m, _, y, b, x, E, v) => {
    c = c || Ft, d = d || Ft;
    const w = c.length, C = d.length, S = Math.min(w, C);
    let T;
    for (T = 0; T < S; T++) {
      const F = d[T] = v ? nt(d[T]) : Ce(d[T]);
      $(
        c[T],
        F,
        m,
        null,
        y,
        b,
        x,
        E,
        v
      );
    }
    w > C ? je(
      c,
      y,
      b,
      !0,
      !1,
      S
    ) : J(
      d,
      m,
      _,
      y,
      b,
      x,
      E,
      v,
      S
    );
  }, Ut = (c, d, m, _, y, b, x, E, v) => {
    let w = 0;
    const C = d.length;
    let S = c.length - 1, T = C - 1;
    for (; w <= S && w <= T; ) {
      const F = c[w], W = d[w] = v ? nt(d[w]) : Ce(d[w]);
      if (Yt(F, W))
        $(
          F,
          W,
          m,
          null,
          y,
          b,
          x,
          E,
          v
        );
      else
        break;
      w++;
    }
    for (; w <= S && w <= T; ) {
      const F = c[S], W = d[T] = v ? nt(d[T]) : Ce(d[T]);
      if (Yt(F, W))
        $(
          F,
          W,
          m,
          null,
          y,
          b,
          x,
          E,
          v
        );
      else
        break;
      S--, T--;
    }
    if (w > S) {
      if (w <= T) {
        const F = T + 1, W = F < C ? d[F].el : _;
        for (; w <= T; )
          $(
            null,
            d[w] = v ? nt(d[w]) : Ce(d[w]),
            m,
            W,
            y,
            b,
            x,
            E,
            v
          ), w++;
      }
    } else if (w > T)
      for (; w <= S; )
        Xe(c[w], y, b, !0), w++;
    else {
      const F = w, W = w, G = /* @__PURE__ */ new Map();
      for (w = W; w <= T; w++) {
        const he = d[w] = v ? nt(d[w]) : Ce(d[w]);
        he.key != null && ({}.NODE_ENV !== "production" && G.has(he.key) && N(
          "Duplicate keys found during update:",
          JSON.stringify(he.key),
          "Make sure keys are unique."
        ), G.set(he.key, w));
      }
      let te, le = 0;
      const xe = T - W + 1;
      let Rt = !1, Pi = 0;
      const Kt = new Array(xe);
      for (w = 0; w < xe; w++)
        Kt[w] = 0;
      for (w = F; w <= S; w++) {
        const he = c[w];
        if (le >= xe) {
          Xe(he, y, b, !0);
          continue;
        }
        let Pe;
        if (he.key != null)
          Pe = G.get(he.key);
        else
          for (te = W; te <= T; te++)
            if (Kt[te - W] === 0 && Yt(he, d[te])) {
              Pe = te;
              break;
            }
        Pe === void 0 ? Xe(he, y, b, !0) : (Kt[Pe - W] = w + 1, Pe >= Pi ? Pi = Pe : Rt = !0, $(
          he,
          d[Pe],
          m,
          null,
          y,
          b,
          x,
          E,
          v
        ), le++);
      }
      const Ai = Rt ? Eu(Kt) : Ft;
      for (te = Ai.length - 1, w = xe - 1; w >= 0; w--) {
        const he = W + w, Pe = d[he], Ri = he + 1 < C ? d[he + 1].el : _;
        Kt[w] === 0 ? $(
          null,
          Pe,
          m,
          Ri,
          y,
          b,
          x,
          E,
          v
        ) : Rt && (te < 0 || w !== Ai[te] ? Pt(Pe, m, Ri, 2) : te--);
      }
    }
  }, Pt = (c, d, m, _, y = null) => {
    const { el: b, type: x, transition: E, children: v, shapeFlag: w } = c;
    if (w & 6) {
      Pt(c.component.subTree, d, m, _);
      return;
    }
    if (w & 128) {
      c.suspense.move(d, m, _);
      return;
    }
    if (w & 64) {
      x.move(c, d, m, At);
      return;
    }
    if (x === we) {
      o(b, d, m);
      for (let S = 0; S < v.length; S++)
        Pt(v[S], d, m, _);
      o(c.anchor, d, m);
      return;
    }
    if (x === Ln) {
      j(c, d, m);
      return;
    }
    if (_ !== 2 && w & 1 && E)
      if (_ === 0)
        E.beforeEnter(b), o(b, d, m), _e(() => E.enter(b), y);
      else {
        const { leave: S, delayLeave: T, afterLeave: F } = E, W = () => o(b, d, m), G = () => {
          S(b, () => {
            W(), F && F();
          });
        };
        T ? T(b, W, G) : G();
      }
    else
      o(b, d, m);
  }, Xe = (c, d, m, _ = !1, y = !1) => {
    const {
      type: b,
      props: x,
      ref: E,
      children: v,
      dynamicChildren: w,
      shapeFlag: C,
      patchFlag: S,
      dirs: T
    } = c;
    if (E != null && Yo(E, null, m, c, !0), C & 256) {
      d.ctx.deactivate(c);
      return;
    }
    const F = C & 1 && T, W = !en(c);
    let G;
    if (W && (G = x && x.onVnodeBeforeUnmount) && Ae(G, d, c), C & 6)
      il(c.component, m, _);
    else {
      if (C & 128) {
        c.suspense.unmount(m, _);
        return;
      }
      F && ht(c, null, d, "beforeUnmount"), C & 64 ? c.type.remove(
        c,
        d,
        m,
        y,
        At,
        _
      ) : w && // #1153: fast path should not be taken for non-stable (v-for) fragments
      (b !== we || S > 0 && S & 64) ? je(
        w,
        d,
        m,
        !1,
        !0
      ) : (b === we && S & 384 || !y && C & 16) && je(v, d, m), _ && yo(c);
    }
    (W && (G = x && x.onVnodeUnmounted) || F) && _e(() => {
      G && Ae(G, d, c), F && ht(c, null, d, "unmounted");
    }, m);
  }, yo = (c) => {
    const { type: d, el: m, anchor: _, transition: y } = c;
    if (d === we) {
      ({}).NODE_ENV !== "production" && c.patchFlag > 0 && c.patchFlag & 2048 && y && !y.persisted ? c.children.forEach((x) => {
        x.type === ye ? i(x.el) : yo(x);
      }) : ol(m, _);
      return;
    }
    if (d === Ln) {
      I(c);
      return;
    }
    const b = () => {
      i(m), y && !y.persisted && y.afterLeave && y.afterLeave();
    };
    if (c.shapeFlag & 1 && y && !y.persisted) {
      const { leave: x, delayLeave: E } = y, v = () => x(m, b);
      E ? E(c.el, b, v) : v();
    } else
      b();
  }, ol = (c, d) => {
    let m;
    for (; c !== d; )
      m = h(c), i(c), c = m;
    i(d);
  }, il = (c, d, m) => {
    ({}).NODE_ENV !== "production" && c.type.__hmrId && cc(c);
    const { bum: _, scope: y, update: b, subTree: x, um: E } = c;
    _ && Gt(_), y.stop(), b && (b.active = !1, Xe(x, c, d, m)), E && _e(E, d), _e(() => {
      c.isUnmounted = !0;
    }, d), d && d.pendingBranch && !d.isUnmounted && c.asyncDep && !c.asyncResolved && c.suspenseId === d.pendingId && (d.deps--, d.deps === 0 && d.resolve()), {}.NODE_ENV !== "production" && mc(c);
  }, je = (c, d, m, _ = !1, y = !1, b = 0) => {
    for (let x = b; x < c.length; x++)
      Xe(c[x], d, m, _, y);
  }, On = (c) => c.shapeFlag & 6 ? On(c.component.subTree) : c.shapeFlag & 128 ? c.suspense.next() : h(c.anchor || c.el);
  let wo = !1;
  const Vi = (c, d, m) => {
    c == null ? d._vnode && Xe(d._vnode, null, null, !0) : $(
      d._vnode || null,
      c,
      d,
      null,
      null,
      null,
      m
    ), wo || (wo = !0, zi(), ur(), wo = !1), d._vnode = c;
  }, At = {
    p: $,
    um: Xe,
    m: Pt,
    r: yo,
    mt: re,
    mc: J,
    pc: Ve,
    pbc: ie,
    n: On,
    o: e
  };
  let Eo, bo;
  return t && ([Eo, bo] = t(
    At
  )), {
    render: Vi,
    hydrate: Eo,
    createApp: ou(Vi, Eo)
  };
}
function Co({ type: e, props: t }, n) {
  return n === "svg" && e === "foreignObject" || n === "mathml" && e === "annotation-xml" && t && t.encoding && t.encoding.includes("html") ? void 0 : n;
}
function mt({ effect: e, update: t }, n) {
  e.allowRecurse = t.allowRecurse = n;
}
function wu(e, t) {
  return (!e || e && !e.pendingBranch) && t && !t.persisted;
}
function In(e, t, n = !1) {
  const o = e.children, i = t.children;
  if (R(o) && R(i))
    for (let s = 0; s < o.length; s++) {
      const r = o[s];
      let l = i[s];
      l.shapeFlag & 1 && !l.dynamicChildren && ((l.patchFlag <= 0 || l.patchFlag === 32) && (l = i[s] = nt(i[s]), l.el = r.el), n || In(r, l)), l.type === wn && (l.el = r.el), {}.NODE_ENV !== "production" && l.type === ye && !l.el && (l.el = r.el);
    }
}
function Eu(e) {
  const t = e.slice(), n = [0];
  let o, i, s, r, l;
  const u = e.length;
  for (o = 0; o < u; o++) {
    const a = e[o];
    if (a !== 0) {
      if (i = n[n.length - 1], e[i] < a) {
        t[o] = i, n.push(o);
        continue;
      }
      for (s = 0, r = n.length - 1; s < r; )
        l = s + r >> 1, e[n[l]] < a ? s = l + 1 : r = l;
      a < e[n[s]] && (s > 0 && (t[o] = n[s - 1]), n[s] = o);
    }
  }
  for (s = n.length, r = n[s - 1]; s-- > 0; )
    n[s] = r, r = t[r];
  return n;
}
function Pr(e) {
  const t = e.subTree.component;
  if (t)
    return t.asyncDep && !t.asyncResolved ? t : Pr(t);
}
const bu = (e) => e.__isTeleport, we = Symbol.for("v-fgt"), wn = Symbol.for("v-txt"), ye = Symbol.for("v-cmt"), Ln = Symbol.for("v-stc"), nn = [];
let De = null;
function Ke(e = !1) {
  nn.push(De = e ? null : []);
}
function vu() {
  nn.pop(), De = nn[nn.length - 1] || null;
}
let an = 1;
function os(e) {
  an += e;
}
function Ar(e) {
  return e.dynamicChildren = an > 0 ? De || Ft : null, vu(), an > 0 && De && De.push(e), e;
}
function Gn(e, t, n, o, i, s) {
  return Ar(
    We(
      e,
      t,
      n,
      o,
      i,
      s,
      !0
    )
  );
}
function En(e, t, n, o, i) {
  return Ar(
    ge(
      e,
      t,
      n,
      o,
      i,
      !0
    )
  );
}
function Bt(e) {
  return e ? e.__v_isVNode === !0 : !1;
}
function Yt(e, t) {
  return {}.NODE_ENV !== "production" && t.shapeFlag & 6 && Mt.has(t.type) ? (e.shapeFlag &= -257, t.shapeFlag &= -513, !1) : e.type === t.type && e.key === t.key;
}
const Nu = (...e) => Mr(
  ...e
), uo = "__vInternal", Rr = ({ key: e }) => e ?? null, Fn = ({
  ref: e,
  ref_key: t,
  ref_for: n
}) => (typeof e == "number" && (e = "" + e), e != null ? oe(e) || fe(e) || M(e) ? { i: ue, r: e, k: t, f: !!n } : e : null);
function We(e, t = null, n = null, o = 0, i = null, s = e === we ? 0 : 1, r = !1, l = !1) {
  const u = {
    __v_isVNode: !0,
    __v_skip: !0,
    type: e,
    props: t,
    key: t && Rr(t),
    ref: t && Fn(t),
    scopeId: lo,
    slotScopeIds: null,
    children: n,
    component: null,
    suspense: null,
    ssContent: null,
    ssFallback: null,
    dirs: null,
    transition: null,
    el: null,
    anchor: null,
    target: null,
    targetAnchor: null,
    staticCount: 0,
    shapeFlag: s,
    patchFlag: o,
    dynamicProps: i,
    dynamicChildren: null,
    appContext: null,
    ctx: ue
  };
  return l ? (Oi(u, n), s & 128 && e.normalize(u)) : n && (u.shapeFlag |= oe(n) ? 8 : 16), {}.NODE_ENV !== "production" && u.key !== u.key && N("VNode created with invalid key (NaN). VNode type:", u.type), an > 0 && // avoid a block node from tracking itself
  !r && // has current parent block
  De && // presence of a patch flag indicates this node needs patching on updates.
  // component nodes also should always be patched, because even if the
  // component doesn't need to update, it needs to persist the instance on to
  // the next vnode so that it can be properly unmounted later.
  (u.patchFlag > 0 || s & 6) && // the EVENTS flag is only for hydration and if it is the only flag, the
  // vnode should not be considered dynamic due to handler caching.
  u.patchFlag !== 32 && De.push(u), u;
}
const ge = {}.NODE_ENV !== "production" ? Nu : Mr;
function Mr(e, t = null, n = null, o = 0, i = null, s = !1) {
  if ((!e || e === Cc) && ({}.NODE_ENV !== "production" && !e && N(`Invalid vnode type when creating vnode: ${e}.`), e = ye), Bt(e)) {
    const l = ct(
      e,
      t,
      !0
      /* mergeRef: true */
    );
    return n && Oi(l, n), an > 0 && !s && De && (l.shapeFlag & 6 ? De[De.indexOf(e)] = l : De.push(l)), l.patchFlag |= -2, l;
  }
  if (jr(e) && (e = e.__vccOpts), t) {
    t = Ou(t);
    let { class: l, style: u } = t;
    l && !oe(l) && (t.class = gn(l)), q(u) && (Lo(u) && !R(u) && (u = ne({}, u)), t.style = Ht(u));
  }
  const r = oe(e) ? 1 : Tc(e) ? 128 : bu(e) ? 64 : q(e) ? 4 : M(e) ? 2 : 0;
  return {}.NODE_ENV !== "production" && r & 4 && Lo(e) && (e = k(e), N(
    "Vue received a Component that was made a reactive object. This can lead to unnecessary performance overhead and should be avoided by marking the component with `markRaw` or using `shallowRef` instead of `ref`.",
    `
Component that was made reactive: `,
    e
  )), We(
    e,
    t,
    n,
    o,
    i,
    r,
    s,
    !0
  );
}
function Ou(e) {
  return e ? Lo(e) || uo in e ? ne({}, e) : e : null;
}
function ct(e, t, n = !1) {
  const { props: o, ref: i, patchFlag: s, children: r } = e, l = t ? Lr(o || {}, t) : o;
  return {
    __v_isVNode: !0,
    __v_skip: !0,
    type: e.type,
    props: l,
    key: l && Rr(l),
    ref: t && t.ref ? (
      // #2078 in the case of <component :is="vnode" ref="extra"/>
      // if the vnode itself already has a ref, cloneVNode will need to merge
      // the refs so the single vnode can be set on multiple refs
      n && i ? R(i) ? i.concat(Fn(t)) : [i, Fn(t)] : Fn(t)
    ) : i,
    scopeId: e.scopeId,
    slotScopeIds: e.slotScopeIds,
    children: {}.NODE_ENV !== "production" && s === -1 && R(r) ? r.map(Ir) : r,
    target: e.target,
    targetAnchor: e.targetAnchor,
    staticCount: e.staticCount,
    shapeFlag: e.shapeFlag,
    // if the vnode is cloned with extra props, we can no longer assume its
    // existing patch flag to be reliable and need to add the FULL_PROPS flag.
    // note: preserve flag for fragments since they use the flag for children
    // fast paths only.
    patchFlag: t && e.type !== we ? s === -1 ? 16 : s | 16 : s,
    dynamicProps: e.dynamicProps,
    dynamicChildren: e.dynamicChildren,
    appContext: e.appContext,
    dirs: e.dirs,
    transition: e.transition,
    // These should technically only be non-null on mounted VNodes. However,
    // they *should* be copied for kept-alive vnodes. So we just always copy
    // them since them being non-null during a mount doesn't affect the logic as
    // they will simply be overwritten.
    component: e.component,
    suspense: e.suspense,
    ssContent: e.ssContent && ct(e.ssContent),
    ssFallback: e.ssFallback && ct(e.ssFallback),
    el: e.el,
    anchor: e.anchor,
    ctx: e.ctx,
    ce: e.ce
  };
}
function Ir(e) {
  const t = ct(e);
  return R(e.children) && (t.children = e.children.map(Ir)), t;
}
function xu(e = " ", t = 0) {
  return ge(wn, null, e, t);
}
function is(e = "", t = !1) {
  return t ? (Ke(), En(ye, null, e)) : ge(ye, null, e);
}
function Ce(e) {
  return e == null || typeof e == "boolean" ? ge(ye) : R(e) ? ge(
    we,
    null,
    // #3666, avoid reference pollution when reusing vnode
    e.slice()
  ) : typeof e == "object" ? nt(e) : ge(wn, null, String(e));
}
function nt(e) {
  return e.el === null && e.patchFlag !== -1 || e.memo ? e : ct(e);
}
function Oi(e, t) {
  let n = 0;
  const { shapeFlag: o } = e;
  if (t == null)
    t = null;
  else if (R(t))
    n = 16;
  else if (typeof t == "object")
    if (o & 65) {
      const i = t.default;
      i && (i._c && (i._d = !1), Oi(e, i()), i._c && (i._d = !0));
      return;
    } else {
      n = 32;
      const i = t._;
      !i && !(uo in t) ? t._ctx = ue : i === 3 && ue && (ue.slots._ === 1 ? t._ = 1 : (t._ = 2, e.patchFlag |= 1024));
    }
  else
    M(t) ? (t = { default: t, _ctx: ue }, n = 32) : (t = String(t), o & 64 ? (n = 16, t = [xu(t)]) : n = 8);
  e.children = t, e.shapeFlag |= n;
}
function Lr(...e) {
  const t = {};
  for (let n = 0; n < e.length; n++) {
    const o = e[n];
    for (const i in o)
      if (i === "class")
        t.class !== o.class && (t.class = gn([t.class, o.class]));
      else if (i === "style")
        t.style = Ht([t.style, o.style]);
      else if (mn(i)) {
        const s = t[i], r = o[i];
        r && s !== r && !(R(s) && s.includes(r)) && (t[i] = s ? [].concat(s, r) : r);
      } else
        i !== "" && (t[i] = o[i]);
  }
  return t;
}
function Ae(e, t, n, o = null) {
  Te(e, t, 7, [
    n,
    o
  ]);
}
const $u = xr();
let Cu = 0;
function Du(e, t, n) {
  const o = e.type, i = (t ? t.appContext : e.appContext) || $u, s = {
    uid: Cu++,
    vnode: e,
    type: o,
    parent: t,
    appContext: i,
    root: null,
    // to be immediately set
    next: null,
    subTree: null,
    // will be set synchronously right after creation
    effect: null,
    update: null,
    // will be set synchronously right after creation
    scope: new Nl(
      !0
      /* detached */
    ),
    render: null,
    proxy: null,
    exposed: null,
    exposeProxy: null,
    withProxy: null,
    provides: t ? t.provides : Object.create(i.provides),
    accessCache: null,
    renderCache: [],
    // local resolved assets
    components: null,
    directives: null,
    // resolved props and emits options
    propsOptions: Cr(o, i),
    emitsOptions: mr(o, i),
    // emit
    emit: null,
    // to be set immediately
    emitted: null,
    // props default value
    propsDefaults: X,
    // inheritAttrs
    inheritAttrs: o.inheritAttrs,
    // state
    ctx: X,
    data: X,
    props: X,
    attrs: X,
    slots: X,
    refs: X,
    setupState: X,
    setupContext: null,
    attrsProxy: null,
    slotsProxy: null,
    // suspense related
    suspense: n,
    suspenseId: n ? n.pendingId : 0,
    asyncDep: null,
    asyncResolved: !1,
    // lifecycle hooks
    // not using enums here because it results in computed properties
    isMounted: !1,
    isUnmounted: !1,
    isDeactivated: !1,
    bc: null,
    c: null,
    bm: null,
    m: null,
    bu: null,
    u: null,
    um: null,
    bum: null,
    da: null,
    a: null,
    rtg: null,
    rtc: null,
    ec: null,
    sp: null
  };
  return {}.NODE_ENV !== "production" ? s.ctx = Gc(s) : s.ctx = { _: s }, s.root = t ? t.root : s, s.emit = wc.bind(null, s), e.ce && e.ce(s), s;
}
let ce = null, qn, Jo;
{
  const e = li(), t = (n, o) => {
    let i;
    return (i = e[n]) || (i = e[n] = []), i.push(o), (s) => {
      i.length > 1 ? i.forEach((r) => r(s)) : i[0](s);
    };
  };
  qn = t(
    "__VUE_INSTANCE_SETTERS__",
    (n) => ce = n
  ), Jo = t(
    "__VUE_SSR_SETTERS__",
    (n) => ao = n
  );
}
const bn = (e) => {
  const t = ce;
  return qn(e), e.scope.on(), () => {
    e.scope.off(), qn(t);
  };
}, ss = () => {
  ce && ce.scope.off(), qn(null);
}, Tu = /* @__PURE__ */ Ye("slot,component");
function Xo(e, t) {
  const n = t.isNativeTag || As;
  (Tu(e) || n(e)) && N(
    "Do not use built-in or reserved HTML elements as component id: " + e
  );
}
function Fr(e) {
  return e.vnode.shapeFlag & 4;
}
let ao = !1;
function Su(e, t = !1) {
  t && Jo(t);
  const { props: n, children: o } = e.vnode, i = Fr(e);
  su(e, n, i, t), hu(e, o);
  const s = i ? Vu(e, t) : void 0;
  return t && Jo(!1), s;
}
function Vu(e, t) {
  var n;
  const o = e.type;
  if ({}.NODE_ENV !== "production") {
    if (o.name && Xo(o.name, e.appContext.config), o.components) {
      const s = Object.keys(o.components);
      for (let r = 0; r < s.length; r++)
        Xo(s[r], e.appContext.config);
    }
    if (o.directives) {
      const s = Object.keys(o.directives);
      for (let r = 0; r < s.length; r++)
        wr(s[r]);
    }
    o.compilerOptions && Pu() && N(
      '"compilerOptions" is only supported when using a build of Vue that includes the runtime compiler. Since you are using a runtime-only build, the options should be passed via your build tool config instead.'
    );
  }
  e.accessCache = /* @__PURE__ */ Object.create(null), e.proxy = er(new Proxy(e.ctx, Nr)), {}.NODE_ENV !== "production" && qc(e);
  const { setup: i } = o;
  if (i) {
    const s = e.setupContext = i.length > 1 ? Ru(e) : null, r = bn(e);
    ft();
    const l = Ue(
      i,
      e,
      0,
      [
        {}.NODE_ENV !== "production" ? It(e.props) : e.props,
        s
      ]
    );
    if (dt(), r(), ii(l)) {
      if (l.then(ss, ss), t)
        return l.then((u) => {
          rs(e, u, t);
        }).catch((u) => {
          _n(u, e, 0);
        });
      if (e.asyncDep = l, {}.NODE_ENV !== "production" && !e.suspense) {
        const u = (n = o.name) != null ? n : "Anonymous";
        N(
          `Component <${u}>: setup function returned a promise, but no <Suspense> boundary was found in the parent component tree. A component with async setup() must be nested in a <Suspense> in order to be rendered.`
        );
      }
    } else
      rs(e, l, t);
  } else
    Hr(e, t);
}
function rs(e, t, n) {
  M(t) ? e.type.__ssrInlineRender ? e.ssrRender = t : e.render = t : q(t) ? ({}.NODE_ENV !== "production" && Bt(t) && N(
    "setup() should not return VNodes directly - return a render function instead."
  ), {}.NODE_ENV !== "production" && (e.devtoolsRawSetupState = t), e.setupState = or(t), {}.NODE_ENV !== "production" && Yc(e)) : {}.NODE_ENV !== "production" && t !== void 0 && N(
    `setup() should return an object. Received: ${t === null ? "null" : typeof t}`
  ), Hr(e, n);
}
let Zo;
const Pu = () => !Zo;
function Hr(e, t, n) {
  const o = e.type;
  if (!e.render) {
    if (!t && Zo && !o.render) {
      const i = o.template || vi(e).template;
      if (i) {
        ({}).NODE_ENV !== "production" && ke(e, "compile");
        const { isCustomElement: s, compilerOptions: r } = e.appContext.config, { delimiters: l, compilerOptions: u } = o, a = ne(
          ne(
            {
              isCustomElement: s,
              delimiters: l
            },
            r
          ),
          u
        );
        o.render = Zo(i, a), {}.NODE_ENV !== "production" && Be(e, "compile");
      }
    }
    e.render = o.render || se;
  }
  {
    const i = bn(e);
    ft();
    try {
      Xc(e);
    } finally {
      dt(), i();
    }
  }
  ({}).NODE_ENV !== "production" && !o.render && e.render === se && !t && (o.template ? N(
    'Component provided template option but runtime compilation is not supported in this build of Vue. Configure your bundler to alias "vue" to "vue/dist/vue.esm-bundler.js".'
  ) : N("Component is missing template or render function."));
}
function ls(e) {
  return e.attrsProxy || (e.attrsProxy = new Proxy(
    e.attrs,
    {}.NODE_ENV !== "production" ? {
      get(t, n) {
        return Wn(), ae(e, "get", "$attrs"), t[n];
      },
      set() {
        return N("setupContext.attrs is readonly."), !1;
      },
      deleteProperty() {
        return N("setupContext.attrs is readonly."), !1;
      }
    } : {
      get(t, n) {
        return ae(e, "get", "$attrs"), t[n];
      }
    }
  ));
}
function Au(e) {
  return e.slotsProxy || (e.slotsProxy = new Proxy(e.slots, {
    get(t, n) {
      return ae(e, "get", "$slots"), t[n];
    }
  }));
}
function Ru(e) {
  const t = (n) => {
    if ({}.NODE_ENV !== "production" && (e.exposed && N("expose() should be called only once per setup()."), n != null)) {
      let o = typeof n;
      o === "object" && (R(n) ? o = "array" : fe(n) && (o = "ref")), o !== "object" && N(
        `expose() should be passed a plain object, received ${o}.`
      );
    }
    e.exposed = n || {};
  };
  return {}.NODE_ENV !== "production" ? Object.freeze({
    get attrs() {
      return ls(e);
    },
    get slots() {
      return Au(e);
    },
    get emit() {
      return (n, ...o) => e.emit(n, ...o);
    },
    expose: t
  }) : {
    get attrs() {
      return ls(e);
    },
    slots: e.slots,
    emit: e.emit,
    expose: t
  };
}
function xi(e) {
  if (e.exposed)
    return e.exposeProxy || (e.exposeProxy = new Proxy(or(er(e.exposed)), {
      get(t, n) {
        if (n in t)
          return t[n];
        if (n in xt)
          return xt[n](e);
      },
      has(t, n) {
        return n in t || n in xt;
      }
    }));
}
const Mu = /(?:^|[-_])(\w)/g, Iu = (e) => e.replace(Mu, (t) => t.toUpperCase()).replace(/[-_]/g, "");
function $i(e, t = !0) {
  return M(e) ? e.displayName || e.name : e.name || t && e.__name;
}
function fo(e, t, n = !1) {
  let o = $i(t);
  if (!o && t.__file) {
    const i = t.__file.match(/([^/\\]+)\.\w+$/);
    i && (o = i[1]);
  }
  if (!o && e && e.parent) {
    const i = (s) => {
      for (const r in s)
        if (s[r] === t)
          return r;
    };
    o = i(
      e.components || e.parent.type.components
    ) || i(e.appContext.components);
  }
  return o ? Iu(o) : n ? "App" : "Anonymous";
}
function jr(e) {
  return M(e) && "__vccOpts" in e;
}
const Lu = (e, t) => Gl(e, t, ao);
function Fu(e, t, n) {
  const o = arguments.length;
  return o === 2 ? q(t) && !R(t) ? Bt(t) ? ge(e, null, [t]) : ge(e, t) : ge(e, null, t) : (o > 3 ? n = Array.prototype.slice.call(arguments, 2) : o === 3 && Bt(n) && (n = [n]), ge(e, t, n));
}
function Hu() {
  if ({}.NODE_ENV === "production" || typeof window > "u")
    return;
  const e = { style: "color:#3ba776" }, t = { style: "color:#1677ff" }, n = { style: "color:#f5222d" }, o = { style: "color:#eb2f96" }, i = {
    header(f) {
      return q(f) ? f.__isVue ? ["div", e, "VueInstance"] : fe(f) ? [
        "div",
        {},
        ["span", e, p(f)],
        "<",
        l(f.value),
        ">"
      ] : bt(f) ? [
        "div",
        {},
        ["span", e, vt(f) ? "ShallowReactive" : "Reactive"],
        "<",
        l(f),
        `>${lt(f) ? " (readonly)" : ""}`
      ] : lt(f) ? [
        "div",
        {},
        ["span", e, vt(f) ? "ShallowReadonly" : "Readonly"],
        "<",
        l(f),
        ">"
      ] : null : null;
    },
    hasBody(f) {
      return f && f.__isVue;
    },
    body(f) {
      if (f && f.__isVue)
        return [
          "div",
          {},
          ...s(f.$)
        ];
    }
  };
  function s(f) {
    const h = [];
    f.type.props && f.props && h.push(r("props", k(f.props))), f.setupState !== X && h.push(r("setup", f.setupState)), f.data !== X && h.push(r("data", k(f.data)));
    const g = u(f, "computed");
    g && h.push(r("computed", g));
    const O = u(f, "inject");
    return O && h.push(r("injected", O)), h.push([
      "div",
      {},
      [
        "span",
        {
          style: o.style + ";opacity:0.66"
        },
        "$ (internal): "
      ],
      ["object", { object: f }]
    ]), h;
  }
  function r(f, h) {
    return h = ne({}, h), Object.keys(h).length ? [
      "div",
      { style: "line-height:1.25em;margin-bottom:0.6em" },
      [
        "div",
        {
          style: "color:#476582"
        },
        f
      ],
      [
        "div",
        {
          style: "padding-left:1.25em"
        },
        ...Object.keys(h).map((g) => [
          "div",
          {},
          ["span", o, g + ": "],
          l(h[g], !1)
        ])
      ]
    ] : ["span", {}];
  }
  function l(f, h = !0) {
    return typeof f == "number" ? ["span", t, f] : typeof f == "string" ? ["span", n, JSON.stringify(f)] : typeof f == "boolean" ? ["span", o, f] : q(f) ? ["object", { object: h ? k(f) : f }] : ["span", n, String(f)];
  }
  function u(f, h) {
    const g = f.type;
    if (M(g))
      return;
    const O = {};
    for (const $ in f.ctx)
      a(g, $, h) && (O[$] = f.ctx[$]);
    return O;
  }
  function a(f, h, g) {
    const O = f[g];
    if (R(O) && O.includes(h) || q(O) && h in O || f.extends && a(f.extends, h, g) || f.mixins && f.mixins.some(($) => a($, h, g)))
      return !0;
  }
  function p(f) {
    return vt(f) ? "ShallowRef" : f.effect ? "ComputedRef" : "Ref";
  }
  window.devtoolsFormatters ? window.devtoolsFormatters.push(i) : window.devtoolsFormatters = [i];
}
const cs = "3.4.19", $t = {}.NODE_ENV !== "production" ? N : se;
/**
* @vue/runtime-dom v3.4.19
* (c) 2018-present Yuxi (Evan) You and Vue contributors
* @license MIT
**/
const ju = "http://www.w3.org/2000/svg", ku = "http://www.w3.org/1998/Math/MathML", ot = typeof document < "u" ? document : null, us = ot && /* @__PURE__ */ ot.createElement("template"), Bu = {
  insert: (e, t, n) => {
    t.insertBefore(e, n || null);
  },
  remove: (e) => {
    const t = e.parentNode;
    t && t.removeChild(e);
  },
  createElement: (e, t, n, o) => {
    const i = t === "svg" ? ot.createElementNS(ju, e) : t === "mathml" ? ot.createElementNS(ku, e) : ot.createElement(e, n ? { is: n } : void 0);
    return e === "select" && o && o.multiple != null && i.setAttribute("multiple", o.multiple), i;
  },
  createText: (e) => ot.createTextNode(e),
  createComment: (e) => ot.createComment(e),
  setText: (e, t) => {
    e.nodeValue = t;
  },
  setElementText: (e, t) => {
    e.textContent = t;
  },
  parentNode: (e) => e.parentNode,
  nextSibling: (e) => e.nextSibling,
  querySelector: (e) => ot.querySelector(e),
  setScopeId(e, t) {
    e.setAttribute(t, "");
  },
  // __UNSAFE__
  // Reason: innerHTML.
  // Static content here can only come from compiled templates.
  // As long as the user only uses trusted templates, this is safe.
  insertStaticContent(e, t, n, o, i, s) {
    const r = n ? n.previousSibling : t.lastChild;
    if (i && (i === s || i.nextSibling))
      for (; t.insertBefore(i.cloneNode(!0), n), !(i === s || !(i = i.nextSibling)); )
        ;
    else {
      us.innerHTML = o === "svg" ? `<svg>${e}</svg>` : o === "mathml" ? `<math>${e}</math>` : e;
      const l = us.content;
      if (o === "svg" || o === "mathml") {
        const u = l.firstChild;
        for (; u.firstChild; )
          l.appendChild(u.firstChild);
        l.removeChild(u);
      }
      t.insertBefore(l, n);
    }
    return [
      // first
      r ? r.nextSibling : t.firstChild,
      // last
      n ? n.previousSibling : t.lastChild
    ];
  }
}, zu = Symbol("_vtc");
function Wu(e, t, n) {
  const o = e[zu];
  o && (t = (t ? [t, ...o] : [...o]).join(" ")), t == null ? e.removeAttribute("class") : n ? e.setAttribute("class", t) : e.className = t;
}
const as = Symbol("_vod"), Uu = Symbol({}.NODE_ENV !== "production" ? "CSS_VAR_TEXT" : ""), Ku = /(^|;)\s*display\s*:/;
function Gu(e, t, n) {
  const o = e.style, i = oe(n), s = o.display;
  let r = !1;
  if (n && !i) {
    if (t && !oe(t))
      for (const l in t)
        n[l] == null && Qo(o, l, "");
    for (const l in n)
      l === "display" && (r = !0), Qo(o, l, n[l]);
  } else if (i) {
    if (t !== n) {
      const l = o[Uu];
      l && (n += ";" + l), o.cssText = n, r = Ku.test(n);
    }
  } else
    t && e.removeAttribute("style");
  as in e && (e[as] = r ? o.display : "", o.display = s);
}
const qu = /[^\\];\s*$/, fs = /\s*!important$/;
function Qo(e, t, n) {
  if (R(n))
    n.forEach((o) => Qo(e, t, o));
  else if (n == null && (n = ""), {}.NODE_ENV !== "production" && qu.test(n) && $t(
    `Unexpected semicolon at the end of '${t}' style value: '${n}'`
  ), t.startsWith("--"))
    e.setProperty(t, n);
  else {
    const o = Yu(e, t);
    fs.test(n) ? e.setProperty(
      Ge(o),
      n.replace(fs, ""),
      "important"
    ) : e[o] = n;
  }
}
const ds = ["Webkit", "Moz", "ms"], Do = {};
function Yu(e, t) {
  const n = Do[t];
  if (n)
    return n;
  let o = Le(t);
  if (o !== "filter" && o in e)
    return Do[t] = o;
  o = Dt(o);
  for (let i = 0; i < ds.length; i++) {
    const s = ds[i] + o;
    if (s in e)
      return Do[t] = s;
  }
  return t;
}
const ps = "http://www.w3.org/1999/xlink";
function Ju(e, t, n, o, i) {
  if (o && t.startsWith("xlink:"))
    n == null ? e.removeAttributeNS(ps, t.slice(6, t.length)) : e.setAttributeNS(ps, t, n);
  else {
    const s = bl(t);
    n == null || s && !Ls(n) ? e.removeAttribute(t) : e.setAttribute(t, s ? "" : n);
  }
}
function Xu(e, t, n, o, i, s, r) {
  if (t === "innerHTML" || t === "textContent") {
    o && r(o, i, s), e[t] = n ?? "";
    return;
  }
  const l = e.tagName;
  if (t === "value" && l !== "PROGRESS" && // custom elements may use _value internally
  !l.includes("-")) {
    e._value = n;
    const a = l === "OPTION" ? e.getAttribute("value") : e.value, p = n ?? "";
    a !== p && (e.value = p), n == null && e.removeAttribute(t);
    return;
  }
  let u = !1;
  if (n === "" || n == null) {
    const a = typeof e[t];
    a === "boolean" ? n = Ls(n) : n == null && a === "string" ? (n = "", u = !0) : a === "number" && (n = 0, u = !0);
  }
  try {
    e[t] = n;
  } catch (a) {
    ({}).NODE_ENV !== "production" && !u && $t(
      `Failed setting prop "${t}" on <${l.toLowerCase()}>: value ${n} is invalid.`,
      a
    );
  }
  u && e.removeAttribute(t);
}
function Zu(e, t, n, o) {
  e.addEventListener(t, n, o);
}
function Qu(e, t, n, o) {
  e.removeEventListener(t, n, o);
}
const hs = Symbol("_vei");
function ea(e, t, n, o, i = null) {
  const s = e[hs] || (e[hs] = {}), r = s[t];
  if (o && r)
    r.value = o;
  else {
    const [l, u] = ta(t);
    if (o) {
      const a = s[t] = ia(o, i);
      Zu(e, l, a, u);
    } else
      r && (Qu(e, l, r, u), s[t] = void 0);
  }
}
const ms = /(?:Once|Passive|Capture)$/;
function ta(e) {
  let t;
  if (ms.test(e)) {
    t = {};
    let o;
    for (; o = e.match(ms); )
      e = e.slice(0, e.length - o[0].length), t[o[0].toLowerCase()] = !0;
  }
  return [e[2] === ":" ? e.slice(3) : Ge(e.slice(2)), t];
}
let To = 0;
const na = /* @__PURE__ */ Promise.resolve(), oa = () => To || (na.then(() => To = 0), To = Date.now());
function ia(e, t) {
  const n = (o) => {
    if (!o._vts)
      o._vts = Date.now();
    else if (o._vts <= n.attached)
      return;
    Te(
      sa(o, n.value),
      t,
      5,
      [o]
    );
  };
  return n.value = e, n.attached = oa(), n;
}
function sa(e, t) {
  if (R(t)) {
    const n = e.stopImmediatePropagation;
    return e.stopImmediatePropagation = () => {
      n.call(e), e._stopped = !0;
    }, t.map((o) => (i) => !i._stopped && o && o(i));
  } else
    return t;
}
const gs = (e) => e.charCodeAt(0) === 111 && e.charCodeAt(1) === 110 && // lowercase letter
e.charCodeAt(2) > 96 && e.charCodeAt(2) < 123, ra = (e, t, n, o, i, s, r, l, u) => {
  const a = i === "svg";
  t === "class" ? Wu(e, o, a) : t === "style" ? Gu(e, n, o) : mn(t) ? jn(t) || ea(e, t, n, o, r) : (t[0] === "." ? (t = t.slice(1), !0) : t[0] === "^" ? (t = t.slice(1), !1) : la(e, t, o, a)) ? Xu(
    e,
    t,
    o,
    s,
    r,
    l,
    u
  ) : (t === "true-value" ? e._trueValue = o : t === "false-value" && (e._falseValue = o), Ju(e, t, o, a));
};
function la(e, t, n, o) {
  if (o)
    return !!(t === "innerHTML" || t === "textContent" || t in e && gs(t) && M(n));
  if (t === "spellcheck" || t === "draggable" || t === "translate" || t === "form" || t === "list" && e.tagName === "INPUT" || t === "type" && e.tagName === "TEXTAREA")
    return !1;
  if (t === "width" || t === "height") {
    const i = e.tagName;
    if (i === "IMG" || i === "VIDEO" || i === "CANVAS" || i === "SOURCE")
      return !1;
  }
  return gs(t) && oe(n) ? !1 : t in e;
}
const ca = {
  esc: "escape",
  space: " ",
  up: "arrow-up",
  left: "arrow-left",
  right: "arrow-right",
  down: "arrow-down",
  delete: "backspace"
}, ua = (e, t) => {
  const n = e._withKeys || (e._withKeys = {}), o = t.join(".");
  return n[o] || (n[o] = (i) => {
    if (!("key" in i))
      return;
    const s = Ge(i.key);
    if (t.some((r) => r === s || ca[r] === s))
      return e(i);
  });
}, aa = /* @__PURE__ */ ne({ patchProp: ra }, Bu);
let _s;
function fa() {
  return _s || (_s = _u(aa));
}
const da = (...e) => {
  const t = fa().createApp(...e);
  ({}).NODE_ENV !== "production" && (ha(t), ma(t));
  const { mount: n } = t;
  return t.mount = (o) => {
    const i = ga(o);
    if (!i)
      return;
    const s = t._component;
    !M(s) && !s.render && !s.template && (s.template = i.innerHTML), i.innerHTML = "";
    const r = n(i, !1, pa(i));
    return i instanceof Element && (i.removeAttribute("v-cloak"), i.setAttribute("data-v-app", "")), r;
  }, t;
};
function pa(e) {
  if (e instanceof SVGElement)
    return "svg";
  if (typeof MathMLElement == "function" && e instanceof MathMLElement)
    return "mathml";
}
function ha(e) {
  Object.defineProperty(e.config, "isNativeTag", {
    value: (t) => _l(t) || yl(t) || wl(t),
    writable: !1
  });
}
function ma(e) {
  {
    const t = e.config.isCustomElement;
    Object.defineProperty(e.config, "isCustomElement", {
      get() {
        return t;
      },
      set() {
        $t(
          "The `isCustomElement` config option is deprecated. Use `compilerOptions.isCustomElement` instead."
        );
      }
    });
    const n = e.config.compilerOptions, o = 'The `compilerOptions` config option is only respected when using a build of Vue.js that includes the runtime compiler (aka "full build"). Since you are using the runtime-only build, `compilerOptions` must be passed to `@vue/compiler-dom` in the build setup instead.\n- For vue-loader: pass it via vue-loader\'s `compilerOptions` loader option.\n- For vue-cli: see https://cli.vuejs.org/guide/webpack.html#modifying-options-of-a-loader\n- For vite: pass it via @vitejs/plugin-vue options. See https://github.com/vitejs/vite-plugin-vue/tree/main/packages/plugin-vue#example-for-passing-options-to-vuecompiler-sfc';
    Object.defineProperty(e.config, "compilerOptions", {
      get() {
        return $t(o), n;
      },
      set() {
        $t(o);
      }
    });
  }
}
function ga(e) {
  if (oe(e)) {
    const t = document.querySelector(e);
    return {}.NODE_ENV !== "production" && !t && $t(
      `Failed to mount app: mount target selector "${e}" returned null.`
    ), t;
  }
  return {}.NODE_ENV !== "production" && window.ShadowRoot && e instanceof window.ShadowRoot && e.mode === "closed" && $t(
    'mounting on a ShadowRoot with `{mode: "closed"}` may lead to unpredictable bugs'
  ), e;
}
/**
* vue v3.4.19
* (c) 2018-present Yuxi (Evan) You and Vue contributors
* @license MIT
**/
function _a() {
  Hu();
}
({}).NODE_ENV !== "production" && _a();
const ya = ["top", "right", "bottom", "left"], ys = ["start", "end"], ws = /* @__PURE__ */ ya.reduce((e, t) => e.concat(t, t + "-" + ys[0], t + "-" + ys[1]), []), St = Math.min, Ne = Math.max, Yn = Math.round, ut = (e) => ({
  x: e,
  y: e
}), wa = {
  left: "right",
  right: "left",
  bottom: "top",
  top: "bottom"
}, Ea = {
  start: "end",
  end: "start"
};
function ei(e, t, n) {
  return Ne(e, St(t, n));
}
function Vt(e, t) {
  return typeof e == "function" ? e(t) : e;
}
function Fe(e) {
  return e.split("-")[0];
}
function Se(e) {
  return e.split("-")[1];
}
function kr(e) {
  return e === "x" ? "y" : "x";
}
function Ci(e) {
  return e === "y" ? "height" : "width";
}
function vn(e) {
  return ["top", "bottom"].includes(Fe(e)) ? "y" : "x";
}
function Di(e) {
  return kr(vn(e));
}
function Br(e, t, n) {
  n === void 0 && (n = !1);
  const o = Se(e), i = Di(e), s = Ci(i);
  let r = i === "x" ? o === (n ? "end" : "start") ? "right" : "left" : o === "start" ? "bottom" : "top";
  return t.reference[s] > t.floating[s] && (r = Xn(r)), [r, Xn(r)];
}
function ba(e) {
  const t = Xn(e);
  return [Jn(e), t, Jn(t)];
}
function Jn(e) {
  return e.replace(/start|end/g, (t) => Ea[t]);
}
function va(e, t, n) {
  const o = ["left", "right"], i = ["right", "left"], s = ["top", "bottom"], r = ["bottom", "top"];
  switch (e) {
    case "top":
    case "bottom":
      return n ? t ? i : o : t ? o : i;
    case "left":
    case "right":
      return t ? s : r;
    default:
      return [];
  }
}
function Na(e, t, n, o) {
  const i = Se(e);
  let s = va(Fe(e), n === "start", o);
  return i && (s = s.map((r) => r + "-" + i), t && (s = s.concat(s.map(Jn)))), s;
}
function Xn(e) {
  return e.replace(/left|right|bottom|top/g, (t) => wa[t]);
}
function Oa(e) {
  return {
    top: 0,
    right: 0,
    bottom: 0,
    left: 0,
    ...e
  };
}
function zr(e) {
  return typeof e != "number" ? Oa(e) : {
    top: e,
    right: e,
    bottom: e,
    left: e
  };
}
function Zn(e) {
  return {
    ...e,
    top: e.y,
    left: e.x,
    right: e.x + e.width,
    bottom: e.y + e.height
  };
}
function Es(e, t, n) {
  let {
    reference: o,
    floating: i
  } = e;
  const s = vn(t), r = Di(t), l = Ci(r), u = Fe(t), a = s === "y", p = o.x + o.width / 2 - i.width / 2, f = o.y + o.height / 2 - i.height / 2, h = o[l] / 2 - i[l] / 2;
  let g;
  switch (u) {
    case "top":
      g = {
        x: p,
        y: o.y - i.height
      };
      break;
    case "bottom":
      g = {
        x: p,
        y: o.y + o.height
      };
      break;
    case "right":
      g = {
        x: o.x + o.width,
        y: f
      };
      break;
    case "left":
      g = {
        x: o.x - i.width,
        y: f
      };
      break;
    default:
      g = {
        x: o.x,
        y: o.y
      };
  }
  switch (Se(t)) {
    case "start":
      g[r] -= h * (n && a ? -1 : 1);
      break;
    case "end":
      g[r] += h * (n && a ? -1 : 1);
      break;
  }
  return g;
}
const xa = async (e, t, n) => {
  const {
    placement: o = "bottom",
    strategy: i = "absolute",
    middleware: s = [],
    platform: r
  } = n, l = s.filter(Boolean), u = await (r.isRTL == null ? void 0 : r.isRTL(t));
  let a = await r.getElementRects({
    reference: e,
    floating: t,
    strategy: i
  }), {
    x: p,
    y: f
  } = Es(a, o, u), h = o, g = {}, O = 0;
  for (let $ = 0; $ < l.length; $++) {
    const {
      name: V,
      fn: P
    } = l[$], {
      x: H,
      y: A,
      data: j,
      reset: I
    } = await P({
      x: p,
      y: f,
      initialPlacement: o,
      placement: h,
      strategy: i,
      middlewareData: g,
      rects: a,
      platform: r,
      elements: {
        reference: e,
        floating: t
      }
    });
    p = H ?? p, f = A ?? f, g = {
      ...g,
      [V]: {
        ...g[V],
        ...j
      }
    }, I && O <= 50 && (O++, typeof I == "object" && (I.placement && (h = I.placement), I.rects && (a = I.rects === !0 ? await r.getElementRects({
      reference: e,
      floating: t,
      strategy: i
    }) : I.rects), {
      x: p,
      y: f
    } = Es(a, h, u)), $ = -1);
  }
  return {
    x: p,
    y: f,
    placement: h,
    strategy: i,
    middlewareData: g
  };
};
async function po(e, t) {
  var n;
  t === void 0 && (t = {});
  const {
    x: o,
    y: i,
    platform: s,
    rects: r,
    elements: l,
    strategy: u
  } = e, {
    boundary: a = "clippingAncestors",
    rootBoundary: p = "viewport",
    elementContext: f = "floating",
    altBoundary: h = !1,
    padding: g = 0
  } = Vt(t, e), O = zr(g), V = l[h ? f === "floating" ? "reference" : "floating" : f], P = Zn(await s.getClippingRect({
    element: (n = await (s.isElement == null ? void 0 : s.isElement(V))) == null || n ? V : V.contextElement || await (s.getDocumentElement == null ? void 0 : s.getDocumentElement(l.floating)),
    boundary: a,
    rootBoundary: p,
    strategy: u
  })), H = f === "floating" ? {
    ...r.floating,
    x: o,
    y: i
  } : r.reference, A = await (s.getOffsetParent == null ? void 0 : s.getOffsetParent(l.floating)), j = await (s.isElement == null ? void 0 : s.isElement(A)) ? await (s.getScale == null ? void 0 : s.getScale(A)) || {
    x: 1,
    y: 1
  } : {
    x: 1,
    y: 1
  }, I = Zn(s.convertOffsetParentRelativeRectToViewportRelativeRect ? await s.convertOffsetParentRelativeRectToViewportRelativeRect({
    elements: l,
    rect: H,
    offsetParent: A,
    strategy: u
  }) : H);
  return {
    top: (P.top - I.top + O.top) / j.y,
    bottom: (I.bottom - P.bottom + O.bottom) / j.y,
    left: (P.left - I.left + O.left) / j.x,
    right: (I.right - P.right + O.right) / j.x
  };
}
const $a = (e) => ({
  name: "arrow",
  options: e,
  async fn(t) {
    const {
      x: n,
      y: o,
      placement: i,
      rects: s,
      platform: r,
      elements: l,
      middlewareData: u
    } = t, {
      element: a,
      padding: p = 0
    } = Vt(e, t) || {};
    if (a == null)
      return {};
    const f = zr(p), h = {
      x: n,
      y: o
    }, g = Di(i), O = Ci(g), $ = await r.getDimensions(a), V = g === "y", P = V ? "top" : "left", H = V ? "bottom" : "right", A = V ? "clientHeight" : "clientWidth", j = s.reference[O] + s.reference[g] - h[g] - s.floating[O], I = h[g] - s.reference[g], K = await (r.getOffsetParent == null ? void 0 : r.getOffsetParent(a));
    let D = K ? K[A] : 0;
    (!D || !await (r.isElement == null ? void 0 : r.isElement(K))) && (D = l.floating[A] || s.floating[O]);
    const Y = j / 2 - I / 2, J = D / 2 - $[O] / 2 - 1, Z = St(f[P], J), ie = St(f[H], J), Q = Z, ve = D - $[O] - ie, de = D / 2 - $[O] / 2 + Y, re = ei(Q, de, ve), ee = !u.arrow && Se(i) != null && de !== re && s.reference[O] / 2 - (de < Q ? Z : ie) - $[O] / 2 < 0, L = ee ? de < Q ? de - Q : de - ve : 0;
    return {
      [g]: h[g] + L,
      data: {
        [g]: re,
        centerOffset: de - re - L,
        ...ee && {
          alignmentOffset: L
        }
      },
      reset: ee
    };
  }
});
function Ca(e, t, n) {
  return (e ? [...n.filter((i) => Se(i) === e), ...n.filter((i) => Se(i) !== e)] : n.filter((i) => Fe(i) === i)).filter((i) => e ? Se(i) === e || (t ? Jn(i) !== i : !1) : !0);
}
const Da = function(e) {
  return e === void 0 && (e = {}), {
    name: "autoPlacement",
    options: e,
    async fn(t) {
      var n, o, i;
      const {
        rects: s,
        middlewareData: r,
        placement: l,
        platform: u,
        elements: a
      } = t, {
        crossAxis: p = !1,
        alignment: f,
        allowedPlacements: h = ws,
        autoAlignment: g = !0,
        ...O
      } = Vt(e, t), $ = f !== void 0 || h === ws ? Ca(f || null, g, h) : h, V = await po(t, O), P = ((n = r.autoPlacement) == null ? void 0 : n.index) || 0, H = $[P];
      if (H == null)
        return {};
      const A = Br(H, s, await (u.isRTL == null ? void 0 : u.isRTL(a.floating)));
      if (l !== H)
        return {
          reset: {
            placement: $[0]
          }
        };
      const j = [V[Fe(H)], V[A[0]], V[A[1]]], I = [...((o = r.autoPlacement) == null ? void 0 : o.overflows) || [], {
        placement: H,
        overflows: j
      }], K = $[P + 1];
      if (K)
        return {
          data: {
            index: P + 1,
            overflows: I
          },
          reset: {
            placement: K
          }
        };
      const D = I.map((Z) => {
        const ie = Se(Z.placement);
        return [Z.placement, ie && p ? (
          // Check along the mainAxis and main crossAxis side.
          Z.overflows.slice(0, 2).reduce((Q, ve) => Q + ve, 0)
        ) : (
          // Check only the mainAxis.
          Z.overflows[0]
        ), Z.overflows];
      }).sort((Z, ie) => Z[1] - ie[1]), J = ((i = D.filter((Z) => Z[2].slice(
        0,
        // Aligned placements should not check their opposite crossAxis
        // side.
        Se(Z[0]) ? 2 : 3
      ).every((ie) => ie <= 0))[0]) == null ? void 0 : i[0]) || D[0][0];
      return J !== l ? {
        data: {
          index: P + 1,
          overflows: I
        },
        reset: {
          placement: J
        }
      } : {};
    }
  };
}, Ta = function(e) {
  return e === void 0 && (e = {}), {
    name: "flip",
    options: e,
    async fn(t) {
      var n, o;
      const {
        placement: i,
        middlewareData: s,
        rects: r,
        initialPlacement: l,
        platform: u,
        elements: a
      } = t, {
        mainAxis: p = !0,
        crossAxis: f = !0,
        fallbackPlacements: h,
        fallbackStrategy: g = "bestFit",
        fallbackAxisSideDirection: O = "none",
        flipAlignment: $ = !0,
        ...V
      } = Vt(e, t);
      if ((n = s.arrow) != null && n.alignmentOffset)
        return {};
      const P = Fe(i), H = Fe(l) === l, A = await (u.isRTL == null ? void 0 : u.isRTL(a.floating)), j = h || (H || !$ ? [Xn(l)] : ba(l));
      !h && O !== "none" && j.push(...Na(l, $, O, A));
      const I = [l, ...j], K = await po(t, V), D = [];
      let Y = ((o = s.flip) == null ? void 0 : o.overflows) || [];
      if (p && D.push(K[P]), f) {
        const Q = Br(i, r, A);
        D.push(K[Q[0]], K[Q[1]]);
      }
      if (Y = [...Y, {
        placement: i,
        overflows: D
      }], !D.every((Q) => Q <= 0)) {
        var J, Z;
        const Q = (((J = s.flip) == null ? void 0 : J.index) || 0) + 1, ve = I[Q];
        if (ve)
          return {
            data: {
              index: Q,
              overflows: Y
            },
            reset: {
              placement: ve
            }
          };
        let de = (Z = Y.filter((re) => re.overflows[0] <= 0).sort((re, ee) => re.overflows[1] - ee.overflows[1])[0]) == null ? void 0 : Z.placement;
        if (!de)
          switch (g) {
            case "bestFit": {
              var ie;
              const re = (ie = Y.map((ee) => [ee.placement, ee.overflows.filter((L) => L > 0).reduce((L, B) => L + B, 0)]).sort((ee, L) => ee[1] - L[1])[0]) == null ? void 0 : ie[0];
              re && (de = re);
              break;
            }
            case "initialPlacement":
              de = l;
              break;
          }
        if (i !== de)
          return {
            reset: {
              placement: de
            }
          };
      }
      return {};
    }
  };
};
async function Sa(e, t) {
  const {
    placement: n,
    platform: o,
    elements: i
  } = e, s = await (o.isRTL == null ? void 0 : o.isRTL(i.floating)), r = Fe(n), l = Se(n), u = vn(n) === "y", a = ["left", "top"].includes(r) ? -1 : 1, p = s && u ? -1 : 1, f = Vt(t, e);
  let {
    mainAxis: h,
    crossAxis: g,
    alignmentAxis: O
  } = typeof f == "number" ? {
    mainAxis: f,
    crossAxis: 0,
    alignmentAxis: null
  } : {
    mainAxis: 0,
    crossAxis: 0,
    alignmentAxis: null,
    ...f
  };
  return l && typeof O == "number" && (g = l === "end" ? O * -1 : O), u ? {
    x: g * p,
    y: h * a
  } : {
    x: h * a,
    y: g * p
  };
}
const Va = function(e) {
  return e === void 0 && (e = 0), {
    name: "offset",
    options: e,
    async fn(t) {
      var n, o;
      const {
        x: i,
        y: s,
        placement: r,
        middlewareData: l
      } = t, u = await Sa(t, e);
      return r === ((n = l.offset) == null ? void 0 : n.placement) && (o = l.arrow) != null && o.alignmentOffset ? {} : {
        x: i + u.x,
        y: s + u.y,
        data: {
          ...u,
          placement: r
        }
      };
    }
  };
}, Pa = function(e) {
  return e === void 0 && (e = {}), {
    name: "shift",
    options: e,
    async fn(t) {
      const {
        x: n,
        y: o,
        placement: i
      } = t, {
        mainAxis: s = !0,
        crossAxis: r = !1,
        limiter: l = {
          fn: (V) => {
            let {
              x: P,
              y: H
            } = V;
            return {
              x: P,
              y: H
            };
          }
        },
        ...u
      } = Vt(e, t), a = {
        x: n,
        y: o
      }, p = await po(t, u), f = vn(Fe(i)), h = kr(f);
      let g = a[h], O = a[f];
      if (s) {
        const V = h === "y" ? "top" : "left", P = h === "y" ? "bottom" : "right", H = g + p[V], A = g - p[P];
        g = ei(H, g, A);
      }
      if (r) {
        const V = f === "y" ? "top" : "left", P = f === "y" ? "bottom" : "right", H = O + p[V], A = O - p[P];
        O = ei(H, O, A);
      }
      const $ = l.fn({
        ...t,
        [h]: g,
        [f]: O
      });
      return {
        ...$,
        data: {
          x: $.x - n,
          y: $.y - o
        }
      };
    }
  };
}, Aa = function(e) {
  return e === void 0 && (e = {}), {
    name: "size",
    options: e,
    async fn(t) {
      const {
        placement: n,
        rects: o,
        platform: i,
        elements: s
      } = t, {
        apply: r = () => {
        },
        ...l
      } = Vt(e, t), u = await po(t, l), a = Fe(n), p = Se(n), f = vn(n) === "y", {
        width: h,
        height: g
      } = o.floating;
      let O, $;
      a === "top" || a === "bottom" ? (O = a, $ = p === (await (i.isRTL == null ? void 0 : i.isRTL(s.floating)) ? "start" : "end") ? "left" : "right") : ($ = a, O = p === "end" ? "top" : "bottom");
      const V = g - u[O], P = h - u[$], H = !t.middlewareData.shift;
      let A = V, j = P;
      if (f) {
        const K = h - u.left - u.right;
        j = p || H ? St(P, K) : K;
      } else {
        const K = g - u.top - u.bottom;
        A = p || H ? St(V, K) : K;
      }
      if (H && !p) {
        const K = Ne(u.left, 0), D = Ne(u.right, 0), Y = Ne(u.top, 0), J = Ne(u.bottom, 0);
        f ? j = h - 2 * (K !== 0 || D !== 0 ? K + D : Ne(u.left, u.right)) : A = g - 2 * (Y !== 0 || J !== 0 ? Y + J : Ne(u.top, u.bottom));
      }
      await r({
        ...t,
        availableWidth: j,
        availableHeight: A
      });
      const I = await i.getDimensions(s.floating);
      return h !== I.width || g !== I.height ? {
        reset: {
          rects: !0
        }
      } : {};
    }
  };
};
function at(e) {
  return Wr(e) ? (e.nodeName || "").toLowerCase() : "#document";
}
function be(e) {
  var t;
  return (e == null || (t = e.ownerDocument) == null ? void 0 : t.defaultView) || window;
}
function pt(e) {
  var t;
  return (t = (Wr(e) ? e.ownerDocument : e.document) || window.document) == null ? void 0 : t.documentElement;
}
function Wr(e) {
  return e instanceof Node || e instanceof be(e).Node;
}
function qe(e) {
  return e instanceof Element || e instanceof be(e).Element;
}
function He(e) {
  return e instanceof HTMLElement || e instanceof be(e).HTMLElement;
}
function bs(e) {
  return typeof ShadowRoot > "u" ? !1 : e instanceof ShadowRoot || e instanceof be(e).ShadowRoot;
}
function Nn(e) {
  const {
    overflow: t,
    overflowX: n,
    overflowY: o,
    display: i
  } = Oe(e);
  return /auto|scroll|overlay|hidden|clip/.test(t + o + n) && !["inline", "contents"].includes(i);
}
function Ra(e) {
  return ["table", "td", "th"].includes(at(e));
}
function Ti(e) {
  const t = Si(), n = Oe(e);
  return n.transform !== "none" || n.perspective !== "none" || (n.containerType ? n.containerType !== "normal" : !1) || !t && (n.backdropFilter ? n.backdropFilter !== "none" : !1) || !t && (n.filter ? n.filter !== "none" : !1) || ["transform", "perspective", "filter"].some((o) => (n.willChange || "").includes(o)) || ["paint", "layout", "strict", "content"].some((o) => (n.contain || "").includes(o));
}
function Ma(e) {
  let t = zt(e);
  for (; He(t) && !ho(t); ) {
    if (Ti(t))
      return t;
    t = zt(t);
  }
  return null;
}
function Si() {
  return typeof CSS > "u" || !CSS.supports ? !1 : CSS.supports("-webkit-backdrop-filter", "none");
}
function ho(e) {
  return ["html", "body", "#document"].includes(at(e));
}
function Oe(e) {
  return be(e).getComputedStyle(e);
}
function mo(e) {
  return qe(e) ? {
    scrollLeft: e.scrollLeft,
    scrollTop: e.scrollTop
  } : {
    scrollLeft: e.pageXOffset,
    scrollTop: e.pageYOffset
  };
}
function zt(e) {
  if (at(e) === "html")
    return e;
  const t = (
    // Step into the shadow DOM of the parent of a slotted node.
    e.assignedSlot || // DOM Element detected.
    e.parentNode || // ShadowRoot detected.
    bs(e) && e.host || // Fallback.
    pt(e)
  );
  return bs(t) ? t.host : t;
}
function Ur(e) {
  const t = zt(e);
  return ho(t) ? e.ownerDocument ? e.ownerDocument.body : e.body : He(t) && Nn(t) ? t : Ur(t);
}
function fn(e, t, n) {
  var o;
  t === void 0 && (t = []), n === void 0 && (n = !0);
  const i = Ur(e), s = i === ((o = e.ownerDocument) == null ? void 0 : o.body), r = be(i);
  return s ? t.concat(r, r.visualViewport || [], Nn(i) ? i : [], r.frameElement && n ? fn(r.frameElement) : []) : t.concat(i, fn(i, [], n));
}
function Kr(e) {
  const t = Oe(e);
  let n = parseFloat(t.width) || 0, o = parseFloat(t.height) || 0;
  const i = He(e), s = i ? e.offsetWidth : n, r = i ? e.offsetHeight : o, l = Yn(n) !== s || Yn(o) !== r;
  return l && (n = s, o = r), {
    width: n,
    height: o,
    $: l
  };
}
function Gr(e) {
  return qe(e) ? e : e.contextElement;
}
function kt(e) {
  const t = Gr(e);
  if (!He(t))
    return ut(1);
  const n = t.getBoundingClientRect(), {
    width: o,
    height: i,
    $: s
  } = Kr(t);
  let r = (s ? Yn(n.width) : n.width) / o, l = (s ? Yn(n.height) : n.height) / i;
  return (!r || !Number.isFinite(r)) && (r = 1), (!l || !Number.isFinite(l)) && (l = 1), {
    x: r,
    y: l
  };
}
const Ia = /* @__PURE__ */ ut(0);
function qr(e) {
  const t = be(e);
  return !Si() || !t.visualViewport ? Ia : {
    x: t.visualViewport.offsetLeft,
    y: t.visualViewport.offsetTop
  };
}
function La(e, t, n) {
  return t === void 0 && (t = !1), !n || t && n !== be(e) ? !1 : t;
}
function dn(e, t, n, o) {
  t === void 0 && (t = !1), n === void 0 && (n = !1);
  const i = e.getBoundingClientRect(), s = Gr(e);
  let r = ut(1);
  t && (o ? qe(o) && (r = kt(o)) : r = kt(e));
  const l = La(s, n, o) ? qr(s) : ut(0);
  let u = (i.left + l.x) / r.x, a = (i.top + l.y) / r.y, p = i.width / r.x, f = i.height / r.y;
  if (s) {
    const h = be(s), g = o && qe(o) ? be(o) : o;
    let O = h, $ = O.frameElement;
    for (; $ && o && g !== O; ) {
      const V = kt($), P = $.getBoundingClientRect(), H = Oe($), A = P.left + ($.clientLeft + parseFloat(H.paddingLeft)) * V.x, j = P.top + ($.clientTop + parseFloat(H.paddingTop)) * V.y;
      u *= V.x, a *= V.y, p *= V.x, f *= V.y, u += A, a += j, O = be($), $ = O.frameElement;
    }
  }
  return Zn({
    width: p,
    height: f,
    x: u,
    y: a
  });
}
const Fa = [":popover-open", ":modal"];
function Yr(e) {
  return Fa.some((t) => {
    try {
      return e.matches(t);
    } catch {
      return !1;
    }
  });
}
function Ha(e) {
  let {
    elements: t,
    rect: n,
    offsetParent: o,
    strategy: i
  } = e;
  const s = i === "fixed", r = pt(o), l = t ? Yr(t.floating) : !1;
  if (o === r || l && s)
    return n;
  let u = {
    scrollLeft: 0,
    scrollTop: 0
  }, a = ut(1);
  const p = ut(0), f = He(o);
  if ((f || !f && !s) && ((at(o) !== "body" || Nn(r)) && (u = mo(o)), He(o))) {
    const h = dn(o);
    a = kt(o), p.x = h.x + o.clientLeft, p.y = h.y + o.clientTop;
  }
  return {
    width: n.width * a.x,
    height: n.height * a.y,
    x: n.x * a.x - u.scrollLeft * a.x + p.x,
    y: n.y * a.y - u.scrollTop * a.y + p.y
  };
}
function ja(e) {
  return Array.from(e.getClientRects());
}
function Jr(e) {
  return dn(pt(e)).left + mo(e).scrollLeft;
}
function ka(e) {
  const t = pt(e), n = mo(e), o = e.ownerDocument.body, i = Ne(t.scrollWidth, t.clientWidth, o.scrollWidth, o.clientWidth), s = Ne(t.scrollHeight, t.clientHeight, o.scrollHeight, o.clientHeight);
  let r = -n.scrollLeft + Jr(e);
  const l = -n.scrollTop;
  return Oe(o).direction === "rtl" && (r += Ne(t.clientWidth, o.clientWidth) - i), {
    width: i,
    height: s,
    x: r,
    y: l
  };
}
function Ba(e, t) {
  const n = be(e), o = pt(e), i = n.visualViewport;
  let s = o.clientWidth, r = o.clientHeight, l = 0, u = 0;
  if (i) {
    s = i.width, r = i.height;
    const a = Si();
    (!a || a && t === "fixed") && (l = i.offsetLeft, u = i.offsetTop);
  }
  return {
    width: s,
    height: r,
    x: l,
    y: u
  };
}
function za(e, t) {
  const n = dn(e, !0, t === "fixed"), o = n.top + e.clientTop, i = n.left + e.clientLeft, s = He(e) ? kt(e) : ut(1), r = e.clientWidth * s.x, l = e.clientHeight * s.y, u = i * s.x, a = o * s.y;
  return {
    width: r,
    height: l,
    x: u,
    y: a
  };
}
function vs(e, t, n) {
  let o;
  if (t === "viewport")
    o = Ba(e, n);
  else if (t === "document")
    o = ka(pt(e));
  else if (qe(t))
    o = za(t, n);
  else {
    const i = qr(e);
    o = {
      ...t,
      x: t.x - i.x,
      y: t.y - i.y
    };
  }
  return Zn(o);
}
function Xr(e, t) {
  const n = zt(e);
  return n === t || !qe(n) || ho(n) ? !1 : Oe(n).position === "fixed" || Xr(n, t);
}
function Wa(e, t) {
  const n = t.get(e);
  if (n)
    return n;
  let o = fn(e, [], !1).filter((l) => qe(l) && at(l) !== "body"), i = null;
  const s = Oe(e).position === "fixed";
  let r = s ? zt(e) : e;
  for (; qe(r) && !ho(r); ) {
    const l = Oe(r), u = Ti(r);
    !u && l.position === "fixed" && (i = null), (s ? !u && !i : !u && l.position === "static" && !!i && ["absolute", "fixed"].includes(i.position) || Nn(r) && !u && Xr(e, r)) ? o = o.filter((p) => p !== r) : i = l, r = zt(r);
  }
  return t.set(e, o), o;
}
function Ua(e) {
  let {
    element: t,
    boundary: n,
    rootBoundary: o,
    strategy: i
  } = e;
  const r = [...n === "clippingAncestors" ? Wa(t, this._c) : [].concat(n), o], l = r[0], u = r.reduce((a, p) => {
    const f = vs(t, p, i);
    return a.top = Ne(f.top, a.top), a.right = St(f.right, a.right), a.bottom = St(f.bottom, a.bottom), a.left = Ne(f.left, a.left), a;
  }, vs(t, l, i));
  return {
    width: u.right - u.left,
    height: u.bottom - u.top,
    x: u.left,
    y: u.top
  };
}
function Ka(e) {
  const {
    width: t,
    height: n
  } = Kr(e);
  return {
    width: t,
    height: n
  };
}
function Ga(e, t, n) {
  const o = He(t), i = pt(t), s = n === "fixed", r = dn(e, !0, s, t);
  let l = {
    scrollLeft: 0,
    scrollTop: 0
  };
  const u = ut(0);
  if (o || !o && !s)
    if ((at(t) !== "body" || Nn(i)) && (l = mo(t)), o) {
      const f = dn(t, !0, s, t);
      u.x = f.x + t.clientLeft, u.y = f.y + t.clientTop;
    } else
      i && (u.x = Jr(i));
  const a = r.left + l.scrollLeft - u.x, p = r.top + l.scrollTop - u.y;
  return {
    x: a,
    y: p,
    width: r.width,
    height: r.height
  };
}
function Ns(e, t) {
  return !He(e) || Oe(e).position === "fixed" ? null : t ? t(e) : e.offsetParent;
}
function Zr(e, t) {
  const n = be(e);
  if (!He(e) || Yr(e))
    return n;
  let o = Ns(e, t);
  for (; o && Ra(o) && Oe(o).position === "static"; )
    o = Ns(o, t);
  return o && (at(o) === "html" || at(o) === "body" && Oe(o).position === "static" && !Ti(o)) ? n : o || Ma(e) || n;
}
const qa = async function(e) {
  const t = this.getOffsetParent || Zr, n = this.getDimensions;
  return {
    reference: Ga(e.reference, await t(e.floating), e.strategy),
    floating: {
      x: 0,
      y: 0,
      ...await n(e.floating)
    }
  };
};
function Ya(e) {
  return Oe(e).direction === "rtl";
}
const Ja = {
  convertOffsetParentRelativeRectToViewportRelativeRect: Ha,
  getDocumentElement: pt,
  getClippingRect: Ua,
  getOffsetParent: Zr,
  getElementRects: qa,
  getClientRects: ja,
  getDimensions: Ka,
  getScale: kt,
  isElement: qe,
  isRTL: Ya
}, Xa = Da, Za = Pa, Qa = Ta, ef = Aa, tf = $a, nf = (e, t, n) => {
  const o = /* @__PURE__ */ new Map(), i = {
    platform: Ja,
    ...n
  }, s = {
    ...i.platform,
    _c: o
  };
  return xa(e, t, {
    ...i,
    platform: s
  });
};
let pn = !1;
if (typeof window < "u") {
  pn = !1;
  try {
    const e = Object.defineProperty({}, "passive", {
      get() {
        pn = !0;
      }
    });
    window.addEventListener("test", null, e);
  } catch {
  }
}
let Qr = !1;
typeof window < "u" && typeof navigator < "u" && (Qr = /iPad|iPhone|iPod/.test(navigator.userAgent) && !window.MSStream);
const el = ["auto", "top", "bottom", "left", "right"].reduce((e, t) => e.concat([
  t,
  `${t}-start`,
  `${t}-end`
]), []), Os = {
  hover: "mouseenter",
  focus: "focus",
  click: "click",
  touch: "touchstart",
  pointer: "pointerdown"
}, xs = {
  hover: "mouseleave",
  focus: "blur",
  click: "click",
  touch: "touchend",
  pointer: "pointerup"
};
function $s(e, t) {
  const n = e.indexOf(t);
  n !== -1 && e.splice(n, 1);
}
function So() {
  return new Promise((e) => requestAnimationFrame(() => {
    requestAnimationFrame(e);
  }));
}
const Ct = {
  // Disable popper components
  disabled: !1,
  // Default position offset along main axis (px)
  distance: 5,
  // Default position offset along cross axis (px)
  skidding: 0,
  // Default container where the tooltip will be appended
  container: "body",
  // Element used to compute position and size boundaries
  boundary: void 0,
  // Skip delay & CSS transitions when another popper is shown, so that the popper appear to instanly move to the new position.
  instantMove: !1,
  // Auto destroy tooltip DOM nodes (ms)
  disposeTimeout: 150,
  // Triggers on the popper itself
  popperTriggers: [],
  // Positioning strategy
  strategy: "absolute",
  // Prevent overflow
  preventOverflow: !0,
  // Flip to the opposite placement if needed
  flip: !0,
  // Shift on the cross axis to prevent the popper from overflowing
  shift: !0,
  // Overflow padding (px)
  overflowPadding: 0,
  // Arrow padding (px)
  arrowPadding: 0,
  // Compute arrow overflow (useful to hide it)
  arrowOverflow: !0,
  /**
   * By default, compute autohide on 'click'.
   */
  autoHideOnMousedown: !1,
  // Themes
  themes: {
    tooltip: {
      // Default tooltip placement relative to target element
      placement: "top",
      // Default events that trigger the tooltip
      triggers: ["hover", "focus", "touch"],
      // Close tooltip on click on tooltip target
      hideTriggers: (e) => [...e, "click"],
      // Delay (ms)
      delay: {
        show: 200,
        hide: 0
      },
      // Update popper on content resize
      handleResize: !1,
      // Enable HTML content in directive
      html: !1,
      // Displayed when tooltip content is loading
      loadingContent: "..."
    },
    dropdown: {
      // Default dropdown placement relative to target element
      placement: "bottom",
      // Default events that trigger the dropdown
      triggers: ["click"],
      // Delay (ms)
      delay: 0,
      // Update popper on content resize
      handleResize: !0,
      // Hide on clock outside
      autoHide: !0
    },
    menu: {
      $extend: "dropdown",
      triggers: ["hover", "focus"],
      popperTriggers: ["hover"],
      delay: {
        show: 0,
        hide: 400
      }
    }
  }
};
function Qn(e, t) {
  let n = Ct.themes[e] || {}, o;
  do
    o = n[t], typeof o > "u" ? n.$extend ? n = Ct.themes[n.$extend] || {} : (n = null, o = Ct[t]) : n = null;
  while (n);
  return o;
}
function of(e) {
  const t = [e];
  let n = Ct.themes[e] || {};
  do
    n.$extend && !n.$resetCss ? (t.push(n.$extend), n = Ct.themes[n.$extend] || {}) : n = null;
  while (n);
  return t.map((o) => `v-popper--theme-${o}`);
}
function Cs(e) {
  const t = [e];
  let n = Ct.themes[e] || {};
  do
    n.$extend ? (t.push(n.$extend), n = Ct.themes[n.$extend] || {}) : n = null;
  while (n);
  return t;
}
const Ee = [];
let gt = null;
const Ds = {};
function Ts(e) {
  let t = Ds[e];
  return t || (t = Ds[e] = []), t;
}
let ti = function() {
};
typeof window < "u" && (ti = window.Element);
function U(e) {
  return function(t) {
    return Qn(t.theme, e);
  };
}
const Vo = "__floating-ui__popper", sf = () => /* @__PURE__ */ wi({
  name: "VPopper",
  provide() {
    return {
      [Vo]: {
        parentPopper: this
      }
    };
  },
  inject: {
    [Vo]: { default: null }
  },
  props: {
    theme: {
      type: String,
      required: !0
    },
    targetNodes: {
      type: Function,
      required: !0
    },
    referenceNode: {
      type: Function,
      default: null
    },
    popperNode: {
      type: Function,
      required: !0
    },
    shown: {
      type: Boolean,
      default: !1
    },
    showGroup: {
      type: String,
      default: null
    },
    // eslint-disable-next-line vue/require-prop-types
    ariaId: {
      default: null
    },
    disabled: {
      type: Boolean,
      default: U("disabled")
    },
    positioningDisabled: {
      type: Boolean,
      default: U("positioningDisabled")
    },
    placement: {
      type: String,
      default: U("placement"),
      validator: (e) => el.includes(e)
    },
    delay: {
      type: [String, Number, Object],
      default: U("delay")
    },
    distance: {
      type: [Number, String],
      default: U("distance")
    },
    skidding: {
      type: [Number, String],
      default: U("skidding")
    },
    triggers: {
      type: Array,
      default: U("triggers")
    },
    showTriggers: {
      type: [Array, Function],
      default: U("showTriggers")
    },
    hideTriggers: {
      type: [Array, Function],
      default: U("hideTriggers")
    },
    popperTriggers: {
      type: Array,
      default: U("popperTriggers")
    },
    popperShowTriggers: {
      type: [Array, Function],
      default: U("popperShowTriggers")
    },
    popperHideTriggers: {
      type: [Array, Function],
      default: U("popperHideTriggers")
    },
    container: {
      type: [String, Object, ti, Boolean],
      default: U("container")
    },
    boundary: {
      type: [String, ti],
      default: U("boundary")
    },
    strategy: {
      type: String,
      validator: (e) => ["absolute", "fixed"].includes(e),
      default: U("strategy")
    },
    autoHide: {
      type: [Boolean, Function],
      default: U("autoHide")
    },
    handleResize: {
      type: Boolean,
      default: U("handleResize")
    },
    instantMove: {
      type: Boolean,
      default: U("instantMove")
    },
    eagerMount: {
      type: Boolean,
      default: U("eagerMount")
    },
    popperClass: {
      type: [String, Array, Object],
      default: U("popperClass")
    },
    computeTransformOrigin: {
      type: Boolean,
      default: U("computeTransformOrigin")
    },
    /**
     * @deprecated
     */
    autoMinSize: {
      type: Boolean,
      default: U("autoMinSize")
    },
    autoSize: {
      type: [Boolean, String],
      default: U("autoSize")
    },
    /**
     * @deprecated
     */
    autoMaxSize: {
      type: Boolean,
      default: U("autoMaxSize")
    },
    autoBoundaryMaxSize: {
      type: Boolean,
      default: U("autoBoundaryMaxSize")
    },
    preventOverflow: {
      type: Boolean,
      default: U("preventOverflow")
    },
    overflowPadding: {
      type: [Number, String],
      default: U("overflowPadding")
    },
    arrowPadding: {
      type: [Number, String],
      default: U("arrowPadding")
    },
    arrowOverflow: {
      type: Boolean,
      default: U("arrowOverflow")
    },
    flip: {
      type: Boolean,
      default: U("flip")
    },
    shift: {
      type: Boolean,
      default: U("shift")
    },
    shiftCrossAxis: {
      type: Boolean,
      default: U("shiftCrossAxis")
    },
    noAutoFocus: {
      type: Boolean,
      default: U("noAutoFocus")
    },
    disposeTimeout: {
      type: Number,
      default: U("disposeTimeout")
    }
  },
  emits: {
    show: () => !0,
    hide: () => !0,
    "update:shown": (e) => !0,
    "apply-show": () => !0,
    "apply-hide": () => !0,
    "close-group": () => !0,
    "close-directive": () => !0,
    "auto-hide": () => !0,
    resize: () => !0
  },
  data() {
    return {
      isShown: !1,
      isMounted: !1,
      skipTransition: !1,
      classes: {
        showFrom: !1,
        showTo: !1,
        hideFrom: !1,
        hideTo: !0
      },
      result: {
        x: 0,
        y: 0,
        placement: "",
        strategy: this.strategy,
        arrow: {
          x: 0,
          y: 0,
          centerOffset: 0
        },
        transformOrigin: null
      },
      randomId: `popper_${[Math.random(), Date.now()].map((e) => e.toString(36).substring(2, 10)).join("_")}`,
      shownChildren: /* @__PURE__ */ new Set(),
      lastAutoHide: !0,
      pendingHide: !1,
      containsGlobalTarget: !1,
      isDisposed: !0,
      mouseDownContains: !1
    };
  },
  computed: {
    popperId() {
      return this.ariaId != null ? this.ariaId : this.randomId;
    },
    shouldMountContent() {
      return this.eagerMount || this.isMounted;
    },
    slotData() {
      return {
        popperId: this.popperId,
        isShown: this.isShown,
        shouldMountContent: this.shouldMountContent,
        skipTransition: this.skipTransition,
        autoHide: typeof this.autoHide == "function" ? this.lastAutoHide : this.autoHide,
        show: this.show,
        hide: this.hide,
        handleResize: this.handleResize,
        onResize: this.onResize,
        classes: {
          ...this.classes,
          popperClass: this.popperClass
        },
        result: this.positioningDisabled ? null : this.result,
        attrs: this.$attrs
      };
    },
    parentPopper() {
      var e;
      return (e = this[Vo]) == null ? void 0 : e.parentPopper;
    },
    hasPopperShowTriggerHover() {
      var e, t;
      return ((e = this.popperTriggers) == null ? void 0 : e.includes("hover")) || ((t = this.popperShowTriggers) == null ? void 0 : t.includes("hover"));
    }
  },
  watch: {
    shown: "$_autoShowHide",
    disabled(e) {
      e ? this.dispose() : this.init();
    },
    async container() {
      this.isShown && (this.$_ensureTeleport(), await this.$_computePosition());
    },
    triggers: {
      handler: "$_refreshListeners",
      deep: !0
    },
    positioningDisabled: "$_refreshListeners",
    ...[
      "placement",
      "distance",
      "skidding",
      "boundary",
      "strategy",
      "overflowPadding",
      "arrowPadding",
      "preventOverflow",
      "shift",
      "shiftCrossAxis",
      "flip"
    ].reduce((e, t) => (e[t] = "$_computePosition", e), {})
  },
  created() {
    this.autoMinSize && console.warn('[floating-ui] `autoMinSize` option is deprecated. Use `autoSize="min"` instead.'), this.autoMaxSize && console.warn("[floating-ui] `autoMaxSize` option is deprecated. Use `autoBoundaryMaxSize` instead.");
  },
  mounted() {
    this.init(), this.$_detachPopperNode();
  },
  activated() {
    this.$_autoShowHide();
  },
  deactivated() {
    this.hide();
  },
  beforeUnmount() {
    this.dispose();
  },
  methods: {
    show({ event: e = null, skipDelay: t = !1, force: n = !1 } = {}) {
      var o, i;
      (o = this.parentPopper) != null && o.lockedChild && this.parentPopper.lockedChild !== this || (this.pendingHide = !1, (n || !this.disabled) && (((i = this.parentPopper) == null ? void 0 : i.lockedChild) === this && (this.parentPopper.lockedChild = null), this.$_scheduleShow(e, t), this.$emit("show"), this.$_showFrameLocked = !0, requestAnimationFrame(() => {
        this.$_showFrameLocked = !1;
      })), this.$emit("update:shown", !0));
    },
    hide({ event: e = null, skipDelay: t = !1 } = {}) {
      var n;
      if (!this.$_hideInProgress) {
        if (this.shownChildren.size > 0) {
          this.pendingHide = !0;
          return;
        }
        if (this.hasPopperShowTriggerHover && this.$_isAimingPopper()) {
          this.parentPopper && (this.parentPopper.lockedChild = this, clearTimeout(this.parentPopper.lockedChildTimer), this.parentPopper.lockedChildTimer = setTimeout(() => {
            this.parentPopper.lockedChild === this && (this.parentPopper.lockedChild.hide({ skipDelay: t }), this.parentPopper.lockedChild = null);
          }, 1e3));
          return;
        }
        ((n = this.parentPopper) == null ? void 0 : n.lockedChild) === this && (this.parentPopper.lockedChild = null), this.pendingHide = !1, this.$_scheduleHide(e, t), this.$emit("hide"), this.$emit("update:shown", !1);
      }
    },
    init() {
      var e;
      this.isDisposed && (this.isDisposed = !1, this.isMounted = !1, this.$_events = [], this.$_preventShow = !1, this.$_referenceNode = ((e = this.referenceNode) == null ? void 0 : e.call(this)) ?? this.$el, this.$_targetNodes = this.targetNodes().filter((t) => t.nodeType === t.ELEMENT_NODE), this.$_popperNode = this.popperNode(), this.$_innerNode = this.$_popperNode.querySelector(".v-popper__inner"), this.$_arrowNode = this.$_popperNode.querySelector(".v-popper__arrow-container"), this.$_swapTargetAttrs("title", "data-original-title"), this.$_detachPopperNode(), this.triggers.length && this.$_addEventListeners(), this.shown && this.show());
    },
    dispose() {
      this.isDisposed || (this.isDisposed = !0, this.$_removeEventListeners(), this.hide({ skipDelay: !0 }), this.$_detachPopperNode(), this.isMounted = !1, this.isShown = !1, this.$_updateParentShownChildren(!1), this.$_swapTargetAttrs("data-original-title", "title"));
    },
    async onResize() {
      this.isShown && (await this.$_computePosition(), this.$emit("resize"));
    },
    async $_computePosition() {
      if (this.isDisposed || this.positioningDisabled)
        return;
      const e = {
        strategy: this.strategy,
        middleware: []
      };
      (this.distance || this.skidding) && e.middleware.push(Va({
        mainAxis: this.distance,
        crossAxis: this.skidding
      }));
      const t = this.placement.startsWith("auto");
      if (t ? e.middleware.push(Xa({
        alignment: this.placement.split("-")[1] ?? ""
      })) : e.placement = this.placement, this.preventOverflow && (this.shift && e.middleware.push(Za({
        padding: this.overflowPadding,
        boundary: this.boundary,
        crossAxis: this.shiftCrossAxis
      })), !t && this.flip && e.middleware.push(Qa({
        padding: this.overflowPadding,
        boundary: this.boundary
      }))), e.middleware.push(tf({
        element: this.$_arrowNode,
        padding: this.arrowPadding
      })), this.arrowOverflow && e.middleware.push({
        name: "arrowOverflow",
        fn: ({ placement: o, rects: i, middlewareData: s }) => {
          let r;
          const { centerOffset: l } = s.arrow;
          return o.startsWith("top") || o.startsWith("bottom") ? r = Math.abs(l) > i.reference.width / 2 : r = Math.abs(l) > i.reference.height / 2, {
            data: {
              overflow: r
            }
          };
        }
      }), this.autoMinSize || this.autoSize) {
        const o = this.autoSize ? this.autoSize : this.autoMinSize ? "min" : null;
        e.middleware.push({
          name: "autoSize",
          fn: ({ rects: i, placement: s, middlewareData: r }) => {
            var a;
            if ((a = r.autoSize) != null && a.skip)
              return {};
            let l, u;
            return s.startsWith("top") || s.startsWith("bottom") ? l = i.reference.width : u = i.reference.height, this.$_innerNode.style[o === "min" ? "minWidth" : o === "max" ? "maxWidth" : "width"] = l != null ? `${l}px` : null, this.$_innerNode.style[o === "min" ? "minHeight" : o === "max" ? "maxHeight" : "height"] = u != null ? `${u}px` : null, {
              data: {
                skip: !0
              },
              reset: {
                rects: !0
              }
            };
          }
        });
      }
      (this.autoMaxSize || this.autoBoundaryMaxSize) && (this.$_innerNode.style.maxWidth = null, this.$_innerNode.style.maxHeight = null, e.middleware.push(ef({
        boundary: this.boundary,
        padding: this.overflowPadding,
        apply: ({ availableWidth: o, availableHeight: i }) => {
          this.$_innerNode.style.maxWidth = o != null ? `${o}px` : null, this.$_innerNode.style.maxHeight = i != null ? `${i}px` : null;
        }
      })));
      const n = await nf(this.$_referenceNode, this.$_popperNode, e);
      Object.assign(this.result, {
        x: n.x,
        y: n.y,
        placement: n.placement,
        strategy: n.strategy,
        arrow: {
          ...n.middlewareData.arrow,
          ...n.middlewareData.arrowOverflow
        }
      });
    },
    $_scheduleShow(e, t = !1) {
      if (this.$_updateParentShownChildren(!0), this.$_hideInProgress = !1, clearTimeout(this.$_scheduleTimer), gt && this.instantMove && gt.instantMove && gt !== this.parentPopper) {
        gt.$_applyHide(!0), this.$_applyShow(!0);
        return;
      }
      t ? this.$_applyShow() : this.$_scheduleTimer = setTimeout(this.$_applyShow.bind(this), this.$_computeDelay("show"));
    },
    $_scheduleHide(e, t = !1) {
      if (this.shownChildren.size > 0) {
        this.pendingHide = !0;
        return;
      }
      this.$_updateParentShownChildren(!1), this.$_hideInProgress = !0, clearTimeout(this.$_scheduleTimer), this.isShown && (gt = this), t ? this.$_applyHide() : this.$_scheduleTimer = setTimeout(this.$_applyHide.bind(this), this.$_computeDelay("hide"));
    },
    $_computeDelay(e) {
      const t = this.delay;
      return parseInt(t && t[e] || t || 0);
    },
    async $_applyShow(e = !1) {
      clearTimeout(this.$_disposeTimer), clearTimeout(this.$_scheduleTimer), this.skipTransition = e, !this.isShown && (this.$_ensureTeleport(), await So(), await this.$_computePosition(), await this.$_applyShowEffect(), this.positioningDisabled || this.$_registerEventListeners([
        ...fn(this.$_referenceNode),
        ...fn(this.$_popperNode)
      ], "scroll", () => {
        this.$_computePosition();
      }));
    },
    async $_applyShowEffect() {
      if (this.$_hideInProgress)
        return;
      if (this.computeTransformOrigin) {
        const t = this.$_referenceNode.getBoundingClientRect(), n = this.$_popperNode.querySelector(".v-popper__wrapper"), o = n.parentNode.getBoundingClientRect(), i = t.x + t.width / 2 - (o.left + n.offsetLeft), s = t.y + t.height / 2 - (o.top + n.offsetTop);
        this.result.transformOrigin = `${i}px ${s}px`;
      }
      this.isShown = !0, this.$_applyAttrsToTarget({
        "aria-describedby": this.popperId,
        "data-popper-shown": ""
      });
      const e = this.showGroup;
      if (e) {
        let t;
        for (let n = 0; n < Ee.length; n++)
          t = Ee[n], t.showGroup !== e && (t.hide(), t.$emit("close-group"));
      }
      Ee.push(this), document.body.classList.add("v-popper--some-open");
      for (const t of Cs(this.theme))
        Ts(t).push(this), document.body.classList.add(`v-popper--some-open--${t}`);
      this.$emit("apply-show"), this.classes.showFrom = !0, this.classes.showTo = !1, this.classes.hideFrom = !1, this.classes.hideTo = !1, await So(), this.classes.showFrom = !1, this.classes.showTo = !0, this.noAutoFocus || this.$_popperNode.focus();
    },
    async $_applyHide(e = !1) {
      if (this.shownChildren.size > 0) {
        this.pendingHide = !0, this.$_hideInProgress = !1;
        return;
      }
      if (clearTimeout(this.$_scheduleTimer), !this.isShown)
        return;
      this.skipTransition = e, $s(Ee, this), Ee.length === 0 && document.body.classList.remove("v-popper--some-open");
      for (const n of Cs(this.theme)) {
        const o = Ts(n);
        $s(o, this), o.length === 0 && document.body.classList.remove(`v-popper--some-open--${n}`);
      }
      gt === this && (gt = null), this.isShown = !1, this.$_applyAttrsToTarget({
        "aria-describedby": void 0,
        "data-popper-shown": void 0
      }), clearTimeout(this.$_disposeTimer);
      const t = this.disposeTimeout;
      t !== null && (this.$_disposeTimer = setTimeout(() => {
        this.$_popperNode && (this.$_detachPopperNode(), this.isMounted = !1);
      }, t)), this.$_removeEventListeners("scroll"), this.$emit("apply-hide"), this.classes.showFrom = !1, this.classes.showTo = !1, this.classes.hideFrom = !0, this.classes.hideTo = !1, await So(), this.classes.hideFrom = !1, this.classes.hideTo = !0;
    },
    $_autoShowHide() {
      this.shown ? this.show() : this.hide();
    },
    $_ensureTeleport() {
      if (this.isDisposed)
        return;
      let e = this.container;
      if (typeof e == "string" ? e = window.document.querySelector(e) : e === !1 && (e = this.$_targetNodes[0].parentNode), !e)
        throw new Error("No container for popover: " + this.container);
      e.appendChild(this.$_popperNode), this.isMounted = !0;
    },
    $_addEventListeners() {
      const e = (n) => {
        this.isShown && !this.$_hideInProgress || (n.usedByTooltip = !0, !this.$_preventShow && this.show({ event: n }));
      };
      this.$_registerTriggerListeners(this.$_targetNodes, Os, this.triggers, this.showTriggers, e), this.$_registerTriggerListeners([this.$_popperNode], Os, this.popperTriggers, this.popperShowTriggers, e);
      const t = (n) => {
        n.usedByTooltip || this.hide({ event: n });
      };
      this.$_registerTriggerListeners(this.$_targetNodes, xs, this.triggers, this.hideTriggers, t), this.$_registerTriggerListeners([this.$_popperNode], xs, this.popperTriggers, this.popperHideTriggers, t);
    },
    $_registerEventListeners(e, t, n) {
      this.$_events.push({ targetNodes: e, eventType: t, handler: n }), e.forEach((o) => o.addEventListener(t, n, pn ? {
        passive: !0
      } : void 0));
    },
    $_registerTriggerListeners(e, t, n, o, i) {
      let s = n;
      o != null && (s = typeof o == "function" ? o(s) : o), s.forEach((r) => {
        const l = t[r];
        l && this.$_registerEventListeners(e, l, i);
      });
    },
    $_removeEventListeners(e) {
      const t = [];
      this.$_events.forEach((n) => {
        const { targetNodes: o, eventType: i, handler: s } = n;
        !e || e === i ? o.forEach((r) => r.removeEventListener(i, s)) : t.push(n);
      }), this.$_events = t;
    },
    $_refreshListeners() {
      this.isDisposed || (this.$_removeEventListeners(), this.$_addEventListeners());
    },
    $_handleGlobalClose(e, t = !1) {
      this.$_showFrameLocked || (this.hide({ event: e }), e.closePopover ? this.$emit("close-directive") : this.$emit("auto-hide"), t && (this.$_preventShow = !0, setTimeout(() => {
        this.$_preventShow = !1;
      }, 300)));
    },
    $_detachPopperNode() {
      this.$_popperNode.parentNode && this.$_popperNode.parentNode.removeChild(this.$_popperNode);
    },
    $_swapTargetAttrs(e, t) {
      for (const n of this.$_targetNodes) {
        const o = n.getAttribute(e);
        o && (n.removeAttribute(e), n.setAttribute(t, o));
      }
    },
    $_applyAttrsToTarget(e) {
      for (const t of this.$_targetNodes)
        for (const n in e) {
          const o = e[n];
          o == null ? t.removeAttribute(n) : t.setAttribute(n, o);
        }
    },
    $_updateParentShownChildren(e) {
      let t = this.parentPopper;
      for (; t; )
        e ? t.shownChildren.add(this.randomId) : (t.shownChildren.delete(this.randomId), t.pendingHide && t.hide()), t = t.parentPopper;
    },
    $_isAimingPopper() {
      const e = this.$_referenceNode.getBoundingClientRect();
      if (on >= e.left && on <= e.right && sn >= e.top && sn <= e.bottom) {
        const t = this.$_popperNode.getBoundingClientRect(), n = on - Qe, o = sn - et, s = t.left + t.width / 2 - Qe + (t.top + t.height / 2) - et + t.width + t.height, r = Qe + n * s, l = et + o * s;
        return Vn(Qe, et, r, l, t.left, t.top, t.left, t.bottom) || // Left edge
        Vn(Qe, et, r, l, t.left, t.top, t.right, t.top) || // Top edge
        Vn(Qe, et, r, l, t.right, t.top, t.right, t.bottom) || // Right edge
        Vn(Qe, et, r, l, t.left, t.bottom, t.right, t.bottom);
      }
      return !1;
    }
  },
  render() {
    return this.$slots.default(this.slotData);
  }
});
if (typeof document < "u" && typeof window < "u") {
  if (Qr) {
    const e = pn ? {
      passive: !0,
      capture: !0
    } : !0;
    document.addEventListener("touchstart", (t) => Ss(t), e), document.addEventListener("touchend", (t) => Vs(t, !0), e);
  } else
    window.addEventListener("mousedown", (e) => Ss(e), !0), window.addEventListener("click", (e) => Vs(e, !1), !0);
  window.addEventListener("resize", cf);
}
function Ss(e, t) {
  for (let n = 0; n < Ee.length; n++) {
    const o = Ee[n];
    try {
      o.mouseDownContains = o.popperNode().contains(e.target);
    } catch {
    }
  }
}
function Vs(e, t) {
  rf(e, t);
}
function rf(e, t) {
  const n = {};
  for (let o = Ee.length - 1; o >= 0; o--) {
    const i = Ee[o];
    try {
      const s = i.containsGlobalTarget = i.mouseDownContains || i.popperNode().contains(e.target);
      i.pendingHide = !1, requestAnimationFrame(() => {
        if (i.pendingHide = !1, !n[i.randomId] && Ps(i, s, e)) {
          if (i.$_handleGlobalClose(e, t), !e.closeAllPopover && e.closePopover && s) {
            let l = i.parentPopper;
            for (; l; )
              n[l.randomId] = !0, l = l.parentPopper;
            return;
          }
          let r = i.parentPopper;
          for (; r && Ps(r, r.containsGlobalTarget, e); ) {
            r.$_handleGlobalClose(e, t);
            r = r.parentPopper;
          }
        }
      });
    } catch {
    }
  }
}
function Ps(e, t, n) {
  return n.closeAllPopover || n.closePopover && t || lf(e, n) && !t;
}
function lf(e, t) {
  if (typeof e.autoHide == "function") {
    const n = e.autoHide(t);
    return e.lastAutoHide = n, n;
  }
  return e.autoHide;
}
function cf() {
  for (let e = 0; e < Ee.length; e++)
    Ee[e].$_computePosition();
}
function Pf() {
  for (let e = 0; e < Ee.length; e++)
    Ee[e].hide();
}
let Qe = 0, et = 0, on = 0, sn = 0;
typeof window < "u" && window.addEventListener("mousemove", (e) => {
  Qe = on, et = sn, on = e.clientX, sn = e.clientY;
}, pn ? {
  passive: !0
} : void 0);
function Vn(e, t, n, o, i, s, r, l) {
  const u = ((r - i) * (t - s) - (l - s) * (e - i)) / ((l - s) * (n - e) - (r - i) * (o - t)), a = ((n - e) * (t - s) - (o - t) * (e - i)) / ((l - s) * (n - e) - (r - i) * (o - t));
  return u >= 0 && u <= 1 && a >= 0 && a <= 1;
}
function uf() {
  var e = window.navigator.userAgent, t = e.indexOf("MSIE ");
  if (t > 0)
    return parseInt(e.substring(t + 5, e.indexOf(".", t)), 10);
  var n = e.indexOf("Trident/");
  if (n > 0) {
    var o = e.indexOf("rv:");
    return parseInt(e.substring(o + 3, e.indexOf(".", o)), 10);
  }
  var i = e.indexOf("Edge/");
  return i > 0 ? parseInt(e.substring(i + 5, e.indexOf(".", i)), 10) : -1;
}
let Hn;
function ni() {
  ni.init || (ni.init = !0, Hn = uf() !== -1);
}
var go = {
  name: "ResizeObserver",
  props: {
    emitOnMount: {
      type: Boolean,
      default: !1
    },
    ignoreWidth: {
      type: Boolean,
      default: !1
    },
    ignoreHeight: {
      type: Boolean,
      default: !1
    }
  },
  emits: [
    "notify"
  ],
  mounted() {
    ni(), rr(() => {
      this._w = this.$el.offsetWidth, this._h = this.$el.offsetHeight, this.emitOnMount && this.emitSize();
    });
    const e = document.createElement("object");
    this._resizeObject = e, e.setAttribute("aria-hidden", "true"), e.setAttribute("tabindex", -1), e.onload = this.addResizeHandlers, e.type = "text/html", Hn && this.$el.appendChild(e), e.data = "about:blank", Hn || this.$el.appendChild(e);
  },
  beforeUnmount() {
    this.removeResizeHandlers();
  },
  methods: {
    compareAndNotify() {
      (!this.ignoreWidth && this._w !== this.$el.offsetWidth || !this.ignoreHeight && this._h !== this.$el.offsetHeight) && (this._w = this.$el.offsetWidth, this._h = this.$el.offsetHeight, this.emitSize());
    },
    emitSize() {
      this.$emit("notify", {
        width: this._w,
        height: this._h
      });
    },
    addResizeHandlers() {
      this._resizeObject.contentDocument.defaultView.addEventListener("resize", this.compareAndNotify), this.compareAndNotify();
    },
    removeResizeHandlers() {
      this._resizeObject && this._resizeObject.onload && (!Hn && this._resizeObject.contentDocument && this._resizeObject.contentDocument.defaultView.removeEventListener("resize", this.compareAndNotify), this.$el.removeChild(this._resizeObject), this._resizeObject.onload = null, this._resizeObject = null);
    }
  }
};
const af = /* @__PURE__ */ vc();
Ec("data-v-b329ee4c");
const ff = {
  class: "resize-observer",
  tabindex: "-1"
};
bc();
const df = /* @__PURE__ */ af((e, t, n, o, i, s) => (Ke(), En("div", ff)));
go.render = df;
go.__scopeId = "data-v-b329ee4c";
go.__file = "src/components/ResizeObserver.vue";
const pf = (e = "theme") => ({
  computed: {
    themeClass() {
      return of(this[e]);
    }
  }
}), hf = /* @__PURE__ */ wi({
  name: "VPopperContent",
  components: {
    ResizeObserver: go
  },
  mixins: [
    pf()
  ],
  props: {
    popperId: String,
    theme: String,
    shown: Boolean,
    mounted: Boolean,
    skipTransition: Boolean,
    autoHide: Boolean,
    handleResize: Boolean,
    classes: Object,
    result: Object
  },
  emits: [
    "hide",
    "resize"
  ],
  methods: {
    toPx(e) {
      return e != null && !isNaN(e) ? `${e}px` : null;
    }
  }
}), tl = (e, t) => {
  const n = e.__vccOpts || e;
  for (const [o, i] of t)
    n[o] = i;
  return n;
}, mf = ["id", "aria-hidden", "tabindex", "data-popper-placement"], gf = {
  ref: "inner",
  class: "v-popper__inner"
}, _f = /* @__PURE__ */ We("div", { class: "v-popper__arrow-outer" }, null, -1), yf = /* @__PURE__ */ We("div", { class: "v-popper__arrow-inner" }, null, -1), wf = [
  _f,
  yf
];
function Ef(e, t, n, o, i, s) {
  const r = zo("ResizeObserver");
  return Ke(), Gn("div", {
    id: e.popperId,
    ref: "popover",
    class: gn(["v-popper__popper", [
      e.themeClass,
      e.classes.popperClass,
      {
        "v-popper__popper--shown": e.shown,
        "v-popper__popper--hidden": !e.shown,
        "v-popper__popper--show-from": e.classes.showFrom,
        "v-popper__popper--show-to": e.classes.showTo,
        "v-popper__popper--hide-from": e.classes.hideFrom,
        "v-popper__popper--hide-to": e.classes.hideTo,
        "v-popper__popper--skip-transition": e.skipTransition,
        "v-popper__popper--arrow-overflow": e.result && e.result.arrow.overflow,
        "v-popper__popper--no-positioning": !e.result
      }
    ]]),
    style: Ht(e.result ? {
      position: e.result.strategy,
      transform: `translate3d(${Math.round(e.result.x)}px,${Math.round(e.result.y)}px,0)`
    } : void 0),
    "aria-hidden": e.shown ? "false" : "true",
    tabindex: e.autoHide ? 0 : void 0,
    "data-popper-placement": e.result ? e.result.placement : void 0,
    onKeyup: t[2] || (t[2] = ua((l) => e.autoHide && e.$emit("hide"), ["esc"]))
  }, [
    We("div", {
      class: "v-popper__backdrop",
      onClick: t[0] || (t[0] = (l) => e.autoHide && e.$emit("hide"))
    }),
    We("div", {
      class: "v-popper__wrapper",
      style: Ht(e.result ? {
        transformOrigin: e.result.transformOrigin
      } : void 0)
    }, [
      We("div", gf, [
        e.mounted ? (Ke(), Gn(we, { key: 0 }, [
          We("div", null, [
            Kc(e.$slots, "default")
          ]),
          e.handleResize ? (Ke(), En(r, {
            key: 0,
            onNotify: t[1] || (t[1] = (l) => e.$emit("resize", l))
          })) : is("", !0)
        ], 64)) : is("", !0)
      ], 512),
      We("div", {
        ref: "arrow",
        class: "v-popper__arrow-container",
        style: Ht(e.result ? {
          left: e.toPx(e.result.arrow.x),
          top: e.toPx(e.result.arrow.y)
        } : void 0)
      }, wf, 4)
    ], 4)
  ], 46, mf);
}
const bf = /* @__PURE__ */ tl(hf, [["render", Ef]]), vf = {
  methods: {
    show(...e) {
      return this.$refs.popper.show(...e);
    },
    hide(...e) {
      return this.$refs.popper.hide(...e);
    },
    dispose(...e) {
      return this.$refs.popper.dispose(...e);
    },
    onResize(...e) {
      return this.$refs.popper.onResize(...e);
    }
  }
}, Nf = /* @__PURE__ */ wi({
  name: "VTooltipDirective",
  components: {
    Popper: sf(),
    PopperContent: bf
  },
  mixins: [
    vf
  ],
  inheritAttrs: !1,
  props: {
    theme: {
      type: String,
      default: "tooltip"
    },
    html: {
      type: Boolean,
      default: (e) => Qn(e.theme, "html")
    },
    content: {
      type: [String, Number, Function],
      default: null
    },
    loadingContent: {
      type: String,
      default: (e) => Qn(e.theme, "loadingContent")
    },
    targetNodes: {
      type: Function,
      required: !0
    }
  },
  data() {
    return {
      asyncContent: null
    };
  },
  computed: {
    isContentAsync() {
      return typeof this.content == "function";
    },
    loading() {
      return this.isContentAsync && this.asyncContent == null;
    },
    finalContent() {
      return this.isContentAsync ? this.loading ? this.loadingContent : this.asyncContent : this.content;
    }
  },
  watch: {
    content: {
      handler() {
        this.fetchContent(!0);
      },
      immediate: !0
    },
    async finalContent() {
      await this.$nextTick(), this.$refs.popper.onResize();
    }
  },
  created() {
    this.$_fetchId = 0;
  },
  methods: {
    fetchContent(e) {
      if (typeof this.content == "function" && this.$_isShown && (e || !this.$_loading && this.asyncContent == null)) {
        this.asyncContent = null, this.$_loading = !0;
        const t = ++this.$_fetchId, n = this.content(this);
        n.then ? n.then((o) => this.onResult(t, o)) : this.onResult(t, n);
      }
    },
    onResult(e, t) {
      e === this.$_fetchId && (this.$_loading = !1, this.asyncContent = t);
    },
    onShow() {
      this.$_isShown = !0, this.fetchContent();
    },
    onHide() {
      this.$_isShown = !1;
    }
  }
}), Of = ["innerHTML"], xf = ["textContent"];
function $f(e, t, n, o, i, s) {
  const r = zo("PopperContent"), l = zo("Popper");
  return Ke(), En(l, Lr({ ref: "popper" }, e.$attrs, {
    theme: e.theme,
    "target-nodes": e.targetNodes,
    "popper-node": () => e.$refs.popperContent.$el,
    onApplyShow: e.onShow,
    onApplyHide: e.onHide
  }), {
    default: zn(({
      popperId: u,
      isShown: a,
      shouldMountContent: p,
      skipTransition: f,
      autoHide: h,
      hide: g,
      handleResize: O,
      onResize: $,
      classes: V,
      result: P
    }) => [
      ge(r, {
        ref: "popperContent",
        class: gn({
          "v-popper--tooltip-loading": e.loading
        }),
        "popper-id": u,
        theme: e.theme,
        shown: a,
        mounted: p,
        "skip-transition": f,
        "auto-hide": h,
        "handle-resize": O,
        classes: V,
        result: P,
        onHide: g,
        onResize: $
      }, {
        default: zn(() => [
          e.html ? (Ke(), Gn("div", {
            key: 0,
            innerHTML: e.finalContent
          }, null, 8, Of)) : (Ke(), Gn("div", {
            key: 1,
            textContent: vl(e.finalContent)
          }, null, 8, xf))
        ]),
        _: 2
      }, 1032, ["class", "popper-id", "theme", "shown", "mounted", "skip-transition", "auto-hide", "handle-resize", "classes", "result", "onHide", "onResize"])
    ]),
    _: 1
  }, 16, ["theme", "target-nodes", "popper-node", "onApplyShow", "onApplyHide"]);
}
const Cf = /* @__PURE__ */ tl(Nf, [["render", $f]]), nl = "v-popper--has-tooltip";
function Df(e, t) {
  let n = e.placement;
  if (!n && t)
    for (const o of el)
      t[o] && (n = o);
  return n || (n = Qn(e.theme || "tooltip", "placement")), n;
}
function Tf(e, t, n) {
  let o;
  const i = typeof t;
  return i === "string" ? o = { content: t } : t && i === "object" ? o = t : o = { content: !1 }, o.placement = Df(o, n), o.targetNodes = () => [e], o.referenceNode = () => e, o;
}
let Po, hn, Sf = 0;
function Vf() {
  if (Po)
    return;
  hn = Fo([]), Po = da({
    name: "VTooltipDirectiveApp",
    setup() {
      return {
        directives: hn
      };
    },
    render() {
      return this.directives.map((t) => Fu(Cf, {
        ...t.options,
        shown: t.shown || t.options.shown,
        key: t.id
      }));
    },
    devtools: {
      hide: !0
    }
  });
  const e = document.createElement("div");
  document.body.appendChild(e), Po.mount(e);
}
function Af(e, t, n) {
  Vf();
  const o = Fo(Tf(e, t, n)), i = Fo(!1), s = {
    id: Sf++,
    options: o,
    shown: i
  };
  return hn.value.push(s), e.classList && e.classList.add(nl), e.$_popper = {
    options: o,
    item: s,
    show() {
      i.value = !0;
    },
    hide() {
      i.value = !1;
    }
  };
}
function Rf(e) {
  if (e.$_popper) {
    const t = hn.value.indexOf(e.$_popper.item);
    t !== -1 && hn.value.splice(t, 1), delete e.$_popper, delete e.$_popperOldShown, delete e.$_popperMountTarget;
  }
  e.classList && e.classList.remove(nl);
}
export {
  Af as createTooltip,
  Rf as destroyTooltip,
  Pf as hideAllPoppers,
  cf as recomputeAllPoppers
};
